diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6931cda9d..f4138802a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -227,8 +227,6 @@ message(STATUS "Target architecture: ${ARCHITECTURE}")
 # Configure C++ standard
 # ===========================
 
-# boost asio's concept usage doesn't play nicely with some compilers yet.
-add_definitions(-DBOOST_ASIO_DISABLE_CONCEPTS)
 # boost can have issues compiling with C++17 and up on newer versions of Clang.
 add_definitions(-DBOOST_NO_CXX98_FUNCTION_BASE)
 set(CMAKE_CXX_STANDARD 20)
