diff -ur pam-python-1.0.6.orig/src/pam_python.c pam-python-1.0.6/src/pam_python.c
--- pam-python-1.0.6.orig/src/pam_python.c	2012-04-12 05:35:24.000000000 +0200
+++ pam-python-1.0.6/src/pam_python.c	2012-10-19 11:25:06.570212299 +0200
@@ -32,6 +32,7 @@
 #endif
 
 #undef	_POSIX_C_SOURCE
+#undef	_XOPEN_SOURCE
 
 #include <Python.h>
 #include <dlfcn.h>
@@ -1496,7 +1497,7 @@
       pamHandle->pamh, PAM_XAUTHDATA, (const void**)&xauth_data);
   if (check_pam_result(pamHandle, pam_result) == -1)
     goto error_exit;
-  if (xauth_data == 0)
+  if (xauth_data == 0 || (xauth_data->namelen == 0 && xauth_data->datalen == 0))
   {
     result = Py_None;
     Py_INCREF(result);
diff -ur pam-python-1.0.6.orig/src/setup.py pam-python-1.0.6/src/setup.py
--- pam-python-1.0.6.orig/src/pam_python.c.orig	2018-01-17 13:11:22.910978900 +0100
+++ pam-python-1.0.6/src/pam_python.c	2018-01-17 13:13:28.710991740 +0100
@@ -2174,7 +2174,7 @@
     goto error_exit;
   }
   dot = strrchr(user_module_name, '.');
-  if (dot != 0 || strcmp(dot, ".py") == 0)
+  if (dot != 0 && strcmp(dot, ".py") == 0)
     *dot = '\0';
   *user_module = PyModule_New(user_module_name);
   if (*user_module == 0)
