diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 10533571..e17b1387 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -337,7 +337,7 @@ include(CreateDirectoryGroups)
 create_target_directory_groups(dynarmic)
 
 target_include_directories(dynarmic
-                           PUBLIC ../include
+                           PUBLIC ../include ../externals/xbyak/xbyak
                            PRIVATE .)
 target_compile_options(dynarmic PRIVATE ${DYNARMIC_CXX_FLAGS})
 target_link_libraries(dynarmic
