From 78759d98dc6cf20d770e0aadc87efdc500de4359 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Mon, 20 Dec 2021 19:20:26 -0500
Subject: [PATCH] Add support for linking against system opus

---
 CMakeLists.txt           | 2 ++
 externals/CMakeLists.txt | 6 +++---
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a810e11c22a..1a7b1d85ab6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,6 +33,8 @@ option(ENABLE_CUBEB "Enables the cubeb audio backend" ON)
 
 option(USE_DISCORD_PRESENCE "Enables Discord Rich Presence" OFF)
 
+option(YUZU_USE_BUNDLED_OPUS "Compile bundled opus" ON)
+
 # Default to a Release build
 get_property(IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
 if (NOT IS_MULTI_CONFIG AND NOT CMAKE_BUILD_TYPE)
diff --git a/externals/CMakeLists.txt b/externals/CMakeLists.txt
index 64d1e6aec49..26ead64f267 100644
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -116,8 +116,8 @@ if (ENABLE_WEB_SERVICE)
 endif()
 
 # Opus
-find_package(opus 1.3)
-if (NOT opus_FOUND)
-    message(STATUS "opus 1.3 or newer not found, falling back to externals")
+if (YUZU_USE_BUNDLED_OPUS)
     add_subdirectory(opus EXCLUDE_FROM_ALL)
+else()
+    find_package(opus 1.3 REQUIRED)
 endif()
