--- b/src/video_core/vulkan_common/vulkan_wrapper.cpp
+++ a/src/video_core/vulkan_common/vulkan_wrapper.cpp
@@ -325,8 +325,6 @@
         return "VK_PIPELINE_COMPILE_REQUIRED_EXT";
     case VkResult::VK_RESULT_MAX_ENUM:
         return "VK_RESULT_MAX_ENUM";
-    case VkResult::VK_ERROR_COMPRESSION_EXHAUSTED_EXT:
-        return "VK_ERROR_COMPRESSION_EXHAUSTED_EXT";
     }
     return "Unknown";
 }
