From c4541bd0016dc013e71ba9b9de18e5075906feae Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Mon, 18 Jan 2021 22:14:22 -0500
Subject: [PATCH 2/5] Fix libzip typo

---
 externals/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/externals/CMakeLists.txt b/externals/CMakeLists.txt
index 9eba12c0c..50ed8d1c7 100644
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -70,7 +70,7 @@ add_subdirectory(sirit)
 
 # libzip
 find_package(Libzip 1.5)
-if (NOT LIBZIP_FOUND)
+if (NOT Libzip_FOUND)
     message(STATUS "libzip 1.5 or newer not found, falling back to externals")
     add_subdirectory(libzip EXCLUDE_FROM_ALL)
 endif()
-- 
2.30.0

