diff --git a/externals/CMakeLists.txt b/externals/CMakeLists.txt
index a76a3d800..814d654c9 100644
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -59,8 +59,9 @@ if (YUZU_USE_EXTERNAL_SDL2)
         # Yuzu itself needs: Events Joystick Haptic Sensor Timers Audio
         # Yuzu-cmd also needs: Video (depends on Loadso/Dlopen)
         set(SDL_UNUSED_SUBSYSTEMS
-            Atomic Render Power Threads
-            File CPUinfo Filesystem Locale)
+            Render Power
+            File CPUinfo Filesystem Locale
+            Wayland)
         foreach(_SUB ${SDL_UNUSED_SUBSYSTEMS})
           string(TOUPPER ${_SUB} _OPT)
           option(SDL_${_OPT} "" OFF)
