From d9dc483661328c90f280ce30af073b7e3b663b93 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Mon, 18 Jan 2021 23:45:19 -0500
Subject: [PATCH 6/6] Allow system cubeb

---
 CMakeLists.txt           | 1 +
 externals/CMakeLists.txt | 8 ++++++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 825ea832c..b8f19be74 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,6 +28,7 @@ option(USE_DISCORD_PRESENCE "Enables Discord Rich Presence" OFF)
 
 option(USE_SYSTEM_OPUS "Enable system opus" OFF)
 option(USE_SYSTEM_INIH "Enable system inih" OFF)
+option(USE_SYSTEM_CUBEB "Enable system cubeb" OFF)
 
 if (NOT ENABLE_WEB_SERVICE)
     set(YUZU_ENABLE_BOXCAT OFF)
diff --git a/externals/CMakeLists.txt b/externals/CMakeLists.txt
index 50ed8d1c7..54295bb4f 100644
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -55,8 +55,12 @@ add_subdirectory(soundtouch)
 
 # Cubeb
 if(ENABLE_CUBEB)
-    set(BUILD_TESTS OFF CACHE BOOL "")
-    add_subdirectory(cubeb EXCLUDE_FROM_ALL)
+    if (NOT USE_SYSTEM_CUBEB)
+        set(BUILD_TESTS OFF CACHE BOOL "")
+        add_subdirectory(cubeb EXCLUDE_FROM_ALL)
+    else()
+        find_package(cubeb REQUIRED)
+    endif()
 endif()
 
 # DiscordRPC
-- 
2.30.0

