diff --git a/build_cmake/Framework/CMakeLists.txt b/build_cmake/Framework/CMakeLists.txt
index 0109c6d..65cc58c 100644
--- a/build_cmake/Framework/CMakeLists.txt
+++ b/build_cmake/Framework/CMakeLists.txt
@@ -19,14 +19,14 @@ if(NOT TARGET ghc_filesystem)
 endif()
 LIST(APPEND PROJECT_LIBS ghc_filesystem)
 
-find_package(ICUUC)
+find_package(ICUUC REQUIRED)
 if(ICUUC_FOUND)
 	LIST(APPEND PROJECT_INCLUDES ${ICUUC_INCLUDE_DIRS})
 	LIST(APPEND PROJECT_LIBS ${ICUUC_LIBRARIES})
 	add_definitions(-DHAS_ICU)
 endif()
 
-find_package(OpenSSL)
+find_package(OpenSSL REQUIRED)
 if(OPENSSL_FOUND)
 	LIST(APPEND PROJECT_INCLUDES ${OPENSSL_INCLUDE_DIR})
 	LIST(APPEND PROJECT_LIBS ${OPENSSL_LIBRARIES})
@@ -149,7 +149,7 @@ if(ANDROID)
 		../../src/android/javax_crypto_Mac.cpp
 		../../src/android/javax_crypto_spec_SecretKeySpec.cpp
 		../../src/PosixFileStream.cpp
-		
+
 		../../include/android/android_content_ContentResolver.h
 		../../include/android/android_database_Cursor.h
 		../../include/android/android_net_Uri.h
diff --git a/build_cmake/FrameworkOpenGl/CMakeLists.txt b/build_cmake/FrameworkOpenGl/CMakeLists.txt
index c242179..67ab556 100644
--- a/build_cmake/FrameworkOpenGl/CMakeLists.txt
+++ b/build_cmake/FrameworkOpenGl/CMakeLists.txt
@@ -16,7 +16,7 @@ endif()
 
 set(USE_GLEW OFF)
 if(NOT USE_GLES AND NOT APPLE)
-	find_package(GLEW)
+	find_package(GLEW REQUIRED)
 	if(GLEW_FOUND)
 		list(APPEND PROJECT_LIBS GLEW::GLEW)
 	else()
diff --git a/build_cmake/FrameworkSqlite/CMakeLists.txt b/build_cmake/FrameworkSqlite/CMakeLists.txt
index bc05147..9a874c0 100644
--- a/build_cmake/FrameworkSqlite/CMakeLists.txt
+++ b/build_cmake/FrameworkSqlite/CMakeLists.txt
@@ -3,7 +3,7 @@ project(Framework_Sqlite)
 
 include(../Framework_Common.cmake)
 
-find_package(SQLite3)
+find_package(SQLite3 REQUIRED)
 if(SQLite3_FOUND)
 	LIST(APPEND PROJECT_LIBS SQLite::SQLite3)
 else()
