diff --git a/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt b/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt
index c242179..67ab556 100644
--- a/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt
+++ b/deps/Framework/build_cmake/FrameworkOpenGl/CMakeLists.txt
@@ -16,7 +16,7 @@ endif()
 
 set(USE_GLEW OFF)
 if(NOT USE_GLES AND NOT APPLE)
-	find_package(GLEW)
+	find_package(GLEW REQUIRED)
 	if(GLEW_FOUND)
 		list(APPEND PROJECT_LIBS GLEW::GLEW)
 	else()
diff --git a/deps/Framework/build_cmake/FrameworkSqlite/CMakeLists.txt b/deps/Framework/build_cmake/FrameworkSqlite/CMakeLists.txt
index bc05147..9a874c0 100644
--- a/deps/Framework/build_cmake/FrameworkSqlite/CMakeLists.txt
+++ b/deps/Framework/build_cmake/FrameworkSqlite/CMakeLists.txt
@@ -3,7 +3,7 @@ project(Framework_Sqlite)
 
 include(../Framework_Common.cmake)
 
-find_package(SQLite3)
+find_package(SQLite3 REQUIRED)
 if(SQLite3_FOUND)
 	LIST(APPEND PROJECT_LIBS SQLite::SQLite3)
 else()
diff --git a/src/vulkan/Instance.cpp b/src/vulkan/Instance.cpp
index 8a3b7b3..d7870bc 100644
--- a/deps/Framework/src/vulkan/Instance.cpp
+++ b/deps/Framework/src/vulkan/Instance.cpp
@@ -1,6 +1,7 @@
 #include "vulkan/VulkanDef.h"
 #include "vulkan/Instance.h"
 #include "vulkan/Loader.h"
+#include <stdexcept>
 
 #define SET_PROC_ADDR(functionName) this->functionName = reinterpret_cast<PFN_##functionName>(CLoader::GetInstance().vkGetInstanceProcAddr(m_handle, #functionName));
 
diff --git a/deps/Framework/build_cmake/Framework/CMakeLists.txt b/deps/Framework/build_cmake/Framework/CMakeLists.txt
index dbcef49..a9572cd 100644
--- a/deps/Framework/build_cmake/Framework/CMakeLists.txt
+++ b/deps/Framework/build_cmake/Framework/CMakeLists.txt
@@ -19,14 +19,14 @@ endif()
 LIST(APPEND PROJECT_LIBS ghc_filesystem)
 
 if(NOT APPLE)
-	find_package(ICUUC)
+	find_package(ICUUC REQUIRED)
 	if(ICUUC_FOUND)
 		LIST(APPEND PROJECT_INCLUDES ${ICUUC_INCLUDE_DIRS})
 		LIST(APPEND PROJECT_LIBS ${ICUUC_LIBRARIES})
 		add_definitions(-DHAS_ICU)
 	endif()
 
-	find_package(OpenSSL)
+	find_package(OpenSSL REQUIRED)
 	if(OPENSSL_FOUND)
 		LIST(APPEND PROJECT_INCLUDES ${OPENSSL_INCLUDE_DIR})
 		LIST(APPEND PROJECT_LIBS ${OPENSSL_LIBRARIES})
@@ -160,7 +160,7 @@ if(ANDROID)
 		../../src/android/javax_crypto_Mac.cpp
 		../../src/android/javax_crypto_spec_SecretKeySpec.cpp
 		../../src/PosixFileStream.cpp
-		
+
 		../../include/android/android_content_ContentResolver.h
 		../../include/android/android_database_Cursor.h
 		../../include/android/android_net_Uri.h
diff --git a/deps/Framework/tests/TestDefs.h b/deps/Framework/tests/TestDefs.h
index b365c4c..fcacb8f 100644
--- a/deps/Framework/tests/TestDefs.h
+++ b/deps/Framework/tests/TestDefs.h
@@ -1,3 +1,4 @@
 #pragma once
+#include <cstdio>
 
 #define TEST_VERIFY(a) if(!(a)) { printf("Verification failed: '%s'. Aborting.\n", #a); std::abort(); }
