diff --git a/bin/kindle_sync b/bin/kindle_sync
index 56a8474..10aa2bd 100755
--- a/bin/kindle_sync
+++ b/bin/kindle_sync
@@ -115,7 +115,7 @@ class KindleSync
     source_path = Pathname("#{@repo.path + book}.epub")
     target_path = Pathname("#{@device.path + book}.mobi")
     target_path.dirname.mkpath
-    system *%W[/Applications/calibre.app/Contents/MacOS/ebook-convert #{source_path} #{target_path} --output-profile kindle_voyage]
+    system *%W[ebook-convert #{source_path} #{target_path} --output-profile kindle_voyage]
   end
 
   def sync!
@@ -141,7 +141,7 @@ class KindleSync
   def cleanup_book!(book)
     raise if repo_state(book)
     device_state(book).each do |ext|
-      system "trash", "#{@device.path + book}#{ext}"
+      system "rm", "#{@device.path + book}#{ext}"
     end
   end
 
