diff --git a/bin/open_youtube b/bin/open_youtube
index 8e594e6..206055d 100755
--- a/bin/open_youtube
+++ b/bin/open_youtube
@@ -3,5 +3,5 @@
 ARGV.each do |path|
   # This might not be the most reliable Youtube ID extraction regexp ever
   youtube_id = File.basename(path).sub(/\.[^\.]+\z/, "")[/.{11}\z/]
-  system "open", "https://www.youtube.com/watch?v=#{youtube_id}"
+  system "xdg-open", "https://www.youtube.com/watch?v=#{youtube_id}"
 end
