diff --git a/bin/strip_9gag b/bin/strip_9gag
index e49f096..3991924 100755
--- a/bin/strip_9gag
+++ b/bin/strip_9gag
@@ -2,7 +2,7 @@
 
 def convert(src)
   bn = File.basename(src).sub(/\.jp?eg\z/i, "")
-  target = "/home/taw/Desktop/#{bn}.png"
+  target = ENV['HOME'] + "/Desktop/#{bn}.png"
   system "convert", "-crop", "0x0-0-40", src, target
   puts target
 end
@@ -21,4 +21,4 @@ if src =~ /\Ahttp/
   convert("/tmp/#{fn}")
 else
   convert(src)
-end  
+end
