From acb53b8563be97b7cce45d68f5912878117f648d Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Fri, 10 Jan 2025 05:25:37 -0500
Subject: [PATCH 2/3] Improve get_skip_paths

---
 cruft/_commands/utils/generate.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/cruft/_commands/utils/generate.py b/cruft/_commands/utils/generate.py
index 4d8605a..cafc652 100644
--- a/cruft/_commands/utils/generate.py
+++ b/cruft/_commands/utils/generate.py
@@ -1,4 +1,5 @@
 import os
+import re
 import stat
 import sys
 from pathlib import Path
@@ -117,7 +118,8 @@ def _get_skip_paths(cruft_state: CruftState, pyproject_file: Path) -> Set[Union[
             "`toml` package is not installed. Cruft configuration may be ignored.",
             stacklevel=2,
         )
-    return set(map(lambda p: p if "*" in p else Path(p), skip_cruft))
+    non_globs = {p for p in skip_cruft if not re.search(r"(?<!\\)\*", p)}
+    return {Path(p) for p in non_globs} | (set(skip_cruft) - non_globs)
 
 
 def _get_deleted_files(template_dir: Path, project_dir: Path):
-- 
2.47.1

