diff --git a/app-portage/gcruft/gcruft-0.1.1.ebuild b/app-portage/gcruft/gcruft-0.1.1.ebuild
index 6f6efd9..823a65b 100644
--- a/app-portage/gcruft/gcruft-0.1.1.ebuild
+++ b/app-portage/gcruft/gcruft-0.1.1.ebuild
@@ -52,6 +52,9 @@ src_install() {
 	insinto $cfdir
 	doins "${S}"/config.pm
 	dodir $cfdir/exceptions
+
+	echo "CONFIG_PROTECT=\"${cfdir}\"" > 99gcruft
+	doenvd 99gcruft
 }
 
 pkg_postinst() {
