diff --git a/src/debugger/dbg_memory.c b/src/debugger/dbg_memory.c
index e98bf081..c6a762d6 100644
--- a/src/debugger/dbg_memory.c
+++ b/src/debugger/dbg_memory.c
@@ -97,9 +97,17 @@ static int read_memory_func(bfd_vma memaddr, bfd_byte *myaddr, unsigned int leng
     return (0);
 }
 
+static int fprintf_styled_nop(void *out __attribute__((unused)),
+                              enum disassembler_style s __attribute__((unused)),
+                              const char *fmt __attribute__((unused)),
+                              ...)
+{
+  return 0;
+}
+
 void init_host_disassembler(void)
 {
-    INIT_DISASSEMBLE_INFO(dis_info, stderr, process_opcode_out);
+    INIT_DISASSEMBLE_INFO(dis_info, stderr, process_opcode_out, fprintf_styled_nop);
     dis_info.fprintf_func = (fprintf_ftype) process_opcode_out;
     dis_info.stream = stderr;
     dis_info.bytes_per_line=1;
