From 9824daf3c240c56e27014604fa886f0e12849038 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sat, 24 Nov 2018 02:57:48 -0500
Subject: [PATCH] Fix ambigious calls to QVariant()

---
 vulkanDeviceInfo.hpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/vulkanDeviceInfo.hpp b/vulkanDeviceInfo.hpp
index 5c8fad1..84fa798 100644
--- a/vulkanDeviceInfo.hpp
+++ b/vulkanDeviceInfo.hpp
@@ -480,7 +480,7 @@ class VulkanDeviceInfo
             pfnGetPhysicalDeviceProperties2KHR(device, &deviceProps2);
             pushProperty2(extension, "maxTransformFeedbackStreams", QVariant(extProps.maxTransformFeedbackStreams));
             pushProperty2(extension, "maxTransformFeedbackBuffers", QVariant(extProps.maxTransformFeedbackBuffers));
-            pushProperty2(extension, "maxTransformFeedbackBufferSize", QVariant(extProps.maxTransformFeedbackBufferSize));
+            pushProperty2(extension, "maxTransformFeedbackBufferSize", QVariant::fromValue(extProps.maxTransformFeedbackBufferSize));
             pushProperty2(extension, "maxTransformFeedbackStreamDataSize", QVariant(extProps.maxTransformFeedbackStreamDataSize));
             pushProperty2(extension, "maxTransformFeedbackBufferDataSize", QVariant(extProps.maxTransformFeedbackBufferDataSize));
             pushProperty2(extension, "maxTransformFeedbackBufferDataStride", QVariant(extProps.maxTransformFeedbackBufferDataStride));
@@ -536,9 +536,9 @@ class VulkanDeviceInfo
             pushProperty2(extension, "maxRecursionDepth", QVariant(extProps.maxRecursionDepth));
             pushProperty2(extension, "maxShaderGroupStride", QVariant(extProps.maxShaderGroupStride));
             pushProperty2(extension, "shaderGroupBaseAlignment", QVariant(extProps.shaderGroupBaseAlignment));
-            pushProperty2(extension, "maxGeometryCount", QVariant(extProps.maxGeometryCount));
-            pushProperty2(extension, "maxInstanceCount", QVariant(extProps.maxInstanceCount));
-            pushProperty2(extension, "maxTriangleCount", QVariant(extProps.maxTriangleCount));
+            pushProperty2(extension, "maxGeometryCount", QVariant::fromValue(extProps.maxGeometryCount));
+            pushProperty2(extension, "maxInstanceCount", QVariant::fromValue(extProps.maxInstanceCount));
+            pushProperty2(extension, "maxTriangleCount", QVariant::fromValue(extProps.maxTriangleCount));
             pushProperty2(extension, "maxDescriptorSetAccelerationStructures", QVariant(extProps.maxDescriptorSetAccelerationStructures));
         }
         // VK_NV_mesh_shader
@@ -629,7 +629,7 @@ class VulkanDeviceInfo
                     deviceProps2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2_KHR;
                     deviceProps2.pNext = &extProps;
                     pfnGetPhysicalDeviceProperties2KHR(device, &deviceProps2);
-                    properties2.push_back(Property2("maxPerSetDescriptors", QVariant(extProps.maxPerSetDescriptors), extName));
+                    properties2.push_back(Property2("maxPerSetDescriptors", QVariant::fromValue(extProps.maxPerSetDescriptors), extName));
                     properties2.push_back(Property2("maxMemoryAllocationSize", QVariant::fromValue(extProps.maxMemoryAllocationSize), extName));
                 }
             }
