From 8910bb726f2c1e83e882aeeb764cf72c9fdc1390 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sat, 20 May 2023 03:58:45 -0400
Subject: [PATCH] Linux support

---
 CMakeLists.txt | 29 ++++++++---------------------
 fsrcnn.cpp     | 12 ++++++------
 main.c         |  4 ++--
 3 files changed, 16 insertions(+), 29 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4d3296..e660078 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,29 +1,16 @@
+cmake_minimum_required(VERSION 3.1)
 project(vs_fsrcnn_vulkan)
 
-cmake_minimum_required(VERSION 3.1)
+include(GNUInstallDirs)
 
 set(CMAKE_BUILD_TYPE Release)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
-set(OpenCV "D:/ProgramData/opencv/build")
-include_directories(${OpenCV}/include)
-link_directories(${OpenCV}/x64/vc15/lib)
-
-set(Vulkan "D:/ProgramData/VulkanSDK/1.1.126.0")
-include_directories(${Vulkan}/Include)
-link_directories(${Vulkan}/Lib)
-
-set(NCNN_Build "D:/Workspace/ncnn-master/build-vs2017/install")
-include_directories(${NCNN_Build}/include/ncnn)
-link_directories(${NCNN_Build}/lib)
-
-set(VapourSynthSDK "D:/Program Files/VapourSynth/sdk")
-include_directories(${VapourSynthSDK}/include/vapoursynth)
-link_directories(${VapourSynthSDK}/lib64)
+find_package(OpenCV REQUIRED)
+find_package(Vulkan REQUIRED)
+find_package(ncnn REQUIRED)
 
 add_library(vs_fsrcnn_vulkan SHARED main.c fsrcnn.cpp)
-target_link_libraries(vs_fsrcnn_vulkan vapoursynth)
-target_link_libraries(vs_fsrcnn_vulkan vsscript)
-target_link_libraries(vs_fsrcnn_vulkan opencv_world412)
-target_link_libraries(vs_fsrcnn_vulkan vulkan-1)
-target_link_libraries(vs_fsrcnn_vulkan ncnn)
+target_link_libraries(vs_fsrcnn_vulkan ${OpenCV_LIBS} ${VULKAN_LIBRARIES} ${NCNN_LIBRARIES})
+
+install(TARGETS vs_fsrcnn_vulkan LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/vapoursynth)
diff --git a/fsrcnn.cpp b/fsrcnn.cpp
index 77686df..5548b92 100644
--- a/fsrcnn.cpp
+++ b/fsrcnn.cpp
@@ -2,8 +2,8 @@
 #include <wchar.h>
 
 // VapourSynth Library
-#include "VapourSynth.h"
-#include "VSHelper.h"
+#include <vapoursynth/VapourSynth.h>
+#include <vapoursynth/VSHelper.h>
 
 // Image decode and encode
 #include <opencv2/core.hpp>
@@ -12,9 +12,9 @@
 #include <opencv2/core/hal/interface.h>
 
 // NCNN Library
-#include "platform.h"
-#include "net.h"
-#include "gpu.h"
+#include <ncnn/platform.h>
+#include <ncnn/net.h>
+#include <ncnn/gpu.h>
 
 // FSRCNN Model
 #include "fsrcnn.id.h"
@@ -133,7 +133,7 @@ void new_to_gray(const ncnn::Mat& m, float* gray)
 	int remain = size;
 	for (; remain > 0; remain--)
 	{
-		*gray = min(max(*ptr, 0), 255);
+		*gray = std::min(std::max(*ptr, 0.0f), 255.0f);
 		gray++;
 		ptr++;
 	}
diff --git a/main.c b/main.c
index ff63e1d..d6f9b00 100644
--- a/main.c
+++ b/main.c
@@ -1,6 +1,6 @@
 // VapourSynth Library
-#include "VapourSynth.h"
-#include "VSHelper.h"
+#include <vapoursynth/VapourSynth.h>
+#include <vapoursynth/VSHelper.h>
 
 // FSRCNN
 #include "fsrcnn.h"
-- 
2.40.1

