From 0838f8c373029ace041b6122cbd0f6c6356d4917 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 16 Dec 2021 09:38:00 -0500
Subject: [PATCH 2/2] more path fixes

---
 padd.sh | 43 ++++++++++++++++++++++++-------------------
 1 file changed, 24 insertions(+), 19 deletions(-)

diff --git a/padd.sh b/padd.sh
index 2ba4e90..249418a 100755
--- a/padd.sh
+++ b/padd.sh
@@ -16,6 +16,9 @@ LC_NUMERIC=C
 
 ############################################ VARIABLES #############################################
 
+pihole_version=${XDG_RUNTIME_DIR:-/tmp}/padd.pihole_version
+padd_pid=${XDG_RUNTIME_DIR:-/tmp}/padd.pid
+
 # VERSION
 padd_version="v3.6.2"
 
@@ -118,7 +121,7 @@ pihole_logo_script_retro_3="${green_text}'  ${red_text}'   ${magenta_text}' ${ye
 
 GetFTLData() {
     local ftl_port LINE
-    ftl_port=$(cat /run/pihole-FTL.port 2> /dev/null)
+    ftl_port=$(cat /run/pihole/FTL.port 2> /dev/null)
     if [[ -n "$ftl_port" ]]; then
         # Open connection to FTL
         exec 3<>"/dev/tcp/127.0.0.1/$ftl_port"
@@ -380,7 +383,7 @@ GetNetworkInformation() {
     # if the DHCP Router variable isn't set
     # Issue 3: https://github.com/jpmck/PADD/issues/3
     if [ -z ${DHCP_ROUTER+x} ]; then
-      DHCP_ROUTER=$(/sbin/ip route | awk '/default/ { printf "%s\t", $3 }')
+      DHCP_ROUTER=$(/bin/ip route | awk '/default/ { printf "%s\t", $3 }')
     fi
 
     dhcp_info=" Router:   ${DHCP_ROUTER}"
@@ -475,10 +478,10 @@ GetPiholeInformation() {
 
 GetVersionInformation() {
   # Check if version status has been saved
-  if [ -e "piHoleVersion" ]; then # the file exists...
+  if [ -e "$pihole_version" ]; then # the file exists...
     # the file exits, use it
-    # shellcheck disable=SC1091
-    source piHoleVersion
+    # shellcheck disable=SC1091 source=/dev/null
+    source "$pihole_version"
 
     # Today is...
     today=$(date +%Y%m%d)
@@ -486,7 +489,7 @@ GetVersionInformation() {
     # was the last check today?
     if [ "${today}" != "${last_check}" ]; then # no, it wasn't today
       # Remove the Pi-hole version file...
-      rm -f piHoleVersion
+      rm -f "$pihole_version"
     fi
 
   else # the file doesn't exist, create it...
@@ -593,7 +596,7 @@ GetVersionInformation() {
     fi
 
     # write it all to the file
-    echo "last_check=${today}" > ./piHoleVersion
+    echo "last_check=${today}" > "$pihole_version"
     {
       echo "core_version=$core_version"
       echo "core_version_latest=$core_version_latest"     
@@ -619,7 +622,7 @@ GetVersionInformation() {
       echo "mini_status_=\"$mini_status_\""
       echo "tiny_status_=\"$tiny_status_\""        
       echo "full_status_=\"$full_status_\""
-    } >> ./piHoleVersion
+    } >> "$pihole_version"
 
     # there's a file now
   fi
@@ -638,6 +641,7 @@ CleanEcho() {
 # wrapper - printf
 CleanPrintf() {
 # tput el
+  # shellcheck disable=SC2059
   printf "$@"
 }
 
@@ -1033,6 +1037,7 @@ json_extract() {
     local pair_regex="\"${key}\"[[:space:]]*:[[:space:]]*(${value_regex})"
 
     if [[ ${json} =~ ${pair_regex} ]]; then
+        # shellcheck disable=SC2001
         sed 's/^"\|"$//g' <<< "${BASH_REMATCH[1]}"
     else
         return 1
@@ -1057,12 +1062,12 @@ StartupRoutine(){
     # Get PID of PADD
     pid=$$
     echo -ne " [■·········]  10%\\r"
-    echo ${pid} > ./PADD.pid
+    echo ${pid} > "$padd_pid"
 
     # Check for updates
     echo -ne " [■■········]  20%\\r"
-    if [ -e "piHoleVersion" ]; then
-      rm -f piHoleVersion
+    if [ -e "$pihole_version" ]; then
+      rm -f "$pihole_version"
       echo -ne " [■■■·······]  30%\\r"
     else
       echo -ne " [■■■·······]  30%\\r"
@@ -1092,13 +1097,13 @@ StartupRoutine(){
     # Get PID of PADD
     pid=$$
     echo "- Writing PID (${pid}) to file."
-    echo ${pid} > ./PADD.pid
+    echo ${pid} > "$padd_pid"
 
     # Check for updates
     echo "- Checking for version file."
-    if [ -e "piHoleVersion" ]; then
+    if [ -e "$pihole_version" ]; then
       echo "  - Found and deleted."
-      rm -f piHoleVersion
+      rm -f "$pihole_version"
     else
       echo "  - Not found."
     fi
@@ -1132,13 +1137,13 @@ StartupRoutine(){
     # Get PID of PADD
     pid=$$
     echo "- Writing PID (${pid}) to file..."
-    echo ${pid} > ./PADD.pid
+    echo ${pid} > "$padd_pid"
 
     # Check for updates
     echo "- Checking for PADD version file..."
-    if [ -e "piHoleVersion" ]; then
+    if [ -e "$pihole_version" ]; then
       echo "  - PADD version file found... deleting."
-      rm -f piHoleVersion
+      rm -f "$pihole_version"
     else
       echo "  - PADD version file not found."
     fi
@@ -1179,7 +1184,7 @@ NormalPADD() {
     SizeChecker
 
     # Get Config variables
-    . /etc/pihole/setupVars.conf
+    . /var/lib/pihole/setupVars.conf
 
     # Move the cursor to top left of console to redraw
     tput cup 0 0
@@ -1236,7 +1241,7 @@ if [[ $# = 0 ]]; then
 
   # Get Our Config Values
   # shellcheck disable=SC1091
-  . /etc/pihole/setupVars.conf
+  . /var/lib/pihole/setupVars.conf
 
   SizeChecker
 
-- 
2.34.1

