From 9ebf063f24ab1fcedde527e40af5aab9b334a13a Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 16 Dec 2021 09:31:40 -0500
Subject: [PATCH 1/2] padd: shellcheck issues

---
 padd.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/padd.sh b/padd.sh
index e8bdd09..2ba4e90 100755
--- a/padd.sh
+++ b/padd.sh
@@ -317,7 +317,7 @@ GetNetworkInformation() {
 
   full_hostname=${pi_hostname}
   # does the Pi-hole have a domain set?
-  if ! [ -z ${PIHOLE_DOMAIN+x} ]; then
+  if [ -n "${PIHOLE_DOMAIN+x}" ]; then
     # is Pi-hole acting as DHCP server?
     if [[ "${DHCP_ACTIVE}" == "true" ]]; then
       count=${pi_hostname}"."${PIHOLE_DOMAIN}
@@ -1033,7 +1033,7 @@ json_extract() {
     local pair_regex="\"${key}\"[[:space:]]*:[[:space:]]*(${value_regex})"
 
     if [[ ${json} =~ ${pair_regex} ]]; then
-        echo $(sed 's/^"\|"$//g' <<< "${BASH_REMATCH[1]}")
+        sed 's/^"\|"$//g' <<< "${BASH_REMATCH[1]}"
     else
         return 1
     fi
-- 
2.34.1

