From 811d1bd62cf6c858200f9df0c9aa304e94a8507f Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 16 Dec 2021 10:42:15 -0500
Subject: [PATCH 3/3] version checking workaround

---
 padd.sh | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/padd.sh b/padd.sh
index 249418a..f757dab 100755
--- a/padd.sh
+++ b/padd.sh
@@ -495,8 +495,8 @@ GetVersionInformation() {
   else # the file doesn't exist, create it...
     # Gather core version information...
     read -r -a core_versions <<< "$(pihole -v -p)"
-    core_version=$(echo "${core_versions[3]}" | tr -d '\r\n[:alpha:]')
-    core_version_latest=${core_versions[5]//)}
+    core_version=$(echo "${core_versions[4]}" | tr -d '\r\n[:alpha:]')
+    core_version_latest=${core_version}

     if [[ "${core_version_latest}" == "ERROR" ]]; then
       core_version_heatmap=${yellow_text}
@@ -513,8 +513,8 @@ GetVersionInformation() {

     # Gather web version information...
     read -r -a web_versions <<< "$(pihole -v -a)"
-    web_version=$(echo "${web_versions[3]}" | tr -d '\r\n[:alpha:]')
-    web_version_latest=${web_versions[5]//)}
+    web_version=$(echo "${web_versions[4]}" | tr -d '\r\n[:alpha:]')
+    web_version_latest=${web_version}
     if [[ "${web_version_latest}" == "ERROR" ]]; then
       web_version_heatmap=${yellow_text}
     else
@@ -530,8 +530,8 @@ GetVersionInformation() {

     # Gather FTL version information...
     read -r -a ftl_versions <<< "$(pihole -v -f)"
-    ftl_version=$(echo "${ftl_versions[3]}" | tr -d '\r\n[:alpha:]')
-    ftl_version_latest=${ftl_versions[5]//)}
+    ftl_version=$(echo "${ftl_versions[4]}" | tr -d '\r\n[:alpha:]')
+    ftl_version_latest=${ftl_version}
     if [[ "${ftl_version_latest}" == "ERROR" ]]; then
       ftl_version_heatmap=${yellow_text}
     else
--
2.34.1

