From 4a94ccf0662453cfc9bdaa7ed2397b5cdab1db4d Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sun, 1 Jan 2023 15:10:05 -0500
Subject: [PATCH] more path fixes

---
 padd.sh | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/padd.sh b/padd.sh
index bbae894..ebcdff8 100755
--- a/padd.sh
+++ b/padd.sh
@@ -245,8 +245,8 @@ GetSystemInformation() {
   elif [ -f /tmp/sysinfo/model ]; then
     sys_model=$(tr -d '\0' < /tmp/sysinfo/model)
   elif [ -n "${DOCKER_VERSION}" ]; then
-    # Docker image. DOCKER_VERSION is read from /etc/pihole/versions
-    sys_model="Container"
+    # Docker image. DOCKER_VERSION is read from /var/lib/pihole/localversions
+    sys_model="Docker tag ${DOCKER_VERSION}"
   fi
 
   # Cleaning device model from useless OEM information
@@ -429,7 +429,7 @@ fi
 
 GetVersionInformation() {
   # Check if version status has been saved
-  # all info is sourced from /etc/pihole/versions
+  # all info is sourced from /var/lib/pihole/localversions
 
   out_of_date_flag=false
 
@@ -1197,7 +1197,7 @@ OutputJSON() {
 
 StartupRoutine(){
   # Get config variables
-  . /etc/pihole/setupVars.conf
+  . /var/lib/pihole/setupVars.conf
 
   # Clear the screen and move cursor to (0,0).
   # This mimics the 'clear' command.
@@ -1210,7 +1210,7 @@ StartupRoutine(){
   moveYOffset
 
   # Get versions information
-  . /etc/pihole/versions
+  . /var/lib/pihole/localversions
 
   if [ "$1" = "pico" ] || [ "$1" = "nano" ] || [ "$1" = "micro" ]; then
     moveXOffset; PrintLogo "$1"
@@ -1331,7 +1331,7 @@ NormalPADD() {
 
     # Get uptime, CPU load, temp, etc. every 5 seconds
     if [ $((now - LastCheckSystemInformation)) -ge 5 ]; then
-      . /etc/pihole/setupVars.conf
+      . /var/lib/pihole/setupVars.conf
       GetSystemInformation
       LastCheckSystemInformation="${now}"
     fi
@@ -1356,7 +1356,7 @@ NormalPADD() {
 
     # Get Pi-hole components version information every 30 seconds
     if [ $((now - LastCheckVersionInformation)) -ge 30 ]; then
-      . /etc/pihole/versions
+      . /var/lib/pihole/localversions
       GetVersionInformation
       LastCheckVersionInformation="${now}"
     fi
@@ -1431,7 +1431,7 @@ main(){
 
     # If setupVars.conf is not present, then PADD is not running on a Pi-hole
     # and we are not able to start as StartupRoutine() will fail below
-    if [ ! -f /etc/pihole/setupVars.conf ]; then
+    if [ ! -f /var/lib/pihole/setupVars.conf ]; then
       printf "%b" "${check_box_bad} Error!\n    PADD only works in conjunction with Pi-hole!\n"
       exit 1
     fi
-- 
2.39.1

