From 849cecb2e4649a253c5ed4ed4e5c5219dac9a1f6 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 16 Dec 2021 09:38:00 -0500
Subject: [PATCH] more path fixes

---
 padd.sh | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/padd.sh b/padd.sh
index 5468f3f..7ca81f6 100755
--- a/padd.sh
+++ b/padd.sh
@@ -25,6 +25,9 @@ cd "$tmpdir/padd_$(id -u)/" > /dev/null || {

 ############################################ VARIABLES #############################################

+pihole_version=${XDG_RUNTIME_DIR:-/tmp}/padd.pihole_version
+padd_pid=${XDG_RUNTIME_DIR:-/tmp}/padd.pid
+
 # VERSION
 padd_version="v3.7.1"

@@ -1015,7 +1018,7 @@ }

 StartupRoutine(){
     # Get config variables
-  . /etc/pihole/setupVars.conf
+  . /var/lib/pihole/setupVars.conf

   if [ "$1" = "pico" ] || [ "$1" = "nano" ] || [ "$1" = "micro" ]; then
     PrintLogo "$1"
--
2.35.1

