diff --git a/libraries/libplist/src/xplist.c b/libraries/libplist/src/xplist.c
index d8f6458..a1b5255 100644
--- a/libraries/libplist/src/xplist.c
+++ b/libraries/libplist/src/xplist.c
@@ -385,9 +385,6 @@ static void parse_date(const char *strval, struct TM *btime)
     if (!btime) return;
     memset(btime, 0, sizeof(struct tm));
     if (!strval) return;
-#ifdef HAVE_STRPTIME
-    strptime((char*)strval, "%Y-%m-%dT%H:%M:%SZ", btime);
-#else
 #ifdef USE_TM64
     #define PLIST_SSCANF_FORMAT "%lld-%d-%dT%d:%d:%dZ"
 #else
@@ -396,7 +393,6 @@ static void parse_date(const char *strval, struct TM *btime)
     sscanf(strval, PLIST_SSCANF_FORMAT, &btime->tm_year, &btime->tm_mon, &btime->tm_mday, &btime->tm_hour, &btime->tm_min, &btime->tm_sec);
     btime->tm_year-=1900;
     btime->tm_mon--;
-#endif
     btime->tm_isdst=0;
 }
 
