From 43c916710dfce02c6b2cc9e94699fd2b05c28b05 Mon Sep 17 00:00:00 2001
From: Rosalie Wanders <rosalie@mailbox.org>
Date: Sun, 21 Nov 2021 23:11:19 +0100
Subject: [PATCH] ci: fix build failure

---
 src/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e3e82dbbb..8a7abd620 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -496,6 +496,7 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQ
 endif()
 
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED YES)
 
 add_library( ${GLideN64_DLL_NAME} SHARED ${GLideN64_SOURCES} ${PATH_REVISION})
 
