diff --git a/src/xenia/kernel/xam/xam_profile.cc b/src/xenia/kernel/xam/xam_profile.cc
index 670cb8663..3f58b9100 100644
--- a/src/xenia/kernel/xam/xam_profile.cc
+++ b/src/xenia/kernel/xam/xam_profile.cc
@@ -25,14 +25,14 @@ dword_result_t XamProfileFindAccount_entry(
 
   account_ptr.Zero();
 
-  const auto& account =
+  const X_XAMACCOUNTINFO* account =
       kernel_state()->xam_state()->profile_manager()->GetAccount(offline_xuid);
 
   if (!account) {
     return X_ERROR_NO_SUCH_USER;
   }
 
-  std::memcpy(account_ptr, &account, sizeof(X_XAMACCOUNTINFO));
+  std::memcpy(account_ptr, account, sizeof(X_XAMACCOUNTINFO));
 
   if (device_id) {
     *device_id = 1;
