diff -urN mimeo-2018.12-old/Mimeo.py mimeo-2018.12-new/Mimeo.py
--- mimeo-2018.12-old/Mimeo.py	2018-12-24 13:58:47.000000000 -0500
+++ mimeo-2018.12-new/Mimeo.py	2019-03-02 14:30:04.056459649 -0500
@@ -2619,7 +2619,7 @@
 
 
 
-if __name__ == '__main__':
+def main2():
   logging.basicConfig(
     format='%(levelname)s: %(message)s',
     level=logging.DEBUG if ('--debug' in sys.argv[1:]) else logging.WARNING
diff -urN mimeo-2018.12-old/setup.py mimeo-2018.12-new/setup.py
--- mimeo-2018.12-old/setup.py	2018-12-24 13:58:47.000000000 -0500
+++ mimeo-2018.12-new/setup.py	2019-03-02 14:33:59.216065845 -0500
@@ -1,6 +1,6 @@
 #!/usr/bin/env python3
 
-from distutils.core import setup
+from setuptools import setup
 import time
 
 setup(
@@ -11,4 +11,9 @@
   author_email='''ac xunilhcra enyx, backwards''',
   url='''http://xyne.archlinux.ca/projects/mimeo''',
   py_modules=['''Mimeo'''],
+  entry_points={
+    'console_scripts': [
+        'mimeo = Mimeo:main2',
+    ],
+  },
 )
