diff --git a/vita3k/renderer/src/vulkan/texture.cpp b/vita3k/renderer/src/vulkan/texture.cpp
index bed1af4e..e1b794a8 100644
--- a/vita3k/renderer/src/vulkan/texture.cpp
+++ b/vita3k/renderer/src/vulkan/texture.cpp
@@ -148,7 +148,7 @@ void VKTextureCacheState::prepare_staging_buffer(bool is_configure) {
 
     if (need_wait) {
         if (staging_buffer->scene_timestamp == context->scene_timestamp) {
-            assert(current_fence == staging_buffer.waiting_fence);
+            assert(current_fence == staging_buffer->waiting_fence);
             // special case, all the staging buffer are occupied by the current scene
             // submit the command buffer and wait for it
             context->prerender_cmd.end();
@@ -489,4 +489,4 @@ void upload_done(VKTextureCacheState &cache) {
 }
 } // namespace texture
 
-} // namespace renderer::vulkan
\ No newline at end of file
+} // namespace renderer::vulkan
