Index: flourish/fSchema.php
===================================================================
--- flourish/fSchema.php	(revision 1042)
+++ flourish/fSchema.php	(working copy)
@@ -1656,7 +1656,8 @@
 			'box'               => 'varchar',
 			'path'              => 'varchar',
 			'polygon'           => 'varchar',
-			'circle'            => 'varchar'
+			'circle'            => 'varchar',
+			'tsvector'          => 'varchar'
 		);
 		
 		$max_min_values = array(
