Index: flourish/fValidation.php
===================================================================
--- flourish/fValidation.php	(revision 1042)
+++ flourish/fValidation.php	(working copy)
@@ -434,11 +434,10 @@
 	 * Adds a rule to make sure at least one field of multiple has a value
 	 * 
 	 * @param  string $field    One of the fields to check for a value
-	 * @param  string $field_2  Another field to check for a value
 	 * @param  string ...
 	 * @return fValidation  The validation object, to allow for method chaining
 	 */
-	public function addOneOrMoreRule($field, $field_2)
+	public function addOneOrMoreRule($field)
 	{
 		$fields = func_get_args();
 		$this->one_or_more_rules[] = $fields;
