--- flourish3/fImage.php	2012-01-06 17:10:33.000000000 -0600
+++ flourish/fImage.php	2012-01-06 17:16:45.000000000 -0600
@@ -157,7 +157,7 @@
 	 * 
 	 * @return void
 	 */
-	static private function determineProcessor()
+	static protected function determineProcessor()
 	{
 		// Determine what processor to use
 		if (self::$processor === NULL) {
@@ -1486,4 +1486,4 @@
  * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
- */
\ No newline at end of file
+ */
