From 44ac77a11dd9af00040dc1749eee6f3c98e08a54 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sun, 27 Oct 2024 06:08:21 -0400
Subject: [PATCH] Build system fix

---
 CMakeLists.txt | 30 ++++--------------------------
 1 file changed, 4 insertions(+), 26 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 98e7e5f..269ac8d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -268,21 +268,7 @@ else()
     add_subdirectory(third_party/ncnn)
 endif()
 
-# Include ExternalProject module
-include(ExternalProject)
-
-# Add librealesrgan-ncnn-vulkan as an external project
-ExternalProject_Add(
-    realesrgan
-    SOURCE_DIR ${PROJECT_SOURCE_DIR}/third_party/librealesrgan_ncnn_vulkan/src
-    CMAKE_ARGS
-        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-        -DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/realesrgan_install
-        -DCMAKE_POSITION_INDEPENDENT_CODE=ON
-        -DUSE_SYSTEM_NCNN=${USE_SYSTEM_NCNN}
-    BUILD_ALWAYS ON
-    INSTALL_COMMAND ${CMAKE_COMMAND} --build . --target install --config ${CMAKE_BUILD_TYPE}
-)
+add_subdirectory(third_party/realesrgan)
 
 # Remove duplicate entries
 list(REMOVE_DUPLICATES ALL_INCLUDE_DIRS)
@@ -301,7 +287,7 @@ else()
 endif()
 
 # Ensure libvideo2x depends on realesrgan being built and installed
-add_dependencies(libvideo2x realesrgan)
+#add_dependencies(libvideo2x realesrgan-ncnn-vulkan)
 
 # Include directories for the shared library
 target_include_directories(libvideo2x PRIVATE
@@ -327,7 +313,7 @@ else()
 endif()
 
 # Link the shared library with the dependencies
-target_link_libraries(libvideo2x PRIVATE ${ALL_LIBRARIES} ${REALESRGAN_LIB})
+target_link_libraries(libvideo2x PRIVATE ${ALL_LIBRARIES} realesrgan-ncnn-vulkan)
 
 if(NOT WIN32)
     if (USE_SYSTEM_NCNN)
@@ -365,7 +351,7 @@ if(WIN32)
 else()
     set(BIN_DESTINATION_DEFAULT "bin")
     set(INCLUDE_DESTINATION_DEFAULT "include/libvideo2x")
-    set(LIB_DESTINATION_DEFAULT "lib")
+    set(LIB_DESTINATION_DEFAULT "lib64")
     set(MODEL_DESTINATION_DEFAULT "share/video2x")
 endif()
 
@@ -440,12 +426,4 @@ if(WIN32)
                     GROUP_READ GROUP_EXECUTE
                     WORLD_READ WORLD_EXECUTE
     )
-else()
-    # Install Unix-specific dependencies
-    install(FILES ${REALESRGAN_LIB}
-        DESTINATION ${INSTALL_LIB_DESTINATION}
-        PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
-                    GROUP_READ GROUP_EXECUTE
-                    WORLD_READ WORLD_EXECUTE
-    )
 endif()
-- 
2.47.0

