diff --git a/api/index.php b/api/index.php
index 311f824..e6ec360 100644
--- a/api/index.php
+++ b/api/index.php
@@ -53,14 +53,14 @@ if($method=="post" || $method=="delete"){
 		HttpUtils::ApiError(404,"Not found");
 	}
 	
-	if($allRows[0]->UserId!=$users[0]->Id && !$users[0]->Admin) HttpUtils::ApiError(403,"Missing Api Key");
+	if($allRows[0]->userid!=$users[0]->Id && !$users[0]->Admin) HttpUtils::ApiError(403,"Missing Api Key");
 	
 	if(UrlUtils::ExistRequestParam("setPrerelease")){
 		
-		$allRows[0]->IsPreRelease=$method=="delete";
+		$allRows[0]->isprerelease=$method=="delete";
 		//echo "IsPreRelease ".$allRows[0]->IsPreRelease;
 	}else{
-		$allRows[0]->Listed=$method=="post";
+		$allRows[0]->listed=$method=="post";
 		//echo "Listed ".$allRows[0]->Listed;
 	}
 	
@@ -71,7 +71,7 @@ if($method=="post" || $method=="delete"){
 
 $isSymbol = UrlUtils::GetRequestParamOrDefault("symbol",null)!=null;
 
-$file = ($allRows[0]->Id.".".$allRows[0]->Version.($isSymbol?".snupkg":".nupkg"));
+$file = ($allRows[0]->id.".".$allRows[0]->version.($isSymbol?".snupkg":".nupkg"));
 
 $path = Path::Combine(Settings::$PackagesRoot,$file);
 
diff --git a/inc/api_nuget.php b/inc/api_nuget.php
index cab774f..5dac20f 100644
--- a/inc/api_nuget.php
+++ b/inc/api_nuget.php
@@ -113,10 +113,10 @@ class ApiNugetBase
     {
         $t = $this->_entryTemplate;
         $t.="  ";
-        if($e->Author==null){
-        	$e->Author="";
+        if($e->author==null){
+        	$e->author="";
 		}
-        $authors = explode(";",$e->Author);
+        $authors = explode(";",$e->author);
         $author = "";
         if(sizeof($authors)>0){
 			for($i=0;$i<sizeof($authors);$i++){
@@ -128,43 +128,43 @@ class ApiNugetBase
 		$baseUrl = trim($baseUrl,"\\/");
 		
         $t= $this->_doReplace("\${BASEURL}",$baseUrl,$t);
-        $t= $this->_doReplace("\${NUSPEC.ID}",$e->Id,$t);
+        $t= $this->_doReplace("\${NUSPEC.ID}",$e->id,$t);
         
         
         $t= $this->_doReplace("\${NUSPEC.IDLOWER}",strtolower($e->Id),$t);
-        $t= $this->_doReplace("\${NUSPEC.TITLE}",$e->Title,$t,true);
-        $t= $this->_doReplace("\${NUSPEC.VERSION}",$e->Version,$t);
-        $t= $this->_doReplace("\${NUSPEC.LICENSEURL}",$e->LicenseUrl,$t,true);
-        $t= $this->_doReplace("\${NUSPEC.PROJECTURL}",$e->ProjectUrl,$t,true);
-        $t= $this->_doReplace("\${NUSPEC.REQUIRELICENSEACCEPTANCE}",$e->RequireLicenseAcceptance?"true":"false",$t);
-        $t= $this->_doReplace("\${NUSPEC.DESCRIPTION}",$e->Description,$t,true);
-		$t= $this->_doReplace("\${NUSPEC.TAGS}",$e->Tags,$t,true);
-		$t= $this->_doReplace("\${NUSPEC.SUMMARY}",$e->Summary,$t,true);
+        $t= $this->_doReplace("\${NUSPEC.TITLE}",$e->title,$t,true);
+        $t= $this->_doReplace("\${NUSPEC.VERSION}",$e->version,$t);
+        $t= $this->_doReplace("\${NUSPEC.LICENSEURL}",$e->licenseurl,$t,true);
+        $t= $this->_doReplace("\${NUSPEC.PROJECTURL}",$e->projecturl,$t,true);
+        $t= $this->_doReplace("\${NUSPEC.REQUIRELICENSEACCEPTANCE}",$e->requirelicenseacceptance?"true":"false",$t);
+        $t= $this->_doReplace("\${NUSPEC.DESCRIPTION}",$e->description,$t,true);
+		$t= $this->_doReplace("\${NUSPEC.TAGS}",$e->tags,$t,true);
+		$t= $this->_doReplace("\${NUSPEC.SUMMARY}",$e->summary,$t,true);
 		$t= $this->_doReplace("\${NUSPEC.RELEASENOTES}",$e->ReleaseNotes,$t,true);
 
         $t= $this->_doReplace("\${NUSPEC.AUTHOR}",$author,$t);
         $t= $this->_doReplace("\${NUSPEC.AUTHORS}",$author,$t);
-        $t= $this->_doReplace("\${DB.PUBLISHED}",$e->Published,$t);
-        $t= $this->_doReplace("\${DB.PACKAGESIZE}",$e->PackageSize,$t);
-        $t= $this->_doReplace("\${DB.PACKAGEHASHALGORITHM}",$e->PackageHashAlgorithm,$t);
-        $t= $this->_doReplace("\${DB.PACKAGEHASH}",$e->PackageHash,$t);
+        $t= $this->_doReplace("\${DB.PUBLISHED}",$e->published,$t);
+        $t= $this->_doReplace("\${DB.PACKAGESIZE}",$e->packagesize,$t);
+        $t= $this->_doReplace("\${DB.PACKAGEHASHALGORITHM}",$e->packagehashalgorithm,$t);
+        $t= $this->_doReplace("\${DB.PACKAGEHASH}",$e->packagehash,$t);
        
-        if(is_string($e->Dependencies) && strlen($e->Dependencies)==0){
+        if(is_string($e->dependencies) && (strlen($e->dependencies)==0 || $e->dependencies === 's:0:"";')){
             $t= str_replace("\${NUSPEC.DEPENDENCIES}","",$t);
-        }else if(is_array($e->Dependencies)){
-            $t= str_replace("\${NUSPEC.DEPENDENCIES}",$this->MakeDepString($e->Dependencies),$t);
+        }else if(is_array($e->dependencies)){
+            $t= str_replace("\${NUSPEC.DEPENDENCIES}",$this->MakeDepString($e->dependencies),$t);
         }
-        $t= $this->_doReplace("\${DB.DOWNLOADCOUNT}",$e->DownloadCount,$t);
-        $t= $this->_doReplace("\${DB.UPDATED}",$e->Published,$t);
+        $t= $this->_doReplace("\${DB.DOWNLOADCOUNT}",$e->downloadcount,$t);
+        $t= $this->_doReplace("\${DB.UPDATED}",$e->published,$t);
 
 		
 		
-        $t= $this->_doReplace("\${DB.ISPRERELEASE}",$e->IsPreRelease?"true":"false",$t);
-		$t= $this->_doReplace("\${DB.ISABSOLUTELATESTVERSION}",$e->IsAbsoluteLatestVersion?"true":"false",$t);
-        $t= $this->_doReplace("\${DB.ISLATESTVERSION}",$e->IsLatestVersion?"true":"false",$t);
+        $t= $this->_doReplace("\${DB.ISPRERELEASE}",$e->isprerelease?"true":"false",$t);
+		$t= $this->_doReplace("\${DB.ISABSOLUTELATESTVERSION}",$e->isabsolutelatestversion?"true":"false",$t);
+        $t= $this->_doReplace("\${DB.ISLATESTVERSION}",$e->islatestversion?"true":"false",$t);
         $t= $this->_doReplace("\${DB.VERSIONDOWNLOADCOUNT}","-1",$t);
-        $t= $this->_doReplace("\${DB.LISTED}",$e->Listed?"true":"false",$t);
-		$t= $this->_doReplace("\${DB.COPYRIGHT}",$e->Copyright,$t,true);
+        $t= $this->_doReplace("\${DB.LISTED}",$e->listed?"true":"false",$t);
+		$t= $this->_doReplace("\${DB.COPYRIGHT}",$e->copyright,$t,true);
 
         return preg_replace('/<!--(.*)-->/Uis', '', $t);
     }
diff --git a/inc/api_packages.php b/inc/api_packages.php
index 9e6397f..66584d9 100644
--- a/inc/api_packages.php
+++ b/inc/api_packages.php
@@ -147,7 +147,7 @@ class PackagesApi extends SmallTextDbApiBase
 	
 	public function dogetbyquery()
 	{
-		$query = UrlUtils::GetRequestParam("Query");
+		$query = trim(UrlUtils::GetRequestParam("Query"));
 		$doGroup = UrlUtils::GetRequestParamOrDefault("DoGroup","false");
 		
 		global $loginController;
diff --git a/inc/commons/mysqldb.php b/inc/commons/mysqldb.php
index 10b0d25..2f22d71 100644
--- a/inc/commons/mysqldb.php
+++ b/inc/commons/mysqldb.php
@@ -97,7 +97,7 @@ class MySqlDb
 			for($i=0;$i<sizeof($r);$i++){
 				$k = $r[$i];
 				$kt = $t[$i];
-				$val = $list[strtolower($k)];
+				$val = $list[$k];
 				if($kt=="object"){
 					$item->$k = unserialize($val);
 				}else if($kt=="number" && $val!=null){
diff --git a/inc/commons/objectsearch.php b/inc/commons/objectsearch.php
index 608fb25..faf5c24 100644
--- a/inc/commons/objectsearch.php
+++ b/inc/commons/objectsearch.php
@@ -9,7 +9,7 @@ class Operator
 	{
 		$t = strtolower($this->Type);
 		
-		if($t=="string"){
+		if($t=="string" || $t=="version"){
 			return "'".$this->Value."'";
 		}
 		$r = $this->Value;
@@ -29,6 +29,9 @@ class Operator
 		}
 		return $r;
 	}
+	public function __toString() {
+		return $this->toString();
+	}
 }
 
 class IdentifyResult
diff --git a/inc/db_nugetpackages.php b/inc/db_nugetpackages.php
index 2843944..cafa451 100644
--- a/inc/db_nugetpackages.php
+++ b/inc/db_nugetpackages.php
@@ -15,7 +15,12 @@ require_once(__ROOT__."/inc/commons/url.php");
 require_once(__ROOT__."/inc/commons/objectsearch.php");
 require_once(__ROOT__."/inc/db_nugetpackagesentity.php");
 
-define('__MYTXTDB_PKG__',Path::Combine(Settings::$DataRoot,"nugetdb_pkg.txt"));
+if(__DB_TYPE__==DBMYSQL){
+	define('__MYTXTDB_PKG__','nugetdb_pkg');
+}
+else{
+	define('__MYTXTDB_PKG__',Path::Combine(Settings::$DataRoot,"nugetdb_pkg.txt"));
+}
 define('__MYTXTDBROWS_PKG__',
       "Version0:|:Version1:|:Version2:|:Version3:|:VersionBeta:|:Version:|:Title:|:Id:|:Author:|:IconUrl:|:LicenseUrl:|:ProjectUrl:|:DownloadCount:|:".
       "RequireLicenseAcceptance:|:Description:|:ReleaseNotes:|:Published:|:Dependencies:|:".
diff --git a/inc/db_users.php b/inc/db_users.php
index dc46b01..e50182a 100644
--- a/inc/db_users.php
+++ b/inc/db_users.php
@@ -14,7 +14,11 @@ require_once(__ROOT__."/inc/commons/objectsearch.php");
 require_once(__ROOT__."/inc/db_usersentity.php");
 
 
-define('__MYTXTDB_USR__',Path::Combine(Settings::$DataRoot,"nugetdb_usrs.txt"));
+if(__DB_TYPE__==DBMYSQL){
+	define('__MYTXTDB_USR__',"nugetdb_usrs");
+}else{
+	define('__MYTXTDB_USR__',Path::Combine(Settings::$DataRoot,"nugetdb_usrs.txt"));
+}
 define('__MYTXTDBROWS_USR__',
       "UserId:|:Name:|:Company:|:Md5Password:|:Packages:|:Enabled:|:Email:|:Token:|:Admin:|:Id");
 define('__MYTXTDBROWS_USR_TYP__',
@@ -89,7 +93,8 @@ class UserDb
 		
         if($doAdd){
             $toInsert["Token"]=Utils::NewGuid();
-			$toInsert["Id"]=Utils::NewGuid();
+						$toInsert["Id"]=Utils::NewGuid();
+						$toInsert["Md5Password"] = password_hash($nugetEntity->Password, PASSWORD_DEFAULT);
             $dbInstance->add_row($toInsert);
         }else{
 			$dbInstance->update_row($toInsert,array("UserId"=>$toInsert["UserId"]));
diff --git a/inc/gerfen/class.stream.php b/inc/gerfen/class.stream.php
index e0c5ac5..91b2bad 100644
--- a/inc/gerfen/class.stream.php
+++ b/inc/gerfen/class.stream.php
@@ -30,7 +30,7 @@ class stream
 
 		$boundary = $this->boundary();
 		
-		if (!count($boundary)) {
+		if (!strlen($boundary)) {
 			return array(
 				'post' => $this->parse($this->input),
 				'file' => array()
@@ -244,7 +244,7 @@ class stream
 			'file' => array()
 		);
 
-		if (count($array['post'] > 0)) {
+		if (count($array['post']) > 0) {
 			foreach($array['post'] as $key => $value) {
 				foreach($value as $k => $v) {
 					if (is_array($v)) {
@@ -258,7 +258,7 @@ class stream
 			}
 		}
 
-		if (count($array['file'] > 0)) {
+		if (count($array['file']) > 0) {
 			foreach($array['file'] as $key => $value) {
 				$results['file'][$key] = $value;
 				/*foreach($value as $k => $v) {
diff --git a/inc/logincontroller.php b/inc/logincontroller.php
index 448f0ea..257f44f 100644
--- a/inc/logincontroller.php
+++ b/inc/logincontroller.php
@@ -13,6 +13,7 @@ class LoginController
 	{
 		session_start();
 		$doLogin = UrlUtils::GetRequestParamOrDefault("DoLogin","xxx","all");
+		$doRegister = UrlUtils::GetRequestParamOrDefault("DoRegister","xxx","all");
 		if("false"==$doLogin) {
 			session_unset();
 			session_destroy(); 
@@ -20,7 +21,20 @@ class LoginController
 			return;
 		}
 		
-		if("true"!=$doLogin && array_key_exists ("UserId",$_SESSION)){
+		if("true"==$doRegister) {
+			$udb = new UserDb();
+			try {
+				$udb->AddRow((object)$_POST, false);
+			} catch (Exception $e) {
+				$result = base64_encode($e->getMessage());
+				$location = UrlUtils::CurrentUrl(Settings::$SiteRoot."?specialType=logon&result=$result");
+				header("Location: ".$location);
+				die();
+			}
+			$location = UrlUtils::CurrentUrl(Settings::$SiteRoot."?specialType=logon");
+			header("Location: ".$location);
+			die();
+		}else if("true"!=$doLogin && array_key_exists ("UserId",$_SESSION)){
 			$this->IsLoggedIn = true;
 			$this->UserId = $_SESSION["UserId"];
 			$this->Admin = $_SESSION["Admin"];
@@ -84,7 +98,6 @@ class LoginController
 		}
 		
 		$uid = UrlUtils::GetRequestParam("UserId","post");
-		$pwd = md5(UrlUtils::GetRequestParam("Password","post"));
 		
 		$udb = new UserDb();
 		$user = null;
@@ -94,7 +107,7 @@ class LoginController
 			
 		if(sizeof($ar)==1){
 			$user = $ar[0];
-			if($user->Md5Password != $pwd) {
+			if (!password_verify(UrlUtils::GetRequestParam("Password","post"), $user->Md5Password)) {
 				$errorCode = -2;
 			} else if($user->Enabled != true) {
 				$errorCode = -3;
diff --git a/inc/partials/_logon.php b/inc/partials/_logon.php
index 34e945f..0b3e612 100644
--- a/inc/partials/_logon.php
+++ b/inc/partials/_logon.php
@@ -17,6 +17,7 @@
 				<div class="col-md-6" >
 					<h3>Register</h3>
 					<form class="form-horizontal" method="POST" action="<?php echo Settings::$SiteRoot;?>" >
+						<input type="hidden" id="DoRegister" name="DoRegister" value="true"/>
 						<div class="form-group">
 							<label class="control-label col-md-2" for="UserId">Username</label>
 							<div class="col-md-10">
diff --git a/inc/partials/_packages.php b/inc/partials/_packages.php
index 4d5eed5..8e82717 100644
--- a/inc/partials/_packages.php
+++ b/inc/partials/_packages.php
@@ -156,20 +156,20 @@ if($searchQuery!=null){
 					?>
 					<tr>
 						<td>
-							<img class="package__icon package__icon--md" src="<?php echo $item->IconUrl;?>"/>
+							<img class="package__icon package__icon--md" src="<?php echo $item->iconurl;?>"/>
 						</td>
 						<td>
 							<b><a href="<?php echo Settings::$SiteRoot;
 							?>?specialType=singlePackage<?php
-							echo "&id=".urlencode($item->Id);
-							echo "&version=".urlencode($item->Version);
-							?>"><?php echo $item->Title?></a></b><br><?php echo " v.".$item->Version;?>
+							echo "&id=".urlencode($item->id);
+							echo "&version=".urlencode($item->version);
+							?>"><?php echo $item->title?></a></b><br><?php echo " v.".$item->version;?>
 						</td>
 						<td>
 							<ul>
 								<li><b>By:</b>
 								<?php
-								$ath = explode(",",$item->Author);
+								$ath = explode(",",$item->author);
 								for($k=0;$k<sizeof($ath);$k++){
 									$v= trim($ath[$k]);
 									if($k>0)echo ",&nbsp;";
@@ -182,9 +182,9 @@ if($searchQuery!=null){
 								}
 								?>
 								</li>
-								<li><b>Description:</b> <?php echo $item->Description;?></li>
-								<li><b>Total downloads:</b> <?php echo $item->DownloadCount;?></li>
-								<li><b>Tags:</b> <?php echo is_array($item->Tags) ? implode(",",$item->Tags):$item->Tags;?></li>
+								<li><b>Description:</b> <?php echo $item->description;?></li>
+								<li><b>Total downloads:</b> <?php echo $item->downloadcount;?></li>
+								<li><b>Tags:</b> <?php echo is_array($item->tags) ? implode(",",$item->tags):$item->tags;?></li>
 							</ul>
 						</td>
 					</tr>
