reverted:
--- b/part/document/katedocument.cpp
+++ a/part/document/katedocument.cpp
@@ -4542,12 +4542,6 @@
   if (remove == 0)
     return;
 
-  // get cursor position of active view
-  KTextEditor::Cursor curPos = KTextEditor::Cursor::invalid();
-  if (activeView()) {
-    curPos = activeView()->cursorPosition();
-  }
-
   editStart();
 
   for (int line = 0; line < lines(); ++line)
@@ -4556,17 +4550,11 @@
 
     // remove trailing spaces in entire document, remove = 2
     // remove trailing spaces of touched lines, remove = 1
+    if (remove == 2 || textline->markedAsModified()) {
-    // remove trailing spaces of lines saved on disk, remove = 1
-    if (remove == 2 || textline->markedAsModified() || textline->markedAsSavedOnDisk()) {
       const int p = textline->lastChar() + 1;
       const int l = textline->length() - p;
+      if (l > 0) {
+        editRemoveText(line, p, l);
-      if (l > 0 ) {
-        // if the cursor is in the trailing space, only delete behind cursor
-        if (curPos.line() != line || curPos.column() <= p || curPos.column() > p + l) {
-          editRemoveText(line, p, l);
-        } else {
-          editRemoveText(line, curPos.column(), l - (curPos.column() - p));
-        }
       }
     } 
