diff --git a/lib/autokey/interface.py b/lib/autokey/interface.py
index afb7776..f164bf8 100644
--- a/lib/autokey/interface.py
+++ b/lib/autokey/interface.py
@@ -665,6 +665,8 @@ class XInterfaceBase(threading.Thread):
 
     def finish_send(self):
         self.__enqueue(self.__ungrabKeyboard)
+        time.sleep(0.1)
+        subprocess.call(["xdotool", "windowfocus", "'xdotool getwindowfocus'"])
 
     def grab_keyboard(self):
         self.__enqueue(self.__grab_keyboard)
