diff --git a/CMakeLists.txt b/CMakeLists.txt
index 82599f2..8979fde 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,16 +72,16 @@ find_package(Threads REQUIRED)
 find_package(SDL2 REQUIRED)
 find_package(CURL REQUIRED)
 find_package(pugixml REQUIRED)
-find_package(imgui REQUIRED)
+add_subdirectory(imgui)
 find_package(RapidJSON REQUIRED)
 find_package(Boost COMPONENTS program_options filesystem nowide REQUIRED)
 find_package(libzip REQUIRED)
-find_package(glslang REQUIRED)
 find_package(ZLIB REQUIRED)
 find_package(zstd MODULE REQUIRED) # MODULE so that zstd::zstd is available
 find_package(OpenSSL COMPONENTS Crypto SSL REQUIRED)
 find_package(glm REQUIRED)
-find_package(fmt 7.0.0 REQUIRED)
+add_subdirectory(fmt)
+add_subdirectory(glslang)
 find_package(PNG REQUIRED)
 
 # glslang versions older than 11.11.0 define targets without a namespace
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e903db4..a85ed5e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -76,7 +76,7 @@ set_property(TARGET CemuBin PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONF
 set_target_properties(CemuBin PROPERTIES
         RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_CURRENT_SOURCE_DIR}/../bin/
         RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_CURRENT_SOURCE_DIR}/../bin/
-        OUTPUT_NAME "Cemu"
+        OUTPUT_NAME "Cemu2"
         )
 
 target_link_libraries(CemuBin PRIVATE
@@ -91,7 +91,7 @@ target_link_libraries(CemuBin PRIVATE
 )
 
 target_link_libraries(CemuBin PRIVATE CemuAsm)
-target_link_libraries(CemuBin PRIVATE SDL2::SDL2 SDL2::SDL2main) # is SDL2main needed?
+target_link_libraries(CemuBin PRIVATE SDL2::SDL2) # is SDL2main needed?
 target_link_libraries(CemuBin PRIVATE imguiImpl OpenGL::GL)
 
 if (ENABLE_WXWIDGETS)
