diff --git a/CMakeLists.txt b/CMakeLists.txt
index 03558265..bbe31560 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -133,9 +133,6 @@ if (DYNARMIC_TESTS_USE_UNICORN)
     find_package(Unicorn REQUIRED)
 endif()
 
-# Pull in externals CMakeLists for libs where available
-add_subdirectory(externals)
-
 # Dynarmic project files
 add_subdirectory(src)
 if (DYNARMIC_TESTS)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9f5b45b4..ff0bb6bc 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -328,9 +328,7 @@ target_compile_options(dynarmic PRIVATE ${DYNARMIC_CXX_FLAGS})
 target_link_libraries(dynarmic
     PRIVATE
         boost
-        fmt::fmt
-        mp
-        xbyak
+        fmt
         $<$<BOOL:DYNARMIC_USE_LLVM>:${llvm_libs}>
 )
 if (DYNARMIC_ENABLE_CPU_FEATURE_DETECTION)
