diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5e1458ce..52a7fcff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -137,9 +137,6 @@ if (DYNARMIC_TESTS_USE_UNICORN)
     find_package(Unicorn REQUIRED)
 endif()
 
-# Pull in externals CMakeLists for libs where available
-add_subdirectory(externals)
-
 # Dynarmic project files
 add_subdirectory(src)
 if (DYNARMIC_TESTS)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 10533571..20cc9be3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -343,12 +343,10 @@ target_compile_options(dynarmic PRIVATE ${DYNARMIC_CXX_FLAGS})
 target_link_libraries(dynarmic
     PRIVATE
         boost
-        fmt::fmt
-        mp
-        tsl::robin_map
-        xbyak
+        fmt
         $<$<BOOL:DYNARMIC_USE_LLVM>:${llvm_libs}>
 )
+target_include_directories(dynarmic PRIVATE ../externals/robin-map/include)
 if (DYNARMIC_ENABLE_CPU_FEATURE_DETECTION)
     target_compile_definitions(dynarmic PRIVATE DYNARMIC_ENABLE_CPU_FEATURE_DETECTION=1)
 endif()
