From bdb675ef5992822391e04501d73d358ae04fd830 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sun, 30 Oct 2022 22:55:47 -0400
Subject: [PATCH 2/3] inih fix

---
 src/citra/config.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/citra/config.cpp b/src/citra/config.cpp
index cb7f4b10d..88ee33252 100644
--- a/src/citra/config.cpp
+++ b/src/citra/config.cpp
@@ -8,7 +8,7 @@
 #include <type_traits>
 #include <unordered_map>
 #include <SDL.h>
-#include <inih/cpp/INIReader.h>
+#include <INIReader.h>
 #include "citra/config.h"
 #include "citra/default_ini.h"
 #include "common/file_util.h"
-- 
2.40.0

