From 65773f0ea02b06e3633378af8d4eef9e639dc8af Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sun, 8 Jan 2023 14:34:44 -0500
Subject: [PATCH 3/3] Disable tests

---
 src/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9f976eedf..119e4de57 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -143,7 +143,6 @@ add_subdirectory(video_core)
 add_subdirectory(audio_core)
 add_subdirectory(network)
 add_subdirectory(input_common)
-add_subdirectory(tests)
 
 if (ENABLE_SDL2)
     add_subdirectory(citra)
-- 
2.40.1

