From 0cdf2050f5e863f9e9cd99fe7d9c270086b279a8 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sun, 30 Oct 2022 22:55:47 -0400
Subject: [PATCH 2/2] inih fix

---
 src/citra/config.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/citra/config.cpp b/src/citra/config.cpp
index 09fda6ca5..09d25ba2a 100644
--- a/src/citra/config.cpp
+++ b/src/citra/config.cpp
@@ -7,7 +7,7 @@
 #include <sstream>
 #include <type_traits>
 #include <SDL.h>
-#include <inih/cpp/INIReader.h>
+#include <INIReader.h>
 #include "citra/config.h"
 #include "citra/default_ini.h"
 #include "common/file_util.h"
-- 
2.41.0

