diff --git a/src/citra/CMakeLists.txt b/src/citra/CMakeLists.txt
index 86066dfe9..28ea54323 100644
--- a/src/citra/CMakeLists.txt
+++ b/src/citra/CMakeLists.txt
@@ -16,7 +16,7 @@ add_executable(citra
 create_target_directory_groups(citra)
 
 target_link_libraries(citra PRIVATE common core input_common network)
-target_link_libraries(citra PRIVATE inih glad lodepng)
+target_link_libraries(citra PRIVATE glad)
 if (MSVC)
     target_link_libraries(citra PRIVATE getopt)
 endif()
