diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index 5d65965fa..d02fb33bb 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -129,8 +129,4 @@ endif()
 
 create_target_directory_groups(common)
 
-target_link_libraries(common PUBLIC fmt microprofile Boost::boost Boost::serialization)
-target_link_libraries(common PRIVATE libzstd_static)
-if (ARCHITECTURE_x86_64)
-    target_link_libraries(common PRIVATE xbyak)
-endif()
+target_link_libraries(common PUBLIC fmt microprofile Boost::boost Boost::serialization zstd)
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 01f18e445..605c560ba 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -475,10 +475,6 @@ target_link_libraries(core PUBLIC common PRIVATE audio_core network video_core)
 target_link_libraries(core PUBLIC Boost::boost PRIVATE cryptopp fmt open_source_archives Boost::serialization)
 
 if (ENABLE_WEB_SERVICE)
-    get_directory_property(OPENSSL_LIBS
-        DIRECTORY ${PROJECT_SOURCE_DIR}/externals/libressl
-        DEFINITION OPENSSL_LIBS)
-
     target_compile_definitions(core PRIVATE -DENABLE_WEB_SERVICE -DCPPHTTPLIB_OPENSSL_SUPPORT)
     target_link_libraries(core PRIVATE web_service ${OPENSSL_LIBS} httplib lurlparser)
     if (ANDROID)
diff --git a/src/tests/CMakeLists.txt b/src/tests/CMakeLists.txt
index c6b222ef0..f685dae08 100644
--- a/src/tests/CMakeLists.txt
+++ b/src/tests/CMakeLists.txt
@@ -24,6 +24,6 @@ endif()
 create_target_directory_groups(tests)
 
 target_link_libraries(tests PRIVATE common core video_core audio_core)
-target_link_libraries(tests PRIVATE ${PLATFORM_LIBRARIES} catch-single-include nihstro-headers Threads::Threads)
+target_link_libraries(tests PRIVATE ${PLATFORM_LIBRARIES} nihstro-headers Threads::Threads)
 
 add_test(NAME tests COMMAND tests)
diff --git a/src/video_core/CMakeLists.txt b/src/video_core/CMakeLists.txt
index a6f9860eb..d282be8b2 100644
--- a/src/video_core/CMakeLists.txt
+++ b/src/video_core/CMakeLists.txt
@@ -143,7 +143,3 @@ create_target_directory_groups(video_core)
 
 target_link_libraries(video_core PUBLIC common core)
 target_link_libraries(video_core PRIVATE glad nihstro-headers Boost::serialization)
-
-if (ARCHITECTURE_x86_64)
-    target_link_libraries(video_core PUBLIC xbyak)
-endif()
diff --git a/src/web_service/CMakeLists.txt b/src/web_service/CMakeLists.txt
index 5695e25f0..dc7b41e6c 100644
--- a/src/web_service/CMakeLists.txt
+++ b/src/web_service/CMakeLists.txt
@@ -13,9 +13,6 @@ add_library(web_service STATIC
 
 create_target_directory_groups(web_service)
 
-get_directory_property(OPENSSL_LIBS
-        DIRECTORY ${PROJECT_SOURCE_DIR}/externals/libressl
-        DEFINITION OPENSSL_LIBS)
 target_compile_definitions(web_service PRIVATE -DCPPHTTPLIB_OPENSSL_SUPPORT)
 target_link_libraries(web_service PRIVATE common network json-headers ${OPENSSL_LIBS} httplib lurlparser cpp-jwt)
 if (ANDROID)
