From 731c80ba47bf40674b7ceb77a657afbeb9bdb349 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sun, 8 Jan 2023 14:34:44 -0500
Subject: [PATCH 3/3] Disable tests

---
 src/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f59607d66..a4df2ab54 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -148,7 +148,6 @@ add_subdirectory(video_core)
 add_subdirectory(audio_core)
 add_subdirectory(network)
 add_subdirectory(input_common)
-add_subdirectory(tests)
 
 if (ENABLE_SDL2)
     add_subdirectory(citra)
-- 
2.40.0

