diff --git a/externals/dynarmic/src/backend/x64/a32_jitstate.h b/externals/dynarmic/src/backend/x64/a32_jitstate.h
index 495bf86a..5bfce6a5 100644
--- a/externals/dynarmic/src/backend/x64/a32_jitstate.h
+++ b/externals/dynarmic/src/backend/x64/a32_jitstate.h
@@ -8,7 +8,7 @@
 
 #include <array>
 
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "common/common_types.h"
 
diff --git a/externals/dynarmic/src/backend/x64/a64_jitstate.h b/externals/dynarmic/src/backend/x64/a64_jitstate.h
index 3fb1c228..a5168be5 100644
--- a/externals/dynarmic/src/backend/x64/a64_jitstate.h
+++ b/externals/dynarmic/src/backend/x64/a64_jitstate.h
@@ -8,7 +8,7 @@
 
 #include <array>
 
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "common/common_types.h"
 #include "frontend/A64/location_descriptor.h"
diff --git a/externals/dynarmic/src/backend/x64/abi.cpp b/externals/dynarmic/src/backend/x64/abi.cpp
index 2dcbce3e..21ae4dc8 100644
--- a/externals/dynarmic/src/backend/x64/abi.cpp
+++ b/externals/dynarmic/src/backend/x64/abi.cpp
@@ -17,7 +17,7 @@
 #include <algorithm>
 #include <vector>
 
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "backend/x64/abi.h"
 #include "backend/x64/block_of_code.h"
diff --git a/externals/dynarmic/src/backend/x64/block_of_code.cpp b/externals/dynarmic/src/backend/x64/block_of_code.cpp
index 470be9ea..398aa9b0 100644
--- a/externals/dynarmic/src/backend/x64/block_of_code.cpp
+++ b/externals/dynarmic/src/backend/x64/block_of_code.cpp
@@ -7,7 +7,7 @@
 #include <array>
 #include <cstring>
 
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "backend/x64/a32_jitstate.h"
 #include "backend/x64/abi.h"
diff --git a/externals/dynarmic/src/backend/x64/block_of_code.h b/externals/dynarmic/src/backend/x64/block_of_code.h
index e629c009..42a54062 100644
--- a/externals/dynarmic/src/backend/x64/block_of_code.h
+++ b/externals/dynarmic/src/backend/x64/block_of_code.h
@@ -11,8 +11,8 @@
 #include <memory>
 #include <type_traits>
 
-#include <xbyak.h>
-#include <xbyak_util.h>
+#include <xbyak/xbyak.h>
+#include <xbyak/xbyak_util.h>
 
 #include "backend/x64/callback.h"
 #include "backend/x64/constant_pool.h"
diff --git a/externals/dynarmic/src/backend/x64/callback.h b/externals/dynarmic/src/backend/x64/callback.h
index d305ccdb..8e675996 100644
--- a/externals/dynarmic/src/backend/x64/callback.h
+++ b/externals/dynarmic/src/backend/x64/callback.h
@@ -9,7 +9,7 @@
 #include <functional>
 #include <vector>
 
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "common/common_types.h"
 
diff --git a/externals/dynarmic/src/backend/x64/constant_pool.h b/externals/dynarmic/src/backend/x64/constant_pool.h
index 65ebaaf1..a0dd5d93 100644
--- a/externals/dynarmic/src/backend/x64/constant_pool.h
+++ b/externals/dynarmic/src/backend/x64/constant_pool.h
@@ -9,7 +9,7 @@
 #include <map>
 #include <tuple>
 
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "common/common_types.h"
 
diff --git a/externals/dynarmic/src/backend/x64/emit_x64.h b/externals/dynarmic/src/backend/x64/emit_x64.h
index 6bab043a..e60ece82 100644
--- a/externals/dynarmic/src/backend/x64/emit_x64.h
+++ b/externals/dynarmic/src/backend/x64/emit_x64.h
@@ -14,7 +14,7 @@
 #include <unordered_set>
 #include <vector>
 
-#include <xbyak_util.h>
+#include <xbyak/xbyak_util.h>
 
 #include "backend/x64/exception_handler.h"
 #include "backend/x64/reg_alloc.h"
diff --git a/externals/dynarmic/src/backend/x64/hostloc.cpp b/externals/dynarmic/src/backend/x64/hostloc.cpp
index 8e3a2f8d..6f31a8cb 100644
--- a/externals/dynarmic/src/backend/x64/hostloc.cpp
+++ b/externals/dynarmic/src/backend/x64/hostloc.cpp
@@ -4,7 +4,7 @@
  * General Public License version 2 or any later version.
  */
 
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "backend/x64/hostloc.h"
 
diff --git a/externals/dynarmic/src/backend/x64/hostloc.h b/externals/dynarmic/src/backend/x64/hostloc.h
index f05dcc40..e7cb1465 100644
--- a/externals/dynarmic/src/backend/x64/hostloc.h
+++ b/externals/dynarmic/src/backend/x64/hostloc.h
@@ -5,7 +5,7 @@
  */
 #pragma once
 
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "common/assert.h"
 #include "common/common_types.h"
diff --git a/externals/dynarmic/src/backend/x64/oparg.h b/externals/dynarmic/src/backend/x64/oparg.h
index 266c1f36..81c33223 100644
--- a/externals/dynarmic/src/backend/x64/oparg.h
+++ b/externals/dynarmic/src/backend/x64/oparg.h
@@ -6,7 +6,7 @@
 
 #pragma once
 
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "common/assert.h"
 
diff --git a/externals/dynarmic/src/backend/x64/reg_alloc.cpp b/externals/dynarmic/src/backend/x64/reg_alloc.cpp
index f91e07a8..d4ad9fe2 100644
--- a/externals/dynarmic/src/backend/x64/reg_alloc.cpp
+++ b/externals/dynarmic/src/backend/x64/reg_alloc.cpp
@@ -9,7 +9,7 @@
 #include <utility>
 
 #include <fmt/ostream.h>
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "backend/x64/abi.h"
 #include "backend/x64/reg_alloc.h"
diff --git a/externals/dynarmic/src/backend/x64/reg_alloc.h b/externals/dynarmic/src/backend/x64/reg_alloc.h
index 341dfe66..bb5b656b 100644
--- a/externals/dynarmic/src/backend/x64/reg_alloc.h
+++ b/externals/dynarmic/src/backend/x64/reg_alloc.h
@@ -12,7 +12,7 @@
 #include <utility>
 #include <vector>
 
-#include <xbyak.h>
+#include <xbyak/xbyak.h>
 
 #include "backend/x64/block_of_code.h"
 #include "backend/x64/hostloc.h"
