diff --git a/externals/dynarmic/tests/A32/fuzz_arm.cpp b/externals/dynarmic/tests/A32/fuzz_arm.cpp
index b377995a..0c7bf73c 100644
--- a/externals/dynarmic/tests/A32/fuzz_arm.cpp
+++ b/externals/dynarmic/tests/A32/fuzz_arm.cpp
@@ -10,7 +10,7 @@
 #include <tuple>
 #include <vector>
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 #include <dynarmic/A32/a32.h>
 
 #include "common/common_types.h"
diff --git a/externals/dynarmic/tests/A32/fuzz_thumb.cpp b/externals/dynarmic/tests/A32/fuzz_thumb.cpp
index 7daae415..03051352 100644
--- a/externals/dynarmic/tests/A32/fuzz_thumb.cpp
+++ b/externals/dynarmic/tests/A32/fuzz_thumb.cpp
@@ -11,7 +11,7 @@
 #include <functional>
 #include <tuple>
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 
 #include <dynarmic/A32/a32.h>
 
diff --git a/externals/dynarmic/tests/A32/test_arm_disassembler.cpp b/externals/dynarmic/tests/A32/test_arm_disassembler.cpp
index 4e4b0bad..bab52e92 100644
--- a/externals/dynarmic/tests/A32/test_arm_disassembler.cpp
+++ b/externals/dynarmic/tests/A32/test_arm_disassembler.cpp
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: 0BSD
  */
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 
 #include "frontend/A32/disassembler/disassembler.h"
 
diff --git a/externals/dynarmic/tests/A32/test_arm_instructions.cpp b/externals/dynarmic/tests/A32/test_arm_instructions.cpp
index c8e5eac4..28645bdd 100644
--- a/externals/dynarmic/tests/A32/test_arm_instructions.cpp
+++ b/externals/dynarmic/tests/A32/test_arm_instructions.cpp
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: 0BSD
  */
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 #include <dynarmic/A32/a32.h>
 
 #include "A32/testenv.h"
diff --git a/externals/dynarmic/tests/A32/test_thumb_instructions.cpp b/externals/dynarmic/tests/A32/test_thumb_instructions.cpp
index 46169839..2d592106 100644
--- a/externals/dynarmic/tests/A32/test_thumb_instructions.cpp
+++ b/externals/dynarmic/tests/A32/test_thumb_instructions.cpp
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: 0BSD
  */
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 
 #include <dynarmic/A32/a32.h>
 
diff --git a/externals/dynarmic/tests/A64/a64.cpp b/externals/dynarmic/tests/A64/a64.cpp
index 1cdb7628..c09d91ae 100644
--- a/externals/dynarmic/tests/A64/a64.cpp
+++ b/externals/dynarmic/tests/A64/a64.cpp
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: 0BSD
  */
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 
 #include <dynarmic/exclusive_monitor.h>
 
diff --git a/externals/dynarmic/tests/A64/fuzz_with_unicorn.cpp b/externals/dynarmic/tests/A64/fuzz_with_unicorn.cpp
index ccafed5f..555b9f0d 100644
--- a/externals/dynarmic/tests/A64/fuzz_with_unicorn.cpp
+++ b/externals/dynarmic/tests/A64/fuzz_with_unicorn.cpp
@@ -8,7 +8,7 @@
 #include <string>
 #include <vector>
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 
 #include "common/common_types.h"
 #include "common/fp/fpcr.h"
diff --git a/externals/dynarmic/tests/A64/misaligned_page_table.cpp b/externals/dynarmic/tests/A64/misaligned_page_table.cpp
index 45633764..dcb6e0a2 100644
--- a/externals/dynarmic/tests/A64/misaligned_page_table.cpp
+++ b/externals/dynarmic/tests/A64/misaligned_page_table.cpp
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: 0BSD
  */
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 
 #include <dynarmic/A64/a64.h>
 
diff --git a/externals/dynarmic/tests/A64/verify_unicorn.cpp b/externals/dynarmic/tests/A64/verify_unicorn.cpp
index afcb820a..de843848 100644
--- a/externals/dynarmic/tests/A64/verify_unicorn.cpp
+++ b/externals/dynarmic/tests/A64/verify_unicorn.cpp
@@ -5,7 +5,7 @@
 
 #include <array>
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 
 #include "rand_int.h"
 #include "testenv.h"
diff --git a/externals/dynarmic/tests/CMakeLists.txt b/externals/dynarmic/tests/CMakeLists.txt
index f4a596c2..10b24850 100644
--- a/externals/dynarmic/tests/CMakeLists.txt
+++ b/externals/dynarmic/tests/CMakeLists.txt
@@ -44,12 +44,12 @@ include(CreateDirectoryGroups)
 create_target_directory_groups(dynarmic_tests)
 create_target_directory_groups(dynarmic_print_info)
 
-target_link_libraries(dynarmic_tests PRIVATE dynarmic boost catch fmt mp xbyak)
+target_link_libraries(dynarmic_tests PRIVATE dynarmic boost fmt)
 target_include_directories(dynarmic_tests PRIVATE . ../src)
 target_compile_options(dynarmic_tests PRIVATE ${DYNARMIC_CXX_FLAGS})
 target_compile_definitions(dynarmic_tests PRIVATE FMT_USE_USER_DEFINED_LITERALS=0)
 
-target_link_libraries(dynarmic_print_info PRIVATE dynarmic boost catch fmt mp)
+target_link_libraries(dynarmic_print_info PRIVATE dynarmic boost fmt)
 target_include_directories(dynarmic_print_info PRIVATE . ../src)
 target_compile_options(dynarmic_print_info PRIVATE ${DYNARMIC_CXX_FLAGS})
 target_compile_definitions(dynarmic_print_info PRIVATE FMT_USE_USER_DEFINED_LITERALS=0)
diff --git a/externals/dynarmic/tests/cpu_info.cpp b/externals/dynarmic/tests/cpu_info.cpp
index e70fc8fc..b0f08a9a 100644
--- a/externals/dynarmic/tests/cpu_info.cpp
+++ b/externals/dynarmic/tests/cpu_info.cpp
@@ -6,9 +6,9 @@
 #include <array>
 #include <utility>
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 #include <fmt/format.h>
-#include <xbyak_util.h>
+#include <xbyak/xbyak_util.h>
 
 TEST_CASE("Host CPU supports", "[a64]") {
     Xbyak::util::Cpu cpu_info;
diff --git a/externals/dynarmic/tests/fp/FPToFixed.cpp b/externals/dynarmic/tests/fp/FPToFixed.cpp
index 896b7310..c4bb5178 100644
--- a/externals/dynarmic/tests/fp/FPToFixed.cpp
+++ b/externals/dynarmic/tests/fp/FPToFixed.cpp
@@ -6,7 +6,7 @@
 #include <tuple>
 #include <vector>
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 
 #include "common/common_types.h"
 #include "common/fp/fpcr.h"
diff --git a/externals/dynarmic/tests/fp/mantissa_util_tests.cpp b/externals/dynarmic/tests/fp/mantissa_util_tests.cpp
index d075849d..328572b9 100644
--- a/externals/dynarmic/tests/fp/mantissa_util_tests.cpp
+++ b/externals/dynarmic/tests/fp/mantissa_util_tests.cpp
@@ -6,7 +6,7 @@
 #include <tuple>
 #include <vector>
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 
 #include "common/common_types.h"
 #include "common/fp/mantissa_util.h"
diff --git a/externals/dynarmic/tests/fp/unpacked_tests.cpp b/externals/dynarmic/tests/fp/unpacked_tests.cpp
index 795271a8..d12b83cf 100644
--- a/externals/dynarmic/tests/fp/unpacked_tests.cpp
+++ b/externals/dynarmic/tests/fp/unpacked_tests.cpp
@@ -6,7 +6,7 @@
 #include <tuple>
 #include <vector>
 
-#include <catch.hpp>
+#include <catch2/catch.hpp>
 
 #include "common/common_types.h"
 #include "common/fp/fpcr.h"
diff --git a/externals/dynarmic/tests/main.cpp b/externals/dynarmic/tests/main.cpp
index a20b397a..73eaf605 100644
--- a/externals/dynarmic/tests/main.cpp
+++ b/externals/dynarmic/tests/main.cpp
@@ -4,4 +4,4 @@
  */
 
 #define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
-#include <catch.hpp>
+#include <catch2/catch.hpp>
