diff -urN --ignore-all-space a/saveblobs.sh b/saveblobs.sh
--- a/saveblobs.sh	2018-12-30 04:15:15.341483210 -0500
+++ b/saveblobs.sh	2018-12-30 04:19:40.703067871 -0500
@@ -1,5 +1,6 @@
-#!/bin/bash
-cd /var/shsh
+#!/usr/bin/env bash
+mkdir -p "${XDG_DATA_HOME:-${HOME}/.local/share}/tsschecker"
+cd ${XDG_DATA_HOME:-${HOME}/.local/share}/tsschecker
 declare -a DIRS=("shsh" "shsh_ota" "shsh_beta_ota")
 for dir in "${DIRS[@]}"
 do
@@ -62,4 +63,4 @@
 	echo "$ret" >>/tmp/tsschecker_saveblobs_fullog.log
 	echo -n $(echo $ret | grep -o "iOS .* for device" | rev | cut -c 12- | rev )
 	if [ $code -eq 0 ]; then echo " ok"; else echo " failed"; echo $ret;fi	
-done <devices.txt
+done <"${1:-/etc/tsschecker/devices.txt}"
