From 697a9441222d63b7ab99c3aec3afe701a822503e Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 16 Dec 2021 08:45:38 -0500
Subject: [PATCH 05/13] pihole: add rc-service to restartDNS

---
 pihole | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pihole b/pihole
index 938cba2a..ececf22d 100755
--- a/pihole
+++ b/pihole
@@ -126,6 +126,8 @@ restartDNS() {
         svc="service pihole-FTL restart"
     elif command -v systemctl &>/dev/null; then
         svc="systemctl restart pihole-FTL"
+    elif command -v rc-service &>/dev/null; then
+        svc="rc-service pihole-FTL restart"
     else
         "Cannot restart Pi-hole"
         return 1
-- 
2.38.0

