From 505e73c49618e2c32c8ef5b75cb1328b3ec84da5 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 13 Jan 2022 11:11:56 -0500
Subject: [PATCH 10/10] fix path in generated DHCP config

---
 advanced/Scripts/webpage.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/advanced/Scripts/webpage.sh b/advanced/Scripts/webpage.sh
index 7bbb7f64..faeba72c 100755
--- a/advanced/Scripts/webpage.sh
+++ b/advanced/Scripts/webpage.sh
@@ -453,7 +453,7 @@ ProcessDHCPSettings() {
 dhcp-authoritative
 dhcp-range=${DHCP_START},${DHCP_END},${leasetime}
 dhcp-option=option:router,${DHCP_ROUTER}
-dhcp-leasefile=/etc/pihole/dhcp.leases
+dhcp-leasefile=/var/lib/pihole/dhcp.leases
 #quiet-dhcp
 " > "${dhcpconfig}"
         chmod 644 "${dhcpconfig}"
-- 
2.34.1

