From 2c974abd94cdf9003973d5246b9fb7a9a4687aa7 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 13 Jan 2022 11:11:56 -0500
Subject: [PATCH 10/14] fix path in generated DHCP config

---
 advanced/Scripts/webpage.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/advanced/Scripts/webpage.sh b/advanced/Scripts/webpage.sh
index 8f3e6fdb..000f52ac 100755
--- a/advanced/Scripts/webpage.sh
+++ b/advanced/Scripts/webpage.sh
@@ -451,7 +451,7 @@ ProcessDHCPSettings() {
 dhcp-authoritative
 dhcp-range=${DHCP_START},${DHCP_END},${leasetime}
 dhcp-option=option:router,${DHCP_ROUTER}
-dhcp-leasefile=/etc/pihole/dhcp.leases
+dhcp-leasefile=/var/lib/pihole/dhcp.leases
 #quiet-dhcp
 " > "${dhcpconfig}"
         chmod 644 "${dhcpconfig}"
-- 
2.37.3

