From d25ee46ae0921c867bd36cfa8130dd69a0e8991b Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 13 Jan 2022 11:11:56 -0500
Subject: [PATCH 10/14] fix path in generated DHCP config

---
 advanced/Scripts/webpage.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/advanced/Scripts/webpage.sh b/advanced/Scripts/webpage.sh
index 30c1e693..0a91ff44 100755
--- a/advanced/Scripts/webpage.sh
+++ b/advanced/Scripts/webpage.sh
@@ -452,7 +452,7 @@ ProcessDHCPSettings() {
 dhcp-authoritative
 dhcp-range=${DHCP_START},${DHCP_END},${leasetime}
 dhcp-option=option:router,${DHCP_ROUTER}
-dhcp-leasefile=/etc/pihole/dhcp.leases
+dhcp-leasefile=/var/lib/pihole/dhcp.leases
 #quiet-dhcp
 " > "${dhcpconfig}"
         chmod 644 "${dhcpconfig}"
-- 
2.35.1

