From fefa975a085cd6eb1a198c22dfb9f413f4b2a559 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 13 Jan 2022 11:11:56 -0500
Subject: [PATCH 10/12] fix path in generated DHCP config

---
 advanced/Scripts/webpage.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/advanced/Scripts/webpage.sh b/advanced/Scripts/webpage.sh
index 5d76ea3a..68669da5 100755
--- a/advanced/Scripts/webpage.sh
+++ b/advanced/Scripts/webpage.sh
@@ -454,7 +454,7 @@ ProcessDHCPSettings() {
 dhcp-authoritative
 dhcp-range=${DHCP_START},${DHCP_END},${leasetime}
 dhcp-option=option:router,${DHCP_ROUTER}
-dhcp-leasefile=/etc/pihole/dhcp.leases
+dhcp-leasefile=/var/lib/pihole/dhcp.leases
 #quiet-dhcp
 " > "${dhcpconfig}"
         chmod 644 "${dhcpconfig}"
-- 
2.35.1

