From 4eeb5e4f445b3f6db31f0f51d8a044dd24575d9f Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Wed, 15 Dec 2021 07:24:10 -0500
Subject: [PATCH 04/14] More path fixes

---
 advanced/Scripts/list.sh | 2 +-
 pihole                   | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/advanced/Scripts/list.sh b/advanced/Scripts/list.sh
index eb5c0fec..99efc8da 100755
--- a/advanced/Scripts/list.sh
+++ b/advanced/Scripts/list.sh
@@ -11,7 +11,7 @@
 # Please see LICENSE file for your rights under this license.
 
 # Globals
-piholeDir="@EPREFIX@/etc/pihole"
+piholeDir="@EPREFIX@/var/lib/pihole"
 GRAVITYDB="@EPREFIX@/var/lib/pihole/gravity.db"
 # Source pihole-FTL from install script
 pihole_FTL="${piholeDir}/pihole-FTL.conf"
diff --git a/pihole b/pihole
index 8c8570c7..4659585a 100755
--- a/pihole
+++ b/pihole
@@ -325,13 +325,13 @@ statusFunc() {
     fi
 
   # Determine if Pi-hole's blocking is enabled
-  if grep -q "BLOCKING_ENABLED=false" @EPREFIX@/etc/pihole/setupVars.conf; then
+  if grep -q "BLOCKING_ENABLED=false" @EPREFIX@/var/lib/pihole/setupVars.conf; then
     # A config is commented out
     case "${1}" in
       "web") echo 0;;
       *) echo -e "  ${CROSS} Pi-hole blocking is disabled";;
     esac
-  elif grep -q "BLOCKING_ENABLED=true" @EPREFIX@/etc/pihole/setupVars.conf;  then
+  elif grep -q "BLOCKING_ENABLED=true" @EPREFIX@/var/lib/pihole/setupVars.conf;  then
     # Configs are set
     case "${1}" in
       "web") echo "$port";;
-- 
2.37.3

