From ea718ac490af47a8b84997047f1de4e4bb102ed3 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 16 Dec 2021 08:45:38 -0500
Subject: [PATCH 05/11] pihole: add rc-service to restartDNS

---
 pihole | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pihole b/pihole
index 5c60998e..905fa915 100755
--- a/pihole
+++ b/pihole
@@ -136,6 +136,8 @@ restartDNS() {
         svc="service pihole-FTL restart"
     elif command -v systemctl &>/dev/null; then
         svc="systemctl restart pihole-FTL"
+    elif command -v rc-service &>/dev/null; then
+        svc="rc-service pihole-FTL restart"
     else
         "Cannot restart Pi-hole"
         return 1
-- 
2.34.1

