From 08c46f1c1123b426d0e73c09b857e2b77c01170e Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 16 Dec 2021 08:46:50 -0500
Subject: [PATCH 06/14] advanced/Scripts/piholeLogFlush: change STATEFILE path

---
 advanced/Scripts/piholeLogFlush.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/advanced/Scripts/piholeLogFlush.sh b/advanced/Scripts/piholeLogFlush.sh
index d316bc41..ad669776 100755
--- a/advanced/Scripts/piholeLogFlush.sh
+++ b/advanced/Scripts/piholeLogFlush.sh
@@ -14,7 +14,7 @@ source ${colfile}
 # In case we're running at the same time as a system logrotate, use a
 # separate logrotate state file to prevent stepping on each other's
 # toes.
-STATEFILE="@EPREFIX@/var/lib/logrotate/pihole"
+STATEFILE="@EPREFIX@/var/lib/pihole/logrotate.state"
 
 # Determine database location
 # Obtain DBFILE=... setting from pihole-FTL.db
-- 
2.37.3

