diff --git a/src/about.go b/src/about.go
index ccdccac..14c76fd 100644
--- a/src/about.go
+++ b/src/about.go
@@ -17,7 +17,7 @@ package drive
 import (
 	"fmt"
 
-	drive "github.com/odeke-em/google-api-go-client/drive/v2"
+    drive "google.golang.org/api/drive/v2"
 	"github.com/odeke-em/log"
 )
 
diff --git a/src/remote.go b/src/remote.go
index 01ca4e0..b6363f0 100644
--- a/src/remote.go
+++ b/src/remote.go
@@ -27,7 +27,7 @@ import (
 
 	"code.google.com/p/goauth2/oauth"
 	"github.com/odeke-em/drive/config"
-	drive "github.com/odeke-em/google-api-go-client/drive/v2"
+    drive "google.golang.org/api/drive/v2"
 	"github.com/odeke-em/statos"
 )
 
diff --git a/src/stat.go b/src/stat.go
index d827b1f..abd393c 100644
--- a/src/stat.go
+++ b/src/stat.go
@@ -19,7 +19,7 @@ import (
 	"sync"
 	"time"
 
-	drive "github.com/odeke-em/google-api-go-client/drive/v2"
+    drive "google.golang.org/api/drive/v2"
 	"github.com/odeke-em/log"
 )
 
diff --git a/src/types.go b/src/types.go
index 9592b7c..a07e8d3 100644
--- a/src/types.go
+++ b/src/types.go
@@ -22,7 +22,7 @@ import (
 	"time"
 
 	"github.com/odeke-em/drive/config"
-	drive "github.com/odeke-em/google-api-go-client/drive/v2"
+    drive "google.golang.org/api/drive/v2"
 )
 
 type Operation int
