diff --git a/rpcs3/Emu/Cell/SPUOpcodes.h b/rpcs3/Emu/Cell/SPUOpcodes.h
index 2db6b6b1f..35c9e0103 100644
--- a/rpcs3/Emu/Cell/SPUOpcodes.h
+++ b/rpcs3/Emu/Cell/SPUOpcodes.h
@@ -1,6 +1,6 @@
 #pragma once
 
-#include "../../../Utilities/BitField.h"
+#include "Utilities/BitField.h"
 
 union spu_opcode_t
 {
diff --git a/rpcs3/Emu/RSX/Overlays/overlays.h b/rpcs3/Emu/RSX/Overlays/overlays.h
index cbfd1d390..6a18c24cb 100644
--- a/rpcs3/Emu/RSX/Overlays/overlays.h
+++ b/rpcs3/Emu/RSX/Overlays/overlays.h
@@ -1,8 +1,8 @@
 #pragma once
 #include "overlay_controls.h"
 
-#include "../../../Utilities/date_time.h"
-#include "../../../Utilities/Thread.h"
+#include "Utilities/date_time.h"
+#include "Utilities/Thread.h"
 #include "../../Io/PadHandler.h"
 #include "Emu/Memory/vm.h"
 #include "Emu/IdManager.h"
