From 71492259ebf26abb25f3b66c3e66e8c8299a4410 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Fri, 8 Oct 2021 11:37:19 -0400
Subject: [PATCH 03/10] Allow use of system SPIRV and glslang

---
 3rdparty/CMakeLists.txt | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/3rdparty/CMakeLists.txt b/3rdparty/CMakeLists.txt
index 4ad5f2783..8e6aeb474 100644
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -91,9 +91,12 @@ endif()
 add_subdirectory(hidapi)
 
 # glslang
-add_subdirectory(glslang EXCLUDE_FROM_ALL)
-add_library(3rdparty_glslang INTERFACE)
-target_link_libraries(3rdparty_glslang INTERFACE SPIRV)
+if(NOT USE_SYSTEM_GLSLANG)
+  add_subdirectory(glslang EXCLUDE_FROM_ALL)
+  add_library(3rdparty_glslang INTERFACE)
+  target_link_libraries(3rdparty_glslang INTERFACE SPIRV)
+endif()
+#add_subdirectory(SPIRV EXCLUDE_FROM_ALL)
 
 
 # yaml-cpp
-- 
2.49.0

