From b198629e20caa6b3c6a8412540f4b7e984ced7c7 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sun, 5 Mar 2023 21:46:38 -0500
Subject: [PATCH 07/10] Remove extra

---
 rpcs3/CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/rpcs3/CMakeLists.txt b/rpcs3/CMakeLists.txt
index 53429411c..d28a618bd 100644
--- a/rpcs3/CMakeLists.txt
+++ b/rpcs3/CMakeLists.txt
@@ -165,8 +165,4 @@ if(UNIX AND NOT APPLE)
         DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
     install(DIRECTORY ../bin/GuiConfigs
         DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
-    install(DIRECTORY ../bin/git
-        DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
-    install(DIRECTORY ../bin/test
-        DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
 endif()
-- 
2.43.0

