From 21648009cd8fe35bc0d48f24a44adfad488e7a43 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sun, 4 Jun 2023 01:50:57 -0400
Subject: [PATCH 08/10] Allow system rtmidi

---
 3rdparty/CMakeLists.txt | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/3rdparty/CMakeLists.txt b/3rdparty/CMakeLists.txt
index 7b0506137..f0838dc7a 100644
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -357,7 +357,11 @@ else()
 endif()
 
 # RTMIDI
-add_subdirectory(rtmidi EXCLUDE_FROM_ALL)
+if (NOT USE_SYSTEM_RTMIDI)
+  add_subdirectory(rtmidi EXCLUDE_FROM_ALL)
+else()
+  pkg_check_modules(PC_RTMIDI REQUIRED IMPORTED_TARGET GLOBAL rtmidi)
+endif()
 
 # add nice ALIAS targets for ease of use
 if(USE_SYSTEM_LIBUSB)
@@ -387,4 +391,4 @@ add_library(3rdparty::libcurl ALIAS 3rdparty_libcurl)
 add_library(3rdparty::soundtouch ALIAS soundtouch)
 add_library(3rdparty::sdl2 ALIAS ${SDL2_TARGET})
 add_library(3rdparty::miniupnpc ALIAS PkgConfig::PC_MINIUNPNPC)
-add_library(3rdparty::rtmidi ALIAS rtmidi)
+add_library(3rdparty::rtmidi ALIAS PkgConfig::PC_RTMIDI)
-- 
2.43.0

