From 5cb73483fa95506be649f066087356bcb23b7f36 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sat, 6 Apr 2024 22:58:57 -0400
Subject: [PATCH 08/10] add missing headers

---
 rpcs3/Emu/NP/np_handler.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rpcs3/Emu/NP/np_handler.cpp b/rpcs3/Emu/NP/np_handler.cpp
index 499b6b917..69a0e9be5 100644
--- a/rpcs3/Emu/NP/np_handler.cpp
+++ b/rpcs3/Emu/NP/np_handler.cpp
@@ -28,6 +28,7 @@
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <netinet/in.h>
+#include <sys/ioctl.h>
 #include <net/if.h>
 #include <arpa/inet.h>
 #include <netdb.h>
-- 
2.46.0

