From 9c0a28c8f2fca193d165782313e9f6dfc3ef14f6 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sun, 5 Mar 2023 21:46:38 -0500
Subject: [PATCH 7/7] Remove extra

---
 rpcs3/CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/rpcs3/CMakeLists.txt b/rpcs3/CMakeLists.txt
index e372c6650..e0fe3856b 100644
--- a/rpcs3/CMakeLists.txt
+++ b/rpcs3/CMakeLists.txt
@@ -215,8 +215,4 @@ if(UNIX AND NOT APPLE)
         DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
     install(DIRECTORY ../bin/GuiConfigs
         DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
-    install(DIRECTORY ../bin/git
-        DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
-    install(DIRECTORY ../bin/test
-        DESTINATION ${CMAKE_INSTALL_DATADIR}/rpcs3)
 endif()
-- 
2.39.2

