From 440a4c67b2213c8a8d8d56decb98f58a781ac279 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Tue, 19 Jan 2021 16:45:37 -0500
Subject: [PATCH 7/8] Fix branch name

---
 rpcs3/rpcs3_version.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rpcs3/rpcs3_version.cpp b/rpcs3/rpcs3_version.cpp
index d554086dd..0643cffd6 100644
--- a/rpcs3/rpcs3_version.cpp
+++ b/rpcs3/rpcs3_version.cpp
@@ -12,7 +12,7 @@ namespace rpcs3
 
 	std::string_view get_full_branch()
 	{
-		return RPCS3_GIT_BRANCH;
+		return "RPCS3/rpcs3/master";
 	}
 
 	std::pair<std::string, std::string> get_commit_and_hash()
-- 
2.30.0

