diff --git a/3rdparty/CMakeLists.txt b/3rdparty/CMakeLists.txt
index cd935aef0..a608a7c24 100644
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -362,16 +362,22 @@ target_link_libraries(3rdparty_openal INTERFACE ${OPENAL_LIBRARY})
 
 # FAudio
 set(FAUDIO_TARGET 3rdparty_dummy_lib)
-if(USE_FAUDIO)
+if(USE_FAUDIO OR USE_SYSTEM_FAUDIO)
 	# FAudio depends on SDL2
 	find_package(SDL2)
 	if (NOT SDL2_FOUND OR SDL2_VERSION VERSION_LESS 2.0.9)
 		message("-- RPCS3: FAudio requires SDL 2.0.9 or newer.")
 	else()
-		set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared library")
-		add_subdirectory(FAudio EXCLUDE_FROM_ALL)
-		target_compile_definitions(FAudio INTERFACE -DHAVE_FAUDIO)
-		set(FAUDIO_TARGET FAudio)
+		if (NOT USE_SYSTEM_FAUDIO)
+			set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared library")
+			add_subdirectory(FAudio EXCLUDE_FROM_ALL)
+			set(FAUDIO_TARGET FAudio)
+			target_compile_definitions(FAudio INTERFACE -DHAVE_FAUDIO)
+		else()
+			pkg_check_modules(FAUDIO FAudio REQUIRED IMPORTED_TARGET GLOBAL)
+			set(FAUDIO_TARGET PkgConfig::FAUDIO)
+			target_compile_definitions(PkgConfig::FAUDIO INTERFACE -DHAVE_FAUDIO)
+		endif()
 	endif()
 endif()
 
diff --git a/rpcs3/Emu/Audio/FAudio/FAudioBackend.h b/rpcs3/Emu/Audio/FAudio/FAudioBackend.h
index 07d406b0a..880a47c54 100644
--- a/rpcs3/Emu/Audio/FAudio/FAudioBackend.h
+++ b/rpcs3/Emu/Audio/FAudio/FAudioBackend.h
@@ -5,7 +5,7 @@
 #endif
 
 #include "Emu/Audio/AudioBackend.h"
-#include "3rdparty/FAudio/include/FAudio.h"
+#include "FAudio/FAudio.h"
 
 class FAudioBackend : public AudioBackend
 {
