From fd334c4f35df77e454e7ea808ed50e3ccfac3f2b Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Fri, 8 Oct 2021 11:37:19 -0400
Subject: [PATCH] Allow use of system SPIRV and glslang

---
 3rdparty/CMakeLists.txt | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/3rdparty/CMakeLists.txt b/3rdparty/CMakeLists.txt
index 925af7391..1ab71c53e 100644
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -72,8 +72,15 @@ add_subdirectory(hidapi)
 
 
 # Vulkan
-add_subdirectory(glslang EXCLUDE_FROM_ALL)
-add_subdirectory(SPIRV EXCLUDE_FROM_ALL)
+if(NOT USE_SYSTEM_GLSLANG)
+  add_subdirectory(glslang EXCLUDE_FROM_ALL)
+endif()
+if(USE_SYSTEM_SPIRV_HEADERS_TOOLS)
+  find_package(SPIRV-Headers REQUIRED)
+  find_package(SPIRV-Tools REQUIRED)
+else()
+  add_subdirectory(SPIRV EXCLUDE_FROM_ALL)
+endif()
 
 
 # yaml-cpp
-- 
2.33.0

