From 9f72f1b387bca422da756c387cd7af29fa2b46cd Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 18 Jan 2024 03:17:04 -0500
Subject: [PATCH 9/9] Add missing glslang includes

---
 rpcs3/Emu/RSX/VK/VKCommonDecompiler.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rpcs3/Emu/RSX/VK/VKCommonDecompiler.cpp b/rpcs3/Emu/RSX/VK/VKCommonDecompiler.cpp
index f91851e22..184d7b44c 100644
--- a/rpcs3/Emu/RSX/VK/VKCommonDecompiler.cpp
+++ b/rpcs3/Emu/RSX/VK/VKCommonDecompiler.cpp
@@ -14,6 +14,8 @@
 #endif
 #endif
 #include <glslang/SPIRV/GlslangToSpv.h>
+#include <glslang/Public/ResourceLimits.h>
+#include <glslang/Public/ShaderLang.h>
 #ifdef _MSC_VER
 #pragma warning(pop)
 #else
-- 
2.44.0

