From 3e21bffb3f1e43476b6931a5b65fb1a0eb59a01f Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Tue, 19 Jan 2021 03:25:14 -0500
Subject: [PATCH 1/8] rpcs3/rpcs3_version: fix typo?

---
 rpcs3/rpcs3_version.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rpcs3/rpcs3_version.cpp b/rpcs3/rpcs3_version.cpp
index b1abaeffa..d554086dd 100644
--- a/rpcs3/rpcs3_version.cpp
+++ b/rpcs3/rpcs3_version.cpp
@@ -12,7 +12,7 @@ namespace rpcs3
 
 	std::string_view get_full_branch()
 	{
-		return RPCS3_GIT_FULL_BRANCH;
+		return RPCS3_GIT_BRANCH;
 	}
 
 	std::pair<std::string, std::string> get_commit_and_hash()
-- 
2.30.0

