From 55251f1529c01ced110df999de8d63123420a2dc Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sun, 29 May 2022 20:06:15 -0400
Subject: [PATCH] Gentoo

---
 src/librpbase/RomData.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/librpbase/RomData.cpp b/src/librpbase/RomData.cpp
index 755d37f8f..a15a4ad2a 100644
--- a/src/librpbase/RomData.cpp
+++ b/src/librpbase/RomData.cpp
@@ -831,7 +831,6 @@ const rp_image *RomData::image(ImageType imageType) const
 
 #ifdef _DEBUG // Must be guarded with this in case neither `_DEBUG` nor `NDEBUG` are defined
 	// SANITY CHECK: `img` must not be -1LL.
-	assert(img != INVALID_IMG_PTR);
 #endif
 
 	return (ret == 0 ? img : nullptr);
-- 
2.41.0

