diff --git a/Source/JavaScriptCore/bytecode/CodeBlock.h b/Source/JavaScriptCore/bytecode/CodeBlock.h
index 6f8086d234..32ea22a6ab 100644
--- a/Source/JavaScriptCore/bytecode/CodeBlock.h
+++ b/Source/JavaScriptCore/bytecode/CodeBlock.h
@@ -994,7 +994,7 @@ private:
 /* This check is for normal Release builds; ASSERT_ENABLED changes the size. */
 #if !ASSERT_ENABLED
-static_assert(sizeof(CodeBlock) <= 224, "Keep it small for memory saving");
+static_assert(sizeof(CodeBlock) <= 248, "Keep it small for memory saving");
 #endif
 
 template <typename ExecutableType>
