diff --git a/cmake/Findfmt.cmake b/cmake/Findfmt.cmake
index c01e9b61..e794464a 100644
--- a/cmake/Findfmt.cmake
+++ b/cmake/Findfmt.cmake
@@ -1,4 +1,5 @@
 option(VCPKG_DEPENDENCY_EXTERNAL_FMT "Use an external version of the fmt library" OFF)
+option(VCPKG_DEPENDENCY_EXTERNAL_FMT_SRC "Use an external source of the fmt library" OFF)
 
 # This option exists to allow the URI to be replaced with a Microsoft-internal URI in official
 # builds which have restricted internet access; see azure-pipelines/signing.yml
@@ -42,6 +43,8 @@ endif()
 
 if(VCPKG_DEPENDENCY_EXTERNAL_FMT)
     find_package(fmt CONFIG REQUIRED)
+elseif(VCPKG_DEPENDENCY_EXTERNAL_FMT_SRC)
+    add_subdirectory("${VCPKG_DEPENDENCY_EXTERNAL_FMT_SRC_DIR}")
 else()
     FetchContent_MakeAvailable(fmt)
 endif()
