diff --git a/source/3rdparty/KHR/CMakeLists.txt b/source/3rdparty/KHR/CMakeLists.txt
index 25fac3a7..c8aa7e60 100644
--- a/source/3rdparty/KHR/CMakeLists.txt
+++ b/source/3rdparty/KHR/CMakeLists.txt
@@ -125,10 +125,6 @@ install(TARGETS ${target}
 )
 
 # Header files
-install(DIRECTORY
-    ${CMAKE_CURRENT_SOURCE_DIR}/include/KHR DESTINATION ${INSTALL_INCLUDE}
-    COMPONENT 3rdparty-dev
-)
 
 # CMake config
 install(EXPORT ${target}-export
