diff -urN cctools-806-old/ld/pass1.c cctools-806/ld/pass1.c
--- cctools-806-old/ld/pass1.c	2009-12-16 13:36:33.000000000 -0600
+++ cctools-809/ld/pass1.c	2011-09-07 01:45:04.000000000 -0500
@@ -508,7 +508,7 @@
 	    pass1_object(file_name, file_addr, file_size, base_name, FALSE,
 			 FALSE, bundle_loader, force_weak);
 	}
-#ifdef VM_SYNC_DEACTIVATE
+#if defined(VM_SYNC_DEACTIVATE) && !defined(_POSIX_C_SOURCE) && !defined(__CYGWIN__)
 	vm_msync(mach_task_self(), (vm_address_t)file_addr,
 		 (vm_size_t)file_size, VM_SYNC_DEACTIVATE);
 #endif /* VM_SYNC_DEACTIVATE */
diff -urN cctools-806-old/ld/pass2.c cctools-806/ld/pass2.c
--- cctools-806-old/ld/pass2.c	2009-12-17 13:02:47.000000000 -0600
+++ cctools-809/ld/pass2.c	2011-09-07 01:45:04.000000000 -0500
@@ -269,7 +269,7 @@
 		    }
 		}
 		output_local_symbols();
-#ifdef VM_SYNC_DEACTIVATE
+#if defined(VM_SYNC_DEACTIVATE) && !defined(_POSIX_C_SOURCE) && !defined(__CYGWIN__)
 		vm_msync(mach_task_self(), (vm_address_t)cur_obj->obj_addr,
 			 (vm_size_t)cur_obj->obj_size, VM_SYNC_DEACTIVATE);
 #endif /* VM_SYNC_DEACTIVATE */

