diff -urN cctools-806-old/ar/archive.c cctools-806/ar/archive.c
--- cctools-806-old/ar/archive.c	1969-12-31 18:00:00.000000000 -0600
+++ cctools-809/ar/archive.c	2011-09-07 01:45:04.000000000 -0500
@@ -84,6 +84,10 @@
 #include "archive.h"
 #include "extern.h"
 
+#ifndef EBADRPC
+#define EBADRPC 72
+#endif
+
 typedef struct ar_hdr HDR;
 static char hb[sizeof(HDR) + 1];	/* real header */
 
@@ -321,7 +325,7 @@
 		 * things for exact binary equality.
 		 */
 		if (getenv("ZERO_AR_DATE") == NULL)
-			tv_sec = (long int)sb->st_mtimespec.tv_sec;
+			tv_sec = (long int)sb->st_mtime;
 		else
 			tv_sec = (long int)0;
 
@@ -340,21 +344,21 @@
 			(void)sprintf(hb, HDR3, name, (long int)tv_sec,
 			    (unsigned int)(u_short)sb->st_uid,
 			    (unsigned int)(u_short)sb->st_gid,
-			    sb->st_mode, sb->st_size, ARFMAG);
+			    sb->st_mode, (long long int)sb->st_size, ARFMAG);
 			lname = 0;
 		} else if (lname > sizeof(hdr->ar_name) || strchr(name, ' '))
 			(void)sprintf(hb, HDR1, AR_EFMT1, (lname + 3) & ~3,
 			    (long int)tv_sec,
 			    (unsigned int)(u_short)sb->st_uid,
 			    (unsigned int)(u_short)sb->st_gid,
-			    sb->st_mode, sb->st_size + ((lname + 3) & ~3),
+			    sb->st_mode, (long long int)(sb->st_size + ((lname + 3) & ~3)),
 			    ARFMAG);
 		else {
 			lname = 0;
 			(void)sprintf(hb, HDR2, name, (long int)tv_sec,
 			    (unsigned int)(u_short)sb->st_uid,
 			    (unsigned int)(u_short)sb->st_gid,
-			    sb->st_mode, sb->st_size, ARFMAG);
+			    sb->st_mode, (long long int)sb->st_size, ARFMAG);
 		}
 		size = sb->st_size;
 	} else {
diff -urN cctools-806-old/ar/archive.h cctools-806/ar/archive.h
--- cctools-806-old/ar/archive.h	2004-11-12 14:16:58.000000000 -0600
+++ cctools-809/ar/archive.h	2011-09-07 01:45:04.000000000 -0500
@@ -112,11 +112,11 @@
 } CHDR;
 
 /* Header format strings. */
-#define	HDR1	"%s%-13d%-12ld%-6u%-6u%-8o%-10qd%2s"
-#define	HDR2	"%-16.16s%-12ld%-6u%-6u%-8o%-10qd%2s"
+#define	HDR1	"%s%-13d%-12ld%-6u%-6u%-8o%-10lld%2s"
+#define	HDR2	"%-16.16s%-12ld%-6u%-6u%-8o%-10lld%2s"
 
 #define	OLDARMAXNAME	15
-#define	HDR3	"%-16.15s%-12ld%-6u%-6u%-8o%-10qd%2s"
+#define	HDR3	"%-16.15s%-12ld%-6u%-6u%-8o%-10lld%2s"
 
 
 #include <sys/cdefs.h> 
diff -urN cctools-806-old/ar/misc.c cctools-806/ar/misc.c
--- cctools-806-old/ar/misc.c	2003-08-12 19:02:39.000000000 -0500
+++ cctools-809/ar/misc.c	2011-09-07 01:45:04.000000000 -0500
@@ -81,6 +81,10 @@
 #include "extern.h"
 #include "pathnames.h"
 
+#ifndef EFTYPE
+#define EFTYPE 79
+#endif
+
 char *tname = "temporary file";		/* temporary file "name" */
 
 int
@@ -105,7 +109,7 @@
 	sigfillset(&set);
 	(void)sigprocmask(SIG_BLOCK, &set, &oset);
 	if ((fd = mkstemp(path)) == -1)
-		error(tname);
+		error2(tname);
         (void)unlink(path);
 	(void)sigprocmask(SIG_SETMASK, &oset, NULL);
 	return (fd);
@@ -169,7 +173,7 @@
 }
 
 void
-error(name)
+error2(name)
 	char *name;
 {
	
 }
