--- cctools-809-b/misc/lipo.c	1969-12-31 18:00:00.000000000 -0600
+++ cctools-809/misc/lipo.c	2011-10-24 23:17:10.000000000 -0500
@@ -1191,21 +1191,21 @@
 		thin->fat_arch.align = 0;
 	    }
 	    else{
-		if(is_llvm_bitcode_from_memory(addr, size, &input->arch_flag,
-					       NULL) != 0){
-		    /* create a thin file struct for it */
-		    thin = new_thin();
-		    thin->name = input->name;
-		    thin->addr = addr;
-		    thin->fat_arch.cputype = input->arch_flag.cputype;
-		    thin->fat_arch.cpusubtype = input->arch_flag.cpusubtype;
-		    thin->fat_arch.offset = 0;
-		    thin->fat_arch.size = size;
-		    thin->fat_arch.align = 0;
-		}
-		else
-		    fatal("can't figure out the architecture type of: %s",
-			  input->name);
+// 		if(is_llvm_bitcode_from_memory(addr, size, &input->arch_flag,
+// 					       NULL) != 0){
+// 		    /* create a thin file struct for it */
+// 		    thin = new_thin();
+// 		    thin->name = input->name;
+// 		    thin->addr = addr;
+// 		    thin->fat_arch.cputype = input->arch_flag.cputype;
+// 		    thin->fat_arch.cpusubtype = input->arch_flag.cpusubtype;
+// 		    thin->fat_arch.offset = 0;
+// 		    thin->fat_arch.size = size;
+// 		    thin->fat_arch.align = 0;
+// 		}
+// 		else
+// 		    fatal("can't figure out the architecture type of: %s",
+// 			  input->name);
 	    }
 	}
 }
@@ -1444,23 +1444,23 @@
 		    if(strncmp(ar_name, SYMDEF, sizeof(SYMDEF) - 1) != 0){
 			ar_addr = addr + offset + ar_name_size;
 			ar_size = strtoul(ar_hdr->ar_size, NULL, 10);
-			if(is_llvm_bitcode_from_memory(ar_addr, ar_size,
-						       &arch_flag, NULL) != 0){
-			    if(*cputype == 0){
-				*cputype = arch_flag.cputype;
-				*cpusubtype = arch_flag.cpusubtype;
-			    }
-			    else if(*cputype != arch_flag.cputype){
-				fatal("archive member %s(%.*s) cputype (%d) "
-				      "and cpusubtype (%d) does not match "
-				      "previous archive members cputype (%d) "
-				      "and cpusubtype (%d) (all members must "
-				      "match)", name, (int)i, ar_name,
-				      arch_flag.cputype, arch_flag.cpusubtype &
-				      ~CPU_SUBTYPE_MASK, *cputype,
-				      (*cpusubtype) & ~CPU_SUBTYPE_MASK);
-			    }
-			}
+// 			if(is_llvm_bitcode_from_memory(ar_addr, ar_size,
+// 						       &arch_flag, NULL) != 0){
+// 			    if(*cputype == 0){
+// 				*cputype = arch_flag.cputype;
+// 				*cpusubtype = arch_flag.cpusubtype;
+// 			    }
+// 			    else if(*cputype != arch_flag.cputype){
+// 				fatal("archive member %s(%.*s) cputype (%d) "
+// 				      "and cpusubtype (%d) does not match "
+// 				      "previous archive members cputype (%d) "
+// 				      "and cpusubtype (%d) (all members must "
+// 				      "match)", name, (int)i, ar_name,
+// 				      arch_flag.cputype, arch_flag.cpusubtype &
+// 				      ~CPU_SUBTYPE_MASK, *cputype,
+// 				      (*cpusubtype) & ~CPU_SUBTYPE_MASK);
+// 			    }
+// 			}
 		    }
 		}
 	    } 
--- cctools-809-b/otool/main.c	1969-12-31 18:00:00.000000000 -0600
+++ cctools-809/otool/main.c	2011-10-24 23:21:16.000000000 -0500
@@ -2403,8 +2403,8 @@
 	    arm_dc = NULL;
 	    thumb_dc = NULL;
 	    if(cputype == CPU_TYPE_ARM){
-		arm_dc = create_arm_llvm_disassembler();
-		thumb_dc = create_thumb_llvm_disassembler();
+// 		arm_dc = create_arm_llvm_disassembler();
+// 		thumb_dc = create_thumb_llvm_disassembler();
 	    }
 	    for(i = offset ; i < size ; ){
 		print_label(cur_addr, TRUE, sorted_symbols, nsorted_symbols);
@@ -2491,10 +2491,10 @@
 		cur_addr += j;
 		i += j;
 	    }
-	    if(arm_dc != NULL)
-		delete_arm_llvm_disassembler(arm_dc);
-	    if(thumb_dc != NULL)
-		delete_arm_llvm_disassembler(thumb_dc);
+// 	    if(arm_dc != NULL)
+// 		delete_arm_llvm_disassembler(arm_dc);
+// 	    if(thumb_dc != NULL)
+// 		delete_arm_llvm_disassembler(thumb_dc);
 	}
 	else{
 	    if(cputype == CPU_TYPE_I386 || cputype == CPU_TYPE_X86_64){
--- cctools-809-b/otool/arm_disasm.c	1969-12-31 18:00:00.000000000 -0600
+++ cctools-809/otool/arm_disasm.c	2011-10-24 23:20:29.000000000 -0500
@@ -461,62 +461,62 @@
 	return(guess_symbol(SymbolValue, dis_info->sorted_symbols,
 			    dis_info->nsorted_symbols, TRUE));
 }
-LLVMDisasmContextRef
-create_arm_llvm_disassembler(
-void)
-{
-    LLVMDisasmContextRef dc;
-
-	dc =
-#ifdef STATIC_LLVM
-	    LLVMCreateDisasm
-#else
-	    llvm_create_disasm
-#endif
-		("arm-apple-darwin10", &dis_info, 1, GetOpInfo, SymbolLookUp);
-	return(dc);
-}
-
-void
-delete_arm_llvm_disassembler(
-LLVMDisasmContextRef dc)
-{
-#ifdef STATIC_LLVM
-	LLVMDisasmDispose
-#else
-	llvm_disasm_dispose
-#endif
-	    (dc);
-}
-
-LLVMDisasmContextRef
-create_thumb_llvm_disassembler(
-void)
-{
-    LLVMDisasmContextRef dc;
-
-	dc =
-#ifdef STATIC_LLVM
-	    LLVMCreateDisasm
-#else
-	    llvm_create_disasm
-#endif
-		("thumbv7-apple-darwin10", &dis_info, 1, GetOpInfo,
-		 SymbolLookUp);
-	return(dc);
-}
-
-void
-delete_thumb_llvm_disassembler(
-LLVMDisasmContextRef dc)
-{
-#ifdef STATIC_LLVM
-	LLVMDisasmDispose
-#else
-	llvm_disasm_dispose
-#endif
-	    (dc);
-}
+// LLVMDisasmContextRef
+// create_arm_llvm_disassembler(
+// void)
+// {
+//     LLVMDisasmContextRef dc;
+// 
+// 	dc =
+// #ifdef STATIC_LLVM
+// 	    LLVMCreateDisasm
+// #else
+// 	    llvm_create_disasm
+// #endif
+// 		("arm-apple-darwin10", &dis_info, 1, GetOpInfo, SymbolLookUp);
+// 	return(dc);
+// }
+
+// void
+// delete_arm_llvm_disassembler(
+// LLVMDisasmContextRef dc)
+// {
+// #ifdef STATIC_LLVM
+// 	LLVMDisasmDispose
+// #else
+// 	llvm_disasm_dispose
+// #endif
+// 	    (dc);
+// }
+
+// LLVMDisasmContextRef
+// create_thumb_llvm_disassembler(
+// void)
+// {
+//     LLVMDisasmContextRef dc;
+// 
+// 	dc =
+// #ifdef STATIC_LLVM
+// 	    LLVMCreateDisasm
+// #else
+// 	    llvm_create_disasm
+// #endif
+// 		("thumbv7-apple-darwin10", &dis_info, 1, GetOpInfo,
+// 		 SymbolLookUp);
+// 	return(dc);
+// }
+
+// void
+// delete_thumb_llvm_disassembler(
+// LLVMDisasmContextRef dc)
+// {
+// #ifdef STATIC_LLVM
+// 	LLVMDisasmDispose
+// #else
+// 	llvm_disasm_dispose
+// #endif
+// 	    (dc);
+// }
 
 #ifdef HACKED_LLVM_DISASSEMBLER_INTERFACE
 // These are defined in lib/Target/ARM/ARMGenInstrNames.inc
@@ -4682,22 +4682,22 @@
     {
       char dst[4096];
       dst[4095] = '\0';
-      if(
-#ifdef STATIC_LLVM
-	 LLVMDisasmInstruction
-#else
-         llvm_disasm_instruction
-#endif
-	    (dc, (uint8_t *)info->sect, size, pc, dst, 4095) != 0)
-	printf("%s", dst);
-      else {
-	if (size == 4)
-	  info->fprintf_func (info->stream, "\t.long\t0x%08x", given);
-	else if (size == 2)
-	  info->fprintf_func (info->stream, "\t.short\t0x%04x", given);
-        else
-	  info->fprintf_func (info->stream, "\tinvalid instruction encoding");
-      }
+//       if(
+// #ifdef STATIC_LLVM
+// 	 LLVMDisasmInstruction
+// #else
+//          llvm_disasm_instruction
+// #endif
+// 	    (dc, (uint8_t *)info->sect, size, pc, dst, 4095) != 0)
+// 	printf("%s", dst);
+//       else {
+// 	if (size == 4)
+// 	  info->fprintf_func (info->stream, "\t.long\t0x%08x", given);
+// 	else if (size == 2)
+// 	  info->fprintf_func (info->stream, "\t.short\t0x%04x", given);
+//         else
+// 	  info->fprintf_func (info->stream, "\tinvalid instruction encoding");
+//       }
     }
 #ifdef HACKED_LLVM_DISASSEMBLER_INTERFACE
   else if (Qflag) /* HACKED interface */
