diff -urN cctools-809-b/include/mach/arm/_structs.h cctools-809/include/mach/arm/_structs.h
--- cctools-809-b/include/mach/arm/_structs.h	1969-12-31 18:00:00.000000000 -0600
+++ cctools-809/include/mach/arm/_structs.h	2011-10-24 23:24:19.000000000 -0500
@@ -25,7 +25,7 @@
 };
 #endif /* __DARWIN_UNIX03 */
 
-#if __DARWIN_UNIX03
+#if 1
 #define _STRUCT_ARM_THREAD_STATE	struct __darwin_arm_thread_state
 _STRUCT_ARM_THREAD_STATE
 {
diff -urN cctools-809-b/include/mach/arm/thread_state.h cctools-809/include/mach/arm/thread_state.h
--- cctools-809-b/include/mach/arm/thread_state.h	1969-12-31 18:00:00.000000000 -0600
+++ cctools-809/include/mach/arm/thread_state.h	2011-10-24 23:24:04.000000000 -0500
@@ -8,6 +8,8 @@
 #ifndef _MACH_ARM_THREAD_STATE_H_
 #define _MACH_ARM_THREAD_STATE_H_
 
+#include "_structs.h"
+
 #define ARM_THREAD_STATE_MAX	(272)
 
 #if defined (__arm__)
