diff --git a/Scripts/Classes/Singletons/Global.gd b/Scripts/Classes/Singletons/Global.gd
index 9884bd9..0e91370 100644
--- a/Scripts/Classes/Singletons/Global.gd
+++ b/Scripts/Classes/Singletons/Global.gd
@@ -169,7 +169,6 @@ func _ready() -> void:
 	get_server_version()
 	if OS.is_debug_build():
 		debug_mode = false
-	setup_discord_rpc()
 	check_for_rom()
 
 func check_for_rom() -> void:
@@ -330,31 +329,19 @@ func close_freeze() -> void:
 var recording_dir = "user://marathon_recordings/"
 
 func setup_discord_rpc() -> void:
-	DiscordRPC.app_id = 1331261692381757562
-	DiscordRPC.start_timestamp = int(Time.get_unix_time_from_system())
-	DiscordRPC.details = "In Title Screen.."
-	if DiscordRPC.get_is_discord_working():
-		DiscordRPC.refresh()
+	return
 
 func set_discord_status(details := "") -> void:
-	DiscordRPC.details = details
-	if DiscordRPC.get_is_discord_working():
-		DiscordRPC.refresh()
+	return
 
 func update_game_status() -> void:
 	var lives_str := str(Global.lives)
 	if Settings.file.difficulty.inf_lives == 1:
 		lives_str = "∞"
 	var string := "Coins = " + str(Global.coins) + " Lives = " + lives_str
-	DiscordRPC.large_image = (Global.level_theme + Global.theme_time).to_lower()
-	DiscordRPC.small_image = Global.current_campaign.to_lower()
-	DiscordRPC.state = string
 
 func refresh_discord_rpc() -> void:
-	if DiscordRPC.get_is_discord_working() == false:
-		return
-	update_game_status()
-	DiscordRPC.refresh()
+	return
 
 func open_marathon_results() -> void:
 	get_node("GameHUD/MarathonResults").open()
