diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8ad7aa7..82821cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,8 +52,7 @@ add_definitions(-DREDUMPER_VERSION_MAJOR=${PROJECT_VERSION_MAJOR}
                 -DREDUMPER_VERSION_BUILD=${REDUMPER_VERSION_BUILD})
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-stdlib=libc++>)
-    add_link_options(-stdlib=libc++ ${REDUMPER_CLANG_LINK_OPTIONS})
+    add_link_options(${REDUMPER_CLANG_LINK_OPTIONS})
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
     add_compile_options(-Wall -Wextra -Werror)
 
