From b08d1a0f6ff05c43f3e220a1590e88ed0b1ebd35 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Sat, 4 Mar 2023 21:40:48 -0500
Subject: [PATCH 2/2] System fmt

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d93dd4b..a32403f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,7 +54,11 @@ add_definitions(-DREDUMPER_VERSION_MAJOR=${PROJECT_VERSION_MAJOR}
 
 # fmt
 # remove this after gcc/clang gets full std::format support
-set(FMT_INCLUDE "${CMAKE_CURRENT_SOURCE_DIR}/fmt/include")
+if (NOT USE_SYSTEM_FMT)
+	set(FMT_INCLUDE "${CMAKE_CURRENT_SOURCE_DIR}/fmt/include")
+else()
+	find_package(fmt)
+endif()
 add_definitions(-DFMT_HEADER_ONLY)
 
 # Visual Studio settings
-- 
2.39.2

