--- Makefile	2012-06-13 07:22:36.000000000 -0700
+++ Makefile.new	2013-09-16 08:14:41.000000000 -0700
@@ -1,7 +1,7 @@
 prefix ?= /usr/local
+real_prefix ?= $(shell echo $(prefix) | sed -e 's:.*portage.*//:/:')
 
 all:
-	$(error "Usage: make [prefix=/install/me/here] install | uninstall")
 
 install:
 	mkdir -p $(prefix)/include/libspotify
@@ -13,9 +13,9 @@
 	cd $(prefix)/lib && ln -sf libspotify.so.12.1.51 libspotify.so.12 && ln -sf libspotify.so.12.1.51 libspotify.so
 
 	mkdir -p $(prefix)/lib/pkgconfig
-	sed -e s:PKG_PREFIX:$(prefix):g <lib/pkgconfig/libspotify.pc >$(prefix)/lib/pkgconfig/libspotify.pc
 
-	ldconfig
+	sed -e s:PKG_PREFIX:$(real_prefix):g <lib/pkgconfig/libspotify.pc > lib/pkgconfig/libspotify.pc2
+	mv lib/pkgconfig/libspotify.pc2 lib/pkgconfig/libspotify.pc
 
 uninstall:
 	rm -rf $(prefix)/include/libspotify
