diff --git a/numba/__init__.py b/numba/__init__.py
index 0e437a600..a8c0cbfeb 100644
--- a/numba/__init__.py
+++ b/numba/__init__.py
@@ -142,8 +142,8 @@ def _ensure_critical_deps():
 
     if numpy_version < (1, 18):
         raise ImportError("Numba needs NumPy 1.18 or greater")
-    elif numpy_version > (1, 23):
-        raise ImportError("Numba needs NumPy 1.23 or less")
+    elif numpy_version > (1, 24):
+        raise ImportError("Numba needs NumPy 1.24 or less")
 
     try:
         import scipy
