diff --git a/Cargo.toml b/Cargo.toml
index 6cd015d..8fe786d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -42,6 +42,7 @@ crossbeam-channel = "0.5.2"
 # noisy-float 0.2 bug
 num-traits = { version = "0.2.14", features = ["i128", "std"] }
 crossbeam-utils = "0.8.11"
+ffmpeg-sys-next = "4.4.0"
 
 [dependencies.ffmpeg]
 package = "ffmpeg-next"
@@ -81,6 +82,3 @@ opt-level = 3
 
 [package.metadata.docs.rs]
 targets = ["x86_64-unknown-linux-gnu"]
-
-[patch.crates-io]
-ffmpeg-sys-next = { rev = "78458039d5fac99354f9cb078869f3be3f3af5fb", git = "https://github.com/kornelski/rust-ffmpeg-sys-1"}
