From a304b908639d55759fd186ac44b4c80356d3ae44 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 13 Jan 2022 12:46:13 -0500
Subject: [PATCH 2/4] settings: disable power off and restart system buttons

Update paths
---
 settings.php | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/settings.php b/settings.php
index 4c55087e..791d44f4 100644
--- a/settings.php
+++ b/settings.php
@@ -10,8 +10,8 @@ require "scripts/pi-hole/php/savesettings.php";
 require_once "scripts/pi-hole/php/FTL.php";
 // Reread ini file as things might have been changed
 // DEFAULT_FTLCONFFILE is set in "scripts/pi-hole/php/FTL.php";
-$setupVars = parse_ini_file("/etc/pihole/setupVars.conf");
-$piholeFTLConf = piholeFTLConfig(DEFAULT_FTLCONFFILE ,true);
+$setupVars = parse_ini_file("/var/lib/pihole/setupVars.conf");
+$piholeFTLConf = piholeFTLConfig(DEFAULT_FTLCONFFILE, true);
 
 // Handling of PHP internal errors
 $last_error = error_get_last();
@@ -337,11 +337,11 @@ if (isset($_GET['tab']) && in_array($_GET['tab'], array("sysadmin", "dns", "piho
                                         </div>
                                         <p class="hidden-md hidden-lg"></p>
                                         <div class="col-md-4">
-                                            <button type="button" class="btn btn-danger confirm-poweroff btn-block">Power off system</button>
+                                            <button type="button" class="btn btn-danger confirm-poweroff btn-block" disabled>Power off system</button>
                                         </div>
                                         <p class="hidden-md hidden-lg"></p>
                                         <div class="col-md-4">
-                                            <button type="button" class="btn btn-danger confirm-reboot btn-block">Restart system</button>
+                                            <button disabled type="button" class="btn btn-danger confirm-reboot btn-block">Restart system</button>
                                         </div>
                                     </div>
 
@@ -555,7 +555,7 @@ if (isset($_GET['tab']) && in_array($_GET['tab'], array("sysadmin", "dns", "piho
                             if ($DHCP) {
                                 // Read leases file
                                 $leasesfile = true;
-                                $dhcpleases = @fopen('/etc/pihole/dhcp.leases', 'r');
+                                $dhcpleases = @fopen('/var/lib/pihole/dhcp.leases', 'r');
                                 if (!is_resource($dhcpleases))
                                     $leasesfile = false;
 
-- 
2.35.1

