From 9e35de1fdfc9818f18c40138b3d34e2bf06780c3 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 13 Jan 2022 12:46:24 -0500
Subject: [PATCH 3/3] footer: remove update message

---
 scripts/pi-hole/php/footer.php | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/scripts/pi-hole/php/footer.php b/scripts/pi-hole/php/footer.php
index e96cf20a..716368d3 100644
--- a/scripts/pi-hole/php/footer.php
+++ b/scripts/pi-hole/php/footer.php
@@ -88,14 +88,6 @@ if (isset($core_commit) || isset($web_commit) || isset($FTL_commit)) {
                         <?php if ($web_update) { ?> &middot; <a class="lookatme" lookatme-text="Update available!" href="<?php echo $webUrl.'/latest'; ?>" rel="noopener" target="_blank">Update available!</a><?php } ?>
                     </li>
                 </ul>
-
-                <p style="margin: 15px 0 0;">
-                    <?php if ($docker_update) { ?>
-                        To install updates, <a href="https://github.com/pi-hole/docker-pi-hole#upgrading-persistence-and-customizations" rel="noopener" target="_blank">replace this old container with a fresh upgraded image</a>.
-                    <?php } elseif ($core_update || $web_update || $FTL_update) { ?>
-                        To install updates, run <code><a href="https://docs.pi-hole.net/main/update/" rel="noopener" target="_blank">pihole -up</a></code>.
-                    <?php } ?>
-                </p>
             </div>
         </div>
     </footer>
-- 
2.44.0

