From dbaff18c741068bb13410cd947b177ce6fbca149 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Wed, 5 Jan 2022 14:56:10 -0500
Subject: [PATCH 3/4] footer: remove update check

---
 scripts/pi-hole/php/footer.php | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/scripts/pi-hole/php/footer.php b/scripts/pi-hole/php/footer.php
index 0d2d2485..8e60490e 100644
--- a/scripts/pi-hole/php/footer.php
+++ b/scripts/pi-hole/php/footer.php
@@ -95,9 +95,6 @@
                         <?php if ($web_update) { ?> &middot; <a class="lookatme" href="<?php echo $webReleasesUrl . "/latest"; ?>" rel="noopener" target="_blank">Update available!</a><?php } ?>
                     </li>
                 </ul>
-                <?php if($core_update || $web_update || $FTL_update) { ?>
-                    <p>To install updates, run <code><a href="https://docs.pi-hole.net/main/update/" rel="noopener" target="_blank">pihole -up</a></code>.</p>
-                <?php } ?>
                 <?php } ?>
             </div>
         </div>
-- 
2.34.1

