diff --git a/scripts/pi-hole/php/header.php b/scripts/pi-hole/php/header.php
index db6519ae..dcb510e6 100644
--- a/scripts/pi-hole/php/header.php
+++ b/scripts/pi-hole/php/header.php
@@ -33,6 +33,10 @@
     {
         $output = rtrim(file_get_contents("/sys/class/hwmon/hwmon0/temp1_input"));
     }
+    elseif (file_exists("/sys/class/hwmon/hwmon1/temp1_input"))
+    {
+        $output = rtrim(file_get_contents("/sys/class/hwmon/hwmon1/temp1_input"));
+    }
     else
     {
         $output = "";
@@ -331,12 +335,7 @@ if($auth) {
                 <div class="pull-left info">
                     <p>Status</p>
                         <?php
-                        $pistatus = pihole_execute('status web');
-                        if (isset($pistatus[0])) {
-                            $pistatus = $pistatus[0];
-                        } else {
-                            $pistatus = null;
-                        }
+                        $pistatus = filetype('/run/pihole/FTL.sock') === 'socket' ? '1' : null;
                         if ($pistatus === "1") {
                             echo '<span id="status"><i class="fa fa-circle text-green-light"></i> Active</span>';
                         } elseif ($pistatus === "0") {
