diff --git a/src/cdrom/cdrom.c b/src/cdrom/cdrom.c
index 52ea97972..723112138 100644
--- a/src/cdrom/cdrom.c
+++ b/src/cdrom/cdrom.c
@@ -1923,7 +1923,7 @@ cdrom_get_current_subchannel(cdrom_t *dev, uint8_t *b, const int msf)
             if (dev->bus_type == CDROM_BUS_ATAPI)
                 break;
             diff = 0;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x01:
             /* Current position. */
             b[1] = subc.attr;
@@ -1971,7 +1971,7 @@ cdrom_get_current_subchannel(cdrom_t *dev, uint8_t *b, const int msf)
             if (b[0] != 0x00)
                 break;
             base += 12;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x02:
             /* UPC  - TODO: Finding and reporting the actual UPC data. */
             memset(&(b[base]), 0x00, 20 - diff);
@@ -1985,7 +1985,7 @@ cdrom_get_current_subchannel(cdrom_t *dev, uint8_t *b, const int msf)
             if (b[0] != 0x00)
                 break;
             base += 16;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x03:
             /* ISRC - TODO: Finding and reporting the actual ISRC data. */
             memset(&(b[base]), 0x00, 20 - diff);
diff --git a/src/cdrom/cdrom_image.c b/src/cdrom/cdrom_image.c
index 6452868ab..2da03f34f 100644
--- a/src/cdrom/cdrom_image.c
+++ b/src/cdrom/cdrom_image.c
@@ -683,7 +683,7 @@ image_cue_get_buffer(char *str, char **line, const int up)
                     done = 1;
                     break;
                 }
-                fallthrough;
+                _86BOX_FALLTHROUGH;
 
             default:
                 if (up && islower((int) *s))
diff --git a/src/cdrom/cdrom_mitsumi.c b/src/cdrom/cdrom_mitsumi.c
index 47501c25c..66fe783b6 100644
--- a/src/cdrom/cdrom_mitsumi.c
+++ b/src/cdrom/cdrom_mitsumi.c
@@ -321,7 +321,7 @@ mitsumi_cdrom_out(uint16_t port, uint8_t val, void *priv)
                                 break;
                             case 5:
                                 dev->readmsf = 0;
-                                fallthrough;
+                                _86BOX_FALLTHROUGH;
                             case 4:
                             case 3:
                                 dev->readmsf |= CD_DCB(val) << ((dev->cmdrd_count - 3) << 3);
diff --git a/src/chipset/ali1621.c b/src/chipset/ali1621.c
index 95602496c..4767cfa42 100644
--- a/src/chipset/ali1621.c
+++ b/src/chipset/ali1621.c
@@ -112,7 +112,7 @@ ali1621_smram_recalc(uint8_t val, ali1621_t *dev)
             switch (val & 0x30) {
                 case 0x10: /* Open. */
                     access_normal = ACCESS_SMRAM_RX;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 0x30: /* Protect. */
                     access_smm |= ACCESS_SMRAM_R;
                     break;
@@ -125,7 +125,7 @@ ali1621_smram_recalc(uint8_t val, ali1621_t *dev)
             switch (val & 0x30) {
                 case 0x10: /* Open. */
                     access_normal |= ACCESS_SMRAM_W;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 0x30: /* Protect. */
                     access_smm |= ACCESS_SMRAM_W;
                     break;
diff --git a/src/chipset/ali6117.c b/src/chipset/ali6117.c
index a9eb96f88..fe5a90740 100644
--- a/src/chipset/ali6117.c
+++ b/src/chipset/ali6117.c
@@ -233,7 +233,7 @@ ali6117_reg_write(uint16_t addr, uint8_t val, void *priv)
 
                 case 0x12:
                     val &= 0xf7;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 case 0x14:
                 case 0x15:
diff --git a/src/chipset/cs8220.c b/src/chipset/cs8220.c
index 34229d536..9a129df2c 100644
--- a/src/chipset/cs8220.c
+++ b/src/chipset/cs8220.c
@@ -165,7 +165,7 @@ cs8220_init(UNUSED(const device_t *info))
             dev->ram_banks[2].virt = 0x00100000;
             dev->ram_banks[2].phys = 0x000a0000;
             dev->ram_banks[2].size = (mem_size << 10) - 0x000a0000;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 640:
             dev->ram_banks[0].virt = 0x00000000;
             dev->ram_banks[0].phys = 0x00000000;
@@ -205,7 +205,7 @@ cs8220_init(UNUSED(const device_t *info))
             dev->ram_banks[2].virt = 0x00100000;
             dev->ram_banks[2].phys = 0x000a0000;
             dev->ram_banks[2].size = (mem_size << 10) - 0x000a0000;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 640:
             dev->ram_banks[0].virt = 0x00000000;
             dev->ram_banks[0].phys = 0x00000000;
diff --git a/src/chipset/opti283.c b/src/chipset/opti283.c
index 12d58fbf3..cdde6a716 100644
--- a/src/chipset/opti283.c
+++ b/src/chipset/opti283.c
@@ -259,7 +259,7 @@ opti283_write(uint16_t addr, uint8_t val, void *priv)
                     }
                     cpu_set_isa_speed((int) round(bus_clk));
                     reset_on_hlt = !!(val & 0x40);
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 }
                 case 0x11:
                 case 0x12:
diff --git a/src/chipset/sis_5513_ide.c b/src/chipset/sis_5513_ide.c
index 9e3a0ad07..83966b45b 100644
--- a/src/chipset/sis_5513_ide.c
+++ b/src/chipset/sis_5513_ide.c
@@ -181,11 +181,11 @@ sis_5513_ide_write(int addr, uint8_t val, void *priv)
                 case 0xd0:
                     if (dev->sis->ide_bits_1_3_writable)
                         val |= 0x0a;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 0x00:
                 case 0xd1:
                     val &= 0xbf;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 0xc0:
                     switch (val & 0x0a) {
                         case 0x00:
@@ -213,12 +213,12 @@ sis_5513_ide_write(int addr, uint8_t val, void *priv)
         /* Primary Base Address */
         case 0x10 ... 0x11:
         case 0x14 ... 0x15:
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         /* Secondary Base Address */
         case 0x18 ... 0x19:
         case 0x1c ... 0x1d:
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         /* Bus Mastering Base Address */
         case 0x20 ... 0x21:
diff --git a/src/chipset/via_pipc.c b/src/chipset/via_pipc.c
index a476b9b99..b0565e1ae 100644
--- a/src/chipset/via_pipc.c
+++ b/src/chipset/via_pipc.c
@@ -1490,7 +1490,7 @@ pipc_write(int func, int addr, uint8_t val, void *priv)
             case 0xd2:
                 if (dev->local == VIA_PIPC_686B)
                     smbus_piix4_setclock(dev->smbus, (val & 0x04) ? 65536 : 16384);
-                fallthrough;
+                _86BOX_FALLTHROUGH;
 
             case 0x90:
             case 0x91:
diff --git a/src/chipset/wd76c10.c b/src/chipset/wd76c10.c
index b70785b37..769eb8b1f 100644
--- a/src/chipset/wd76c10.c
+++ b/src/chipset/wd76c10.c
@@ -750,7 +750,7 @@ wd76c10_shadow_ram_recalc(wd76c10_t *dev)
                 for (uint8_t i = 2; i < 4; i++)
                     bios_states[i] |= 0x03;     /* Read/write. */
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x01:
             for (uint8_t i = 4; i < 8; i++) {
                 bios_states[i] |= 0x01;         /* Read. */
@@ -777,7 +777,7 @@ wd76c10_shadow_ram_recalc(wd76c10_t *dev)
         case 0x02:
             for (uint8_t i = 0; i < vb_top; i++)
                 vbios_states[i] |= 0x04;        /* CSPROM#. */
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x01:
             for (uint8_t i = 4; i < 8; i++) {
                 bios_states[i] |= 0x04;         /* CSPROM#. */
diff --git a/src/cpu/808x.c b/src/cpu/808x.c
index 14e56f23a..5b7012515 100644
--- a/src/cpu/808x.c
+++ b/src/cpu/808x.c
@@ -3449,7 +3449,7 @@ execx86(int cycs)
                         wait_cycs(1, 0);
                         break;
                     }
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 0xD7: /*XLATB*/
                     cpu_state.eaaddr = (BX + AL) & 0xffff;
                     access(4, 8);
diff --git a/src/cpu/cpu.c b/src/cpu/cpu.c
index 8d00da512..f76bf9794 100644
--- a/src/cpu/cpu.c
+++ b/src/cpu/cpu.c
@@ -940,7 +940,7 @@ cpu_set(void)
 #endif  /* USE_DYNAREC */
             x86_setopcodes_2386(ops_2386_386, ops_2386_ibm486_0f);
             cpu_features = CPU_FEATURE_MSR;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case CPU_386SX:
         case CPU_386DX:
             /* In case we get Deskpro 386 emulation */
@@ -1168,7 +1168,7 @@ cpu_set(void)
         case CPU_i486DX_SLENH:
             cpu_features = CPU_FEATURE_CR4 | CPU_FEATURE_VME;
             cpu_CR4_mask = CR4_VME | CR4_PVI | CR4_VME;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case CPU_RAPIDCAD:
         case CPU_i486SX:
         case CPU_i486DX:
@@ -2619,7 +2619,7 @@ cpu_ven_reset(void)
         case CPU_K6_3:
         case CPU_K6_2C:
             msr.amd_psor = (cpu_s->cpu_type >= CPU_K6_3) ? 0x008cULL : 0x018cULL;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case CPU_K6_2:
 #ifdef USE_AMD_K5
         case CPU_K5:
diff --git a/src/cpu/x86_ops_mov_ctrl.h b/src/cpu/x86_ops_mov_ctrl.h
index 3f5d6a4d7..b888a4850 100644
--- a/src/cpu/x86_ops_mov_ctrl.h
+++ b/src/cpu/x86_ops_mov_ctrl.h
@@ -106,7 +106,7 @@ opMOV_r_DRx_a16(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 6:
             cpu_state.regs[cpu_rm].l = dr[6];
             break;
@@ -115,7 +115,7 @@ opMOV_r_DRx_a16(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 7:
             cpu_state.regs[cpu_rm].l = dr[7];
             break;
@@ -150,7 +150,7 @@ opMOV_r_DRx_a32(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 6:
             cpu_state.regs[cpu_rm].l = dr[6];
             break;
@@ -159,7 +159,7 @@ opMOV_r_DRx_a32(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 7:
             cpu_state.regs[cpu_rm].l = dr[7];
             break;
@@ -325,7 +325,7 @@ opMOV_DRx_r_a16(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 6:
             dr[6] = (dr[6] & 0xffff0ff0) | (cpu_state.regs[cpu_rm].l & 0x0000f00f);
             break;
@@ -334,7 +334,7 @@ opMOV_DRx_r_a16(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 7:
             dr[7] = cpu_state.regs[cpu_rm].l | 0x00000400;
             break;
@@ -372,7 +372,7 @@ opMOV_DRx_r_a32(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 6:
             dr[6] = (dr[6] & 0xffff0ff0) | (cpu_state.regs[cpu_rm].l & 0x0000f00f);
             break;
@@ -381,7 +381,7 @@ opMOV_DRx_r_a32(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 7:
             dr[7] = cpu_state.regs[cpu_rm].l | 0x00000400;
             break;
diff --git a/src/cpu/x86_ops_mov_ctrl_2386.h b/src/cpu/x86_ops_mov_ctrl_2386.h
index 0d13cc833..3d2784bdd 100644
--- a/src/cpu/x86_ops_mov_ctrl_2386.h
+++ b/src/cpu/x86_ops_mov_ctrl_2386.h
@@ -104,7 +104,7 @@ opMOV_r_DRx_a16(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 6:
             cpu_state.regs[cpu_rm].l = dr[6];
             break;
@@ -113,7 +113,7 @@ opMOV_r_DRx_a16(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 7:
             cpu_state.regs[cpu_rm].l = dr[7];
             break;
@@ -146,7 +146,7 @@ opMOV_r_DRx_a32(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 6:
             cpu_state.regs[cpu_rm].l = dr[6];
             break;
@@ -155,7 +155,7 @@ opMOV_r_DRx_a32(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 7:
             cpu_state.regs[cpu_rm].l = dr[7];
             break;
@@ -310,7 +310,7 @@ opMOV_DRx_r_a16(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 6:
             dr[6] = (dr[6] & 0xffff0ff0) | (cpu_state.regs[cpu_rm].l & 0x0000f00f);
             break;
@@ -319,7 +319,7 @@ opMOV_DRx_r_a16(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 7:
             dr[7] = cpu_state.regs[cpu_rm].l | 0x00000400;
             break;
@@ -353,7 +353,7 @@ opMOV_DRx_r_a32(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 6:
             dr[6] = (dr[6] & 0xffff0ff0) | (cpu_state.regs[cpu_rm].l & 0x0000f00f);
             break;
@@ -362,7 +362,7 @@ opMOV_DRx_r_a32(uint32_t fetchdat)
                 x86illegal();
                 return 1;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 7:
             dr[7] = cpu_state.regs[cpu_rm].l | 0x00000400;
             break;
diff --git a/src/cpu/x86seg.c b/src/cpu/x86seg.c
index 9b8afa0e5..fb5089027 100644
--- a/src/cpu/x86seg.c
+++ b/src/cpu/x86seg.c
@@ -702,7 +702,7 @@ loadcsjmp(uint16_t seg, uint32_t old_pc)
                                 x86gpf("loadcsjmp(): Non-conforming DPL > CPL", seg2 & 0xfffc);
                                 return;
                             }
-                            fallthrough;
+                            _86BOX_FALLTHROUGH;
                         case 0x1c00:
                         case 0x1d00:
                         case 0x1e00:
@@ -1207,7 +1207,7 @@ loadcscall(uint16_t seg)
                                 x86gpf("loadcscall(): Call PM Gate Inner DPL > CPL", seg2 & 0xfffc);
                                 return;
                             }
-                            fallthrough;
+                            _86BOX_FALLTHROUGH;
                         case 0x1c00:
                         case 0x1d00:
                         case 0x1e00:
@@ -1339,7 +1339,7 @@ pmoderetf(int is32, uint16_t off)
             case 0x1200:
             case 0x1300:
                  /* Data segment, apparently valid when CPL is the same, used by MS LINK for DOS. */
-                 fallthrough;
+                 _86BOX_FALLTHROUGH;
             case 0x1800:
             case 0x1900:
             case 0x1a00:
@@ -1395,7 +1395,7 @@ pmoderetf(int is32, uint16_t off)
             case 0x1200:
             case 0x1300:
                  /* Data segment, apparently valid when CPL is the same, used by MS LINK for DOS. */
-                 fallthrough;
+                 _86BOX_FALLTHROUGH;
             case 0x1800:
             case 0x1900:
             case 0x1a00:
@@ -1622,7 +1622,7 @@ pmodeint(int num, int soft)
                 case 0x1200:
                 case 0x1300:
                      /* Data segment, apparently valid when CPL is the same, used by MS CodeView for DOS. */
-                     fallthrough;
+                     _86BOX_FALLTHROUGH;
                 case 0x1800:
                 case 0x1900:
                 case 0x1a00:
@@ -1738,7 +1738,7 @@ pmodeint(int num, int soft)
                         x86gpf("pmodeint(): DPL != CPL", seg & 0xfffc);
                         return;
                     }
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 0x1c00:
                 case 0x1d00:
                 case 0x1e00:
@@ -2006,7 +2006,7 @@ pmodeiret(int is32)
         case 0x1200:
         case 0x1300:
              /* Data segment, apparently valid when CPL is the same, used by MS CodeView for DOS. */
-             fallthrough;
+             _86BOX_FALLTHROUGH;
         case 0x1800:
         case 0x1900:
         case 0x1a00:
diff --git a/src/device/hwm_lm78.c b/src/device/hwm_lm78.c
index 27741fd6d..523c96df9 100644
--- a/src/device/hwm_lm78.c
+++ b/src/device/hwm_lm78.c
@@ -135,7 +135,7 @@ lm78_nvram_read(UNUSED(void *bus), UNUSED(uint8_t addr), void *priv)
     switch (dev->as99127f.nvram_i2c_state) {
         case 0:
             dev->as99127f.nvram_i2c_state = 1;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 1:
             ret = dev->as99127f.regs[0][0x0b] & 0x3f;
diff --git a/src/device/isamem.c b/src/device/isamem.c
index 0cd5fddea..9042fc9d8 100644
--- a/src/device/isamem.c
+++ b/src/device/isamem.c
@@ -583,7 +583,7 @@ isamem_init(const device_t *info)
         case ISAMEM_LOTECH_EMS_CARD: /* Lotech EMS */
             /* The Lotech EMS cannot have more than 4096KB per board. */
             ems_max = EMS_LOTECH_MAXSIZE;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case ISAMEM_BRXT_CARD:   /* BocaRAM/XT */
             dev->base_addr[0]   = device_get_config_hex16("base");
             dev->total_size     = device_get_config_int("size");
diff --git a/src/device/kbc_at.c b/src/device/kbc_at.c
index 771b95c2a..a06b3a494 100644
--- a/src/device/kbc_at.c
+++ b/src/device/kbc_at.c
@@ -511,7 +511,7 @@ kbc_at_poll_at(atkbc_t *dev)
         case STATE_KBC_AMI_OUT:
             if (dev->status & STAT_OFULL)
                 break;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case STATE_MAIN_IBF:
         default:
 at_main_ibf:
@@ -635,7 +635,7 @@ kbc_at_poll_ps2(atkbc_t *dev)
         case STATE_KBC_AMI_OUT:
             if (dev->status & STAT_OFULL)
                 break;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case STATE_MAIN_IBF:
         default:
             if (dev->status & STAT_IFULL)
diff --git a/src/device/kbc_at_dev.c b/src/device/kbc_at_dev.c
index 81ab0aee9..cd878df0f 100644
--- a/src/device/kbc_at_dev.c
+++ b/src/device/kbc_at_dev.c
@@ -119,7 +119,7 @@ kbc_at_dev_poll(void *priv)
                 dev->port->wantcmd    = 0;
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case DEV_STATE_MAIN_WANT_IN:
             /* Output command response and then return to main loop #2. */
             if ((dev->port->out_new == -1) && (dev->cmd_queue_start != dev->cmd_queue_end)) {
diff --git a/src/device/lpt.c b/src/device/lpt.c
index f7cc6eda0..9e5bfd20b 100644
--- a/src/device/lpt.c
+++ b/src/device/lpt.c
@@ -394,7 +394,7 @@ lpt_write(const uint16_t port, const uint8_t val, void *priv)
                 }
                 break;
             } else
-                fallthrough;
+                _86BOX_FALLTHROUGH;
         case 0x0400:
             switch (dev->ecr >> 5) {
                 default:
@@ -621,7 +621,7 @@ lpt_read(const uint16_t port, void *priv)
                 }
                 break;
             } else
-                fallthrough;
+                _86BOX_FALLTHROUGH;
         case 0x0400:
             switch (dev->ecr >> 5) {
                 default:
@@ -647,7 +647,7 @@ lpt_read(const uint16_t port, void *priv)
                 ret = 0x00;
                 break;
             } else
-                fallthrough;
+                _86BOX_FALLTHROUGH;
         case 0x0401:
             if ((dev->ecr & 0xe0) == 0xe0) {
                 /* CNFGB */
diff --git a/src/device/mouse_serial.c b/src/device/mouse_serial.c
index 45750ef09..956d1311c 100644
--- a/src/device/mouse_serial.c
+++ b/src/device/mouse_serial.c
@@ -604,7 +604,7 @@ ltsermouse_process_command(mouse_t *dev)
         case 0x41:
             /* Absolute Bit Pad One Packed Binary Format */
             mouse_clear_coords();
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x42: /* Relative Bit Pad One Packed Binary Format */
         case 0x53: /* MM Series Data Format */
         case 0x54: /* Three Byte Packed Binary Format */
@@ -692,7 +692,7 @@ ltsermouse_process_data(mouse_t *dev)
         case 0x2a:
             switch (dev->ib) {
                 default:
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 0x6e:
                     dev->bps = 1200;
                     break;
@@ -808,10 +808,10 @@ ltsermouse_write(UNUSED(struct serial_s *serial), void *priv, uint8_t data)
         case STATE_TRANSMIT:
         case STATE_SKIP_REPORT:
             sermouse_set_period(dev, 0.0);
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         default:
             dev->state = STATE_COMMAND;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case STATE_DATA:
             sermouse_timer(dev);
             break;
diff --git a/src/device/pci_bridge.c b/src/device/pci_bridge.c
index d9aeeffb6..16f3c1797 100644
--- a/src/device/pci_bridge.c
+++ b/src/device/pci_bridge.c
@@ -429,7 +429,7 @@ pci_bridge_reset(void *priv)
             dev->regs[0xdc] = 0x01;
             dev->regs[0xde] = 0x01;
             dev->regs[0xe2] = 0x80;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case PCI_BRIDGE_DEC_21150:
             dev->regs[0x06] = 0x80;
             dev->regs[0x07] = 0x02;
diff --git a/src/device/smbus_ali7101.c b/src/device/smbus_ali7101.c
index 49eb44bc6..5142e3651 100644
--- a/src/device/smbus_ali7101.c
+++ b/src/device/smbus_ali7101.c
@@ -193,7 +193,7 @@ smbus_ali7101_write(uint16_t addr, uint8_t val, void *priv)
 
                 case 0x4:          /* block R/W */
                     timer_bytes++; /* count the SMBus length byte now */
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 default:                   /* unknown */
                     dev->next_stat = 0x20; /* raise DEV_ERR */
diff --git a/src/device/smbus_piix4.c b/src/device/smbus_piix4.c
index ae263b484..f17882ace 100644
--- a/src/device/smbus_piix4.c
+++ b/src/device/smbus_piix4.c
@@ -193,7 +193,7 @@ smbus_piix4_write(uint16_t addr, uint8_t val, void *priv)
                             i2c_write(i2c_smbus, smbus_addr, dev->cmd);
                             timer_bytes++;
                         }
-                        fallthrough;
+                        _86BOX_FALLTHROUGH;
 
                     case 0xc:        /* I2C process call */
                         if (!read) { /* word write (only when writing) */
@@ -211,7 +211,7 @@ smbus_piix4_write(uint16_t addr, uint8_t val, void *priv)
 
                     case 0x5:          /* block R/W */
                         timer_bytes++; /* count the SMBus length byte now */
-                        fallthrough;
+                        _86BOX_FALLTHROUGH;
 
                     case 0xd: /* I2C block R/W */
                         i2c_write(i2c_smbus, smbus_addr, dev->cmd);
@@ -243,7 +243,7 @@ smbus_piix4_write(uint16_t addr, uint8_t val, void *priv)
                         /* command write */
                         i2c_write(i2c_smbus, smbus_addr, dev->cmd);
                         timer_bytes++;
-                        fallthrough;
+                        _86BOX_FALLTHROUGH;
 
                     case 0xe:        /* I2C with 7-bit address */
                         if (!read) { /* word write (only when writing) */
diff --git a/src/device/smbus_sis5595.c b/src/device/smbus_sis5595.c
index 42d1452ad..64a5cb29d 100644
--- a/src/device/smbus_sis5595.c
+++ b/src/device/smbus_sis5595.c
@@ -261,7 +261,7 @@ smbus_sis5595_write_data(void *priv, uint8_t val)
                     case 0x5:                    /* block R/W */
                         dev->block_ptr = 0x00;
                         timer_bytes++;           /* count the SMBus length byte now */
-                        fallthrough;
+                        _86BOX_FALLTHROUGH;
 
                     default:                     /* unknown */
                         dev->next_stat = 0x0010; /* raise DEV_ERR */
diff --git a/src/disk/hdc_esdi_at.c b/src/disk/hdc_esdi_at.c
index 4838c23d3..eb588f66a 100644
--- a/src/disk/hdc_esdi_at.c
+++ b/src/disk/hdc_esdi_at.c
@@ -373,7 +373,7 @@ esdi_write(uint16_t port, uint8_t val, void *priv)
                             esdi->command &= ~0x03;
                             if (val & 0x02)
                                 fatal("Read with ECC\n");
-                           fallthrough;
+                           _86BOX_FALLTHROUGH;
 
                         case 0xa0:
                             esdi->status = STAT_BUSY;
@@ -432,7 +432,7 @@ esdi_write(uint16_t port, uint8_t val, void *priv)
 
                         default:
                             esdi_at_log("WD1007: bad command %02X\n", val);
-                            fallthrough;
+                            _86BOX_FALLTHROUGH;
                         case 0xe8: /*???*/
                             esdi->status = STAT_BUSY;
                             esdi_set_callback(esdi, 200 * HDC_TIME);
@@ -878,7 +878,7 @@ format_error:
 
         default:
             esdi_at_log("WD1007: callback on unknown command %02x\n", esdi->command);
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 0xe8:
             esdi->status = STAT_READY | STAT_ERR | STAT_DSC;
diff --git a/src/disk/hdc_ide.c b/src/disk/hdc_ide.c
index fb844da8f..c0a4d70bc 100644
--- a/src/disk/hdc_ide.c
+++ b/src/disk/hdc_ide.c
@@ -1131,7 +1131,7 @@ ide_atapi_callback(ide_t *ide)
 
                     if (ide->command_stop != NULL)
                         ide->command_stop(ide->sc);
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 1:
                     if ((ide->sc->packet_status == PHASE_COMPLETE) &&
                         (ide->sc->callback == 0.0))
@@ -1183,7 +1183,7 @@ ide_atapi_callback(ide_t *ide)
 
                     if (ide->phase_data_out != NULL)
                         (void) ide->phase_data_out(ide->sc);
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 1:
                     if ((ide->sc->packet_status == PHASE_COMPLETE) &&
                         (ide->sc->callback == 0.0))
@@ -1771,7 +1771,7 @@ ide_writeb(uint16_t addr, uint8_t val, void *priv)
                        disabled, the Read Multiple operation is rejected with an Aborted Com-
                        mand error. */
                     ide->blockcount = 0;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 case WIN_READ:
                 case WIN_READ_NORETRY:
@@ -1826,7 +1826,7 @@ ide_writeb(uint16_t addr, uint8_t val, void *priv)
                     /* Turn on the activity indicator *here* so that it gets turned on
                        less times. */
                     ui_sb_update_icon_write(SB_HDD | hdd[ide->hdd_num].bus_type, 1);
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 case WIN_WRITE:
                 case WIN_WRITE_NORETRY:
diff --git a/src/disk/hdc_st506_xt.c b/src/disk/hdc_st506_xt.c
index de2536d83..0094f374f 100644
--- a/src/disk/hdc_st506_xt.c
+++ b/src/disk/hdc_st506_xt.c
@@ -652,7 +652,7 @@ st506_callback(void *priv)
                 st506_complete(dev);
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case CMD_FORMAT_TRACK:
         case CMD_FORMAT_BAD_TRACK:
             switch (dev->state) {
@@ -699,7 +699,7 @@ st506_callback(void *priv)
                 st506_complete(dev);
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case CMD_READ:
 #if 0
         case CMD_READ_LONG:
@@ -807,7 +807,7 @@ read_error_sent:
                 st506_complete(dev);
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case CMD_WRITE:
 #if 0
         case CMD_WRITE_LONG:
@@ -1652,7 +1652,7 @@ st506_init(const device_t *info)
 
         case ST506_XT_TYPE_ST11R: /* Seagate ST-11R (RLL) */
             dev->spt = RLL_SECTORS;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case ST506_XT_TYPE_ST11M: /* Seagate ST-11M (MFM) */
             dev->nr_err   = ERR_NOT_AVAILABLE;
diff --git a/src/disk/hdc_xta.c b/src/disk/hdc_xta.c
index 8c60caad2..d13ff56e1 100644
--- a/src/disk/hdc_xta.c
+++ b/src/disk/hdc_xta.c
@@ -500,7 +500,7 @@ hdc_callback(void *priv)
 
         case CMD_READ_VERIFY:
             no_data = 1;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case CMD_READ_SECTORS:
             if (!drive->present) {
@@ -525,7 +525,7 @@ hdc_callback(void *priv)
                     dev->buf_len = 512;
 
                     dev->state = STATE_SEND;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 case STATE_SEND:
                     /* Activate the status icon. */
@@ -637,7 +637,7 @@ read_error:
                     dev->buf_len = 512;
 
                     dev->state = STATE_RECV;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 case STATE_RECV:
                     /* Activate the status icon. */
diff --git a/src/disk/hdc_xta_ps1.c b/src/disk/hdc_xta_ps1.c
index 4a4273333..e4ec8cc8b 100644
--- a/src/disk/hdc_xta_ps1.c
+++ b/src/disk/hdc_xta_ps1.c
@@ -646,7 +646,7 @@ do_format(hdc_t *dev, drive_t *drive, ccb_t *ccb)
         fcb = (fcb_t *)dev->data;
 #endif
             dev->state = STATE_FINIT;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case STATE_FINIT:
 do_fmt:
@@ -680,7 +680,7 @@ do_fmt:
 
             /* Done with this track. */
             dev->state = STATE_FDONE;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case STATE_FDONE:
             /* One more track done. */
             if (++start_cyl == end_cyl) {
@@ -745,7 +745,7 @@ hdc_callback(void *priv)
     switch (ccb->cmd) {
         case CMD_READ_VERIFY:
             no_data = 1;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case CMD_READ_SECTORS:
             if (!drive->present) {
@@ -772,7 +772,7 @@ hdc_callback(void *priv)
                     dev->buf_len = (128 << dev->ssb.sect_size);
 
                     dev->state = STATE_SEND;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 case STATE_SEND:
                     /* Activate the status icon. */
@@ -938,7 +938,7 @@ do_send:
 
         case CMD_WRITE_VERIFY:
             no_data = 1;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case CMD_WRITE_SECTORS:
             if (!drive->present) {
@@ -965,7 +965,7 @@ do_send:
                     dev->buf_len = (128 << dev->ssb.sect_size);
 
                     dev->state = STATE_RECV;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 case STATE_RECV:
                     /* Activate the status icon. */
diff --git a/src/disk/mo.c b/src/disk/mo.c
index 5ffe74f78..2a962ae7c 100644
--- a/src/disk/mo.c
+++ b/src/disk/mo.c
@@ -470,7 +470,7 @@ mo_update_request_length(mo_t *dev, int len, int block_len)
                     break;
                 }
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         default:
             dev->packet_len = len;
@@ -1277,7 +1277,7 @@ mo_command(scsi_common_t *sc, const uint8_t *cdb)
                 mo_invalid_field(dev, cdb[1]);
                 return;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case GPCMD_SCSI_RESERVE:
         case GPCMD_SCSI_RELEASE:
         case GPCMD_TEST_UNIT_READY:
@@ -1418,7 +1418,7 @@ mo_command(scsi_common_t *sc, const uint8_t *cdb)
                 mo_read_error(dev);
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case GPCMD_WRITE_6:
         case GPCMD_WRITE_10:
         case GPCMD_WRITE_AND_VERIFY_10:
diff --git a/src/disk/rdisk.c b/src/disk/rdisk.c
index 683b1e963..530e69578 100644
--- a/src/disk/rdisk.c
+++ b/src/disk/rdisk.c
@@ -548,7 +548,7 @@ rdisk_update_request_length(rdisk_t *dev, int len, int block_len)
                     break;
                 }
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         default:
             dev->packet_len = len;
@@ -1265,7 +1265,7 @@ rdisk_command(scsi_common_t *sc, const uint8_t *cdb)
                 rdisk_invalid_field(dev, cdb[1]);
                 return;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case GPCMD_SCSI_RESERVE:
         case GPCMD_SCSI_RELEASE:
         case GPCMD_TEST_UNIT_READY:
@@ -1444,7 +1444,7 @@ rdisk_command(scsi_common_t *sc, const uint8_t *cdb)
                 rdisk_command_complete(dev);
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case GPCMD_WRITE_6:
         case GPCMD_WRITE_10:
         case GPCMD_WRITE_AND_VERIFY_10:
diff --git a/src/floppy/fdc.c b/src/floppy/fdc.c
index de0549ca4..f240fdea7 100644
--- a/src/floppy/fdc.c
+++ b/src/floppy/fdc.c
@@ -1023,7 +1023,7 @@ fdc_write(uint16_t addr, uint8_t val, void *priv)
                                 fdc_bad_command(fdc);
                                 break;
                             }
-                            fallthrough;
+                            _86BOX_FALLTHROUGH;
                         case 0x07: /*Recalibrate*/
                             fdc->pnum = 0;
                             fdc->ptot = 1;
@@ -1066,7 +1066,7 @@ fdc_write(uint16_t addr, uint8_t val, void *priv)
                                 fdc_bad_command(fdc);
                                 break;
                             }
-                            fallthrough;
+                            _86BOX_FALLTHROUGH;
                         case 0x10: /*Get version*/
                         case 0x14: /*Unlock*/
                         case 0x94: /*Lock*/
@@ -1194,7 +1194,7 @@ fdc_write(uint16_t addr, uint8_t val, void *priv)
                             case 0x16: /* Verify */
                                 if (fdc->params[0] & 0x80)
                                     fdc->sc = fdc->params[7];
-                                fallthrough;
+                                _86BOX_FALLTHROUGH;
                             case 0x06: /* Read data */
                             case 0x0c: /* Read deleted data */
                                 fdc_io_command_phase1(fdc, 0);
diff --git a/src/floppy/fdd_86f.c b/src/floppy/fdd_86f.c
index 0c64c41e7..06740bb90 100644
--- a/src/floppy/fdd_86f.c
+++ b/src/floppy/fdd_86f.c
@@ -1993,7 +1993,7 @@ d86f_format_track(int drive, int side, int do_write)
                 if (dev->datac == 3)
                     fdc_stop_id_request(d86f_fdc);
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case FMT_PRETRK_SYNC:
         case FMT_SECTOR_DATA_SYNC:
@@ -2336,7 +2336,7 @@ d86f_turbo_poll(int drive, int side)
         case STATE_0D_SPIN_TO_INDEX:
             dev->sector_count = 0;
             dev->datac        = 5;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case STATE_02_SPIN_TO_INDEX:
             dev->state++;
@@ -2381,7 +2381,7 @@ d86f_turbo_poll(int drive, int side)
             dev->last_sector.id.r = dev->req_sector.id.r;
             dev->last_sector.id.n = dev->req_sector.id.n;
             d86f_handler[drive].set_sector(drive, side, dev->last_sector.id.c, dev->last_sector.id.h, dev->last_sector.id.r, dev->last_sector.id.n);
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case STATE_0A_FIND_ID:
             dev->turbo_pos = 0;
diff --git a/src/include/86box/plat.h b/src/include/86box/plat.h
index dcf7e4d53..7bdbc186a 100644
--- a/src/include/86box/plat.h
+++ b/src/include/86box/plat.h
@@ -102,15 +102,15 @@ extern "C" {
 #endif
 
 #ifdef _MSC_VER
-# define fallthrough do {} while (0) /* fallthrough */
+# define _86BOX_FALLTHROUGH do {} while (0) /* fallthrough */
 #else
 # if __has_attribute(fallthrough)
-#  define fallthrough __attribute__((fallthrough))
+#  define _86BOX_FALLTHROUGH __attribute__((fallthrough))
 # else
 #  if __has_attribute(__fallthrough__)
-#   define fallthrough __attribute__((__fallthrough__))
+#   define _86BOX_FALLTHROUGH __attribute__((__fallthrough__))
 #  endif
-#  define fallthrough do {} while (0) /* fallthrough */
+#  define _86BOX_FALLTHROUGH do {} while (0) /* fallthrough */
 # endif
 #endif
 
diff --git a/src/include/86box/plat_fallthrough.h b/src/include/86box/plat_fallthrough.h
index 2ba000848..ca5a94e38 100644
--- a/src/include/86box/plat_fallthrough.h
+++ b/src/include/86box/plat_fallthrough.h
@@ -17,15 +17,15 @@
 
 #ifndef EMU_PLAT_H
 #ifdef _MSC_VER
-# define fallthrough do {} while (0) /* fallthrough */
+# define _86BOX_FALLTHROUGH do {} while (0) /* fallthrough */
 #else
 # if __has_attribute(fallthrough)
-#  define fallthrough __attribute__((fallthrough))
+#  define _86BOX_FALLTHROUGH __attribute__((fallthrough))
 # else
 #  if __has_attribute(__fallthrough__)
-#   define fallthrough __attribute__((__fallthrough__))
+#   define _86BOX_FALLTHROUGH __attribute__((__fallthrough__))
 #  endif
-#  define fallthrough do {} while (0) /* fallthrough */
+#  define _86BOX_FALLTHROUGH do {} while (0) /* fallthrough */
 # endif
 #endif
 #endif
diff --git a/src/machine/m_elt.c b/src/machine/m_elt.c
index b965486d1..8a535c106 100644
--- a/src/machine/m_elt.c
+++ b/src/machine/m_elt.c
@@ -124,7 +124,7 @@ elt_vid_out(uint16_t addr, uint8_t val, void *priv)
         case 0x3d1:
             if (cga->crtcreg >= 32)
                 return;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         default:
             cga->crtcreg &= 31;
diff --git a/src/mem/catalyst_flash.c b/src/mem/catalyst_flash.c
index da9bfc249..8f58e9ad8 100644
--- a/src/mem/catalyst_flash.c
+++ b/src/mem/catalyst_flash.c
@@ -86,7 +86,7 @@ flash_read(uint32_t addr, void *priv)
         case CMD_READ_AUTO_SELECT:
             if (!dev->is_amd)
                 break;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case CMD_READ_SIGNATURE:
             if (dev->is_amd) {
                 if (addr == 0x00000)
diff --git a/src/network/net_ne2000.c b/src/network/net_ne2000.c
index 3e2e7dd38..79e35bd25 100644
--- a/src/network/net_ne2000.c
+++ b/src/network/net_ne2000.c
@@ -916,7 +916,7 @@ nic_pci_write(UNUSED(int func), int addr, uint8_t val, void *priv)
         case 0x10:       /* PCI_BAR */
             val &= 0xe0; /* 0xe0 acc to RTL DS */
             val |= 0x01; /* re-enable IOIN bit */
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 0x11: /* PCI_BAR */
         case 0x12: /* PCI_BAR */
diff --git a/src/network/net_pcnet.c b/src/network/net_pcnet.c
index 0c97d1078..8d98d13ba 100644
--- a/src/network/net_pcnet.c
+++ b/src/network/net_pcnet.c
@@ -2000,7 +2000,7 @@ pcnet_bcr_writew(nic_t *dev, uint16_t rap, uint16_t val)
                     break;
             }
             dev->aCSR[58] = val;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case BCR_LNKST:
         case BCR_LED1:
         case BCR_LED2:
diff --git a/src/network/net_wd8003.c b/src/network/net_wd8003.c
index 100391eb0..f0d0a535a 100644
--- a/src/network/net_wd8003.c
+++ b/src/network/net_wd8003.c
@@ -732,7 +732,7 @@ wd_init(const device_t *info)
         /* Ethernet, MCA, 5x3 interface chip, RAM 16k */
         case WD8003EA:
             dev->board_chip = WE_ID_SOFT_CONFIG;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         /* Ethernet, MCA, no interface chip, RAM 16k */
         case WD8003ETA:
diff --git a/src/pci_dummy.c b/src/pci_dummy.c
index bceb58c22..41c40f4ee 100644
--- a/src/pci_dummy.c
+++ b/src/pci_dummy.c
@@ -218,7 +218,7 @@ pci_dummy_pci_write(int func, int addr, uint8_t val, void *priv)
 
             case 0x10:       /* PCI_BAR */
                 val &= 0xe0; /* 0xe0 acc to RTL DS */
-                fallthrough;
+                _86BOX_FALLTHROUGH;
 
             case 0x11: /* PCI_BAR */
                 /* Remove old I/O. */
diff --git a/src/printer/prt_escp.c b/src/printer/prt_escp.c
index a24072971..b711ec047 100644
--- a/src/printer/prt_escp.c
+++ b/src/printer/prt_escp.c
@@ -1467,7 +1467,7 @@ process_char(escp_t *dev, uint8_t ch)
             dev->curr_x = dev->left_margin;
             if (!dev->autofeed)
                 return 1;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 0x0a: /* Line feed */
             if (dev->font_style & STYLE_DOUBLEWIDTHONELINE) {
diff --git a/src/printer/prt_text.c b/src/printer/prt_text.c
index af601af75..6380180ee 100644
--- a/src/printer/prt_text.c
+++ b/src/printer/prt_text.c
@@ -283,7 +283,7 @@ process_char(prnt_t *dev, uint8_t ch)
             dev->curr_x = 0;
             if (!dev->autofeed)
                 return 1;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 0x0a: /* Line feed */
             dev->curr_x = 0;
diff --git a/src/scsi/scsi_buslogic.c b/src/scsi/scsi_buslogic.c
index daeb0a3e4..46a35fdb9 100644
--- a/src/scsi/scsi_buslogic.c
+++ b/src/scsi/scsi_buslogic.c
@@ -872,7 +872,7 @@ buslogic_cmds(void *priv)
                 dev->Status |= STAT_INVCMD;
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x92:
             if ((bl->chip == CHIP_BUSLOGIC_ISA_542B_1991_12_14) || (bl->chip == CHIP_BUSLOGIC_ISA_545S_1992_10_05) || (bl->chip == CHIP_BUSLOGIC_ISA_542BH_1993_05_23) || (bl->chip == CHIP_BUSLOGIC_MCA_640A_1993_05_23)) {
                 dev->DataReplyLeft = 0;
@@ -1233,7 +1233,7 @@ BuslogicPCIWrite(UNUSED(int func), int addr, uint8_t val, void *priv)
         case 0x10:
             val &= 0xe0;
             val |= 1;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 0x11:
         case 0x12:
@@ -1257,7 +1257,7 @@ BuslogicPCIWrite(UNUSED(int func), int addr, uint8_t val, void *priv)
 
         case 0x14:
             val &= 0xe0;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 0x15:
         case 0x16:
diff --git a/src/scsi/scsi_cdrom.c b/src/scsi/scsi_cdrom.c
index ca5d237b9..d25dc0a77 100644
--- a/src/scsi/scsi_cdrom.c
+++ b/src/scsi/scsi_cdrom.c
@@ -565,7 +565,7 @@ scsi_cdrom_update_request_length(scsi_cdrom_t *dev, int len, const int block_len
                 dev->packet_len   = len;
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x08:
         case 0x28:
         case 0xa8:
@@ -613,7 +613,7 @@ scsi_cdrom_update_request_length(scsi_cdrom_t *dev, int len, const int block_len
 
             if ((dev->drv->bus_type != CDROM_BUS_SCSI) && (dev->block_len != 0))
                 dev->requested_blocks = (dev->packet_len  / dev->block_len);
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         default:
             dev->packet_len   = len;
@@ -1114,7 +1114,7 @@ scsi_cdrom_read_blocks(scsi_cdrom_t *dev)
         case GPCMD_READ_CD_MSF_OLD:
         case GPCMD_READ_CD_MSF:
             msf = 1;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case GPCMD_PLAY_CD:
         case GPCMD_READ_CD_OLD:
         case GPCMD_READ_CD:
@@ -2597,7 +2597,7 @@ scsi_cdrom_command(scsi_common_t *sc, const uint8_t *cdb)
                 scsi_cdrom_illegal_opcode(dev, dev->current_cdb[0]);
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case GPCMD_READ_6:
         case GPCMD_READ_10:
         case GPCMD_READ_12:
@@ -2661,7 +2661,7 @@ scsi_cdrom_command(scsi_common_t *sc, const uint8_t *cdb)
                 case GPCMD_READ_CD_MSF_OLD:
                 case GPCMD_READ_CD_MSF:
                     msf = 1;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case GPCMD_PLAY_CD:
                 case GPCMD_READ_CD_OLD:
                 case GPCMD_READ_CD:
diff --git a/src/scsi/scsi_disk.c b/src/scsi/scsi_disk.c
index 6c59ee7ad..5a144de2d 100644
--- a/src/scsi/scsi_disk.c
+++ b/src/scsi/scsi_disk.c
@@ -361,7 +361,7 @@ scsi_disk_update_request_length(scsi_disk_t *dev, int len, const int block_len)
                     break;
                 }
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         default:
             dev->packet_len = len;
@@ -1009,7 +1009,7 @@ scsi_disk_command(scsi_common_t *sc, const uint8_t *cdb)
                 scsi_disk_invalid_field(dev, cdb[1]);
                 return;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case GPCMD_SCSI_RESERVE:
         case GPCMD_SCSI_RELEASE:
         case GPCMD_TEST_UNIT_READY:
@@ -1148,7 +1148,7 @@ scsi_disk_command(scsi_common_t *sc, const uint8_t *cdb)
                 scsi_disk_command_complete(dev);
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case GPCMD_WRITE_6:
         case GPCMD_WRITE_10:
         case GPCMD_WRITE_AND_VERIFY_10:
diff --git a/src/scsi/scsi_pcscsi.c b/src/scsi/scsi_pcscsi.c
index 11793f5a9..081c84fbd 100644
--- a/src/scsi/scsi_pcscsi.c
+++ b/src/scsi/scsi_pcscsi.c
@@ -1482,7 +1482,7 @@ esp_reg_write(esp_t *dev, uint32_t saddr, uint32_t val)
     switch (saddr) {
         case ESP_TCHI:
             dev->tchi_written = 1;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case ESP_TCLO:
         case ESP_TCMID:
             esp_log("ESP TCW reg%02x = %02x.\n", saddr, val);
diff --git a/src/scsi/scsi_x54x.c b/src/scsi/scsi_x54x.c
index b4256f94d..15dbd7095 100644
--- a/src/scsi/scsi_x54x.c
+++ b/src/scsi/scsi_x54x.c
@@ -484,7 +484,7 @@ x54x_bios_command(x54x_t *x54x, uint8_t max_id, BIOSCMD *cmd, int8_t islba)
 
             default:
                 x54x_log("BIOS: Unimplemented command: %02X\n", cmd->command);
-                fallthrough;
+                _86BOX_FALLTHROUGH;
             case 0x05: /* Format Track, invalid since SCSI has no tracks */
             case 0x0a: /* ???? */
             case 0x0b: /* ???? */
diff --git a/src/sio/sio_f82c710.c b/src/sio/sio_f82c710.c
index 46cb61a40..cd2973a1b 100644
--- a/src/sio/sio_f82c710.c
+++ b/src/sio/sio_f82c710.c
@@ -259,7 +259,7 @@ f82c710_config_write(uint16_t port, uint8_t val, void *priv)
                 case 0x03:
                 case 0x07: case 0x08:
                     /* TODO: Reserved - is it actually writable? */
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 0x09: case 0x0a:
                 case 0x0b:
                     dev->regs[dev->cri] = val;
diff --git a/src/sio/sio_it86x1f.c b/src/sio/sio_it86x1f.c
index 77011de26..3b63804de 100644
--- a/src/sio/sio_it86x1f.c
+++ b/src/sio/sio_it86x1f.c
@@ -325,7 +325,7 @@ it8671f_pnp_config_changed(uint8_t ld, isapnp_device_config_t *config, void *pri
     switch (ld) {
         case 2:
             it8661f_pnp_config_changed(4, config, dev); /* just for logging, should change if IR UART is implemented */
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 0 ... 1:
         case 3:
@@ -429,7 +429,7 @@ it86x1f_pnp_write_vendor_reg(uint8_t ld, uint8_t reg, uint8_t val, void *priv)
 
         case 0x25:
             val &= (CHIP_ID == ITE_IT8661F) ? 0x1f : 0xf0;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 0x26:
             if (ld == dev->gpio_ldn)
@@ -543,7 +543,7 @@ it86x1f_pnp_write_vendor_reg(uint8_t ld, uint8_t reg, uint8_t val, void *priv)
                 case 0x772:
                     if (CHIP_ID != ITE_IT8671F)
                         break;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 case 0x761:
                 case 0x763:
@@ -575,7 +575,7 @@ it86x1f_pnp_write_vendor_reg(uint8_t ld, uint8_t reg, uint8_t val, void *priv)
                 case 0x7fb:
                     if (CHIP_ID == ITE_IT8671F)
                         val &= 0x7f;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 case 0x7f0 ... 0x7f5:
                     dev->ldn_regs[ld][reg & 0x0f] = val;
@@ -662,7 +662,7 @@ it86x1f_write_data(UNUSED(uint16_t port), uint8_t val, void *priv)
                 dev->locked = 1;
                 it86x1f_remap(dev, 0, 0);
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         default:
             isapnp_write_reg(dev->pnp_card, dev->cur_ldn, dev->cur_reg, val);
diff --git a/src/sound/snd_ac97_codec.c b/src/sound/snd_ac97_codec.c
index c028e429d..668fd5560 100644
--- a/src/sound/snd_ac97_codec.c
+++ b/src/sound/snd_ac97_codec.c
@@ -509,7 +509,7 @@ modem_gain:
 
         case 0x50: /* GPIO Pin Sticky */
             dev->regs[0x54 >> 1] &= val; /* clear sticky inputs that are no longer sticky (assumed undefined behavior) */
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 0x52: /* GPIO Pin Wake-up Mask */
             val &= ac97_codecs[dev->model].gpi_mask;
@@ -532,7 +532,7 @@ modem_gain:
                 /* Get actual previous value. */
                 prev = dev->vendor_reg_pages[(i << 3) | ((reg & 0x0e) >> 1)];
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 0x5a ... 0x5e: /* Vendor Reserved */
         case 0x70 ... 0x7a:
diff --git a/src/sound/snd_ad1848.c b/src/sound/snd_ad1848.c
index 44e328842..9fc395697 100644
--- a/src/sound/snd_ad1848.c
+++ b/src/sound/snd_ad1848.c
@@ -315,7 +315,7 @@ ad1848_write(uint16_t addr, uint8_t val, void *priv)
                 case 10:
                     if (ad1848->type < AD1848_TYPE_CS4232)
                         break;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
 
                 case 8:
                     updatefreq = 1;
diff --git a/src/sound/snd_cmi8x38.c b/src/sound/snd_cmi8x38.c
index 9ee003689..a110c284d 100644
--- a/src/sound/snd_cmi8x38.c
+++ b/src/sound/snd_cmi8x38.c
@@ -467,7 +467,7 @@ cmi8x38_sb_mixer_write(uint16_t addr, uint8_t val, void *priv)
             case 0xf8 ... 0xff:
                 if (dev->type == CMEDIA_CMI8338)
                     mixer->regs[mixer->index] = val;
-                fallthrough;
+                _86BOX_FALLTHROUGH;
 
             case 0xf1 ... 0xf7:
                 return;
diff --git a/src/sound/snd_covox.c b/src/sound/snd_covox.c
index 0d6b1dbc1..95e3db919 100644
--- a/src/sound/snd_covox.c
+++ b/src/sound/snd_covox.c
@@ -174,7 +174,7 @@ covox_init(UNUSED(const device_t *info))
     switch (info->local) {
         case COVOX_SOUNDMAN:
             fixed_address = 1;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case COVOX_SOUNDMASTERPLUS:
             has_adlib = 1;
             break;
diff --git a/src/sound/snd_cs423x.c b/src/sound/snd_cs423x.c
index 4fd630e07..3c95d8265 100644
--- a/src/sound/snd_cs423x.c
+++ b/src/sound/snd_cs423x.c
@@ -358,7 +358,7 @@ cs423x_write(uint16_t addr, uint8_t val, void *priv)
                     switch (val) {
                         case 0x55: /* Disable PnP Key */
                             dev->pnp_enable = 0;
-                            fallthrough;
+                            _86BOX_FALLTHROUGH;
 
                         case 0x5a: /* Update Hardware Configuration Data */
                             cs423x_pnp_enable(dev, 0, 1);
diff --git a/src/sound/snd_gus.c b/src/sound/snd_gus.c
index 8d70e61c5..b858b3908 100644
--- a/src/sound/snd_gus.c
+++ b/src/sound/snd_gus.c
@@ -727,7 +727,7 @@ gus_write(uint16_t addr, uint8_t val, void *priv)
                 else if (gus->irq != -1)
                     picint(1 << gus->irq);
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x20d:
             gus->sb_2xc = val;
             break;
@@ -771,7 +771,7 @@ gus_write(uint16_t addr, uint8_t val, void *priv)
                 }
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x706:
             if (gus->type == GUS_CLASSIC_37) {
                 gus->ics2101.addr = val & 0x3F;
@@ -1022,7 +1022,7 @@ gus_read(uint16_t addr, void *priv)
         case 0x388:
             if ((gus->type == GUS_ACE) && !device_get_config_int("adlib_ports"))
                 break;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x208:
             if (gus->tctrl & GUS_TIMER_CTRL_AUTO)
                 val = gus->sb_2xa;
diff --git a/src/sound/snd_sb_dsp.c b/src/sound/snd_sb_dsp.c
index bfc2847ed..addfbd96b 100644
--- a/src/sound/snd_sb_dsp.c
+++ b/src/sound/snd_sb_dsp.c
@@ -1380,7 +1380,7 @@ sb_exec_command(sb_dsp_t *dsp)
         case 0x17: /* 2-bit ADPCM output with reference */
             dsp->sbref  = dsp->dma_readb(dsp->dma_priv);
             dsp->sbstep = 0;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x16: /* 2-bit ADPCM output */
             sb_start_dma(dsp, 1, 0, ADPCM_2, dsp->sb_data[0] + (dsp->sb_data[1] << 8));
             dsp->sbdat2 = dsp->dma_readb(dsp->dma_priv);
@@ -1542,7 +1542,7 @@ sb_exec_command(sb_dsp_t *dsp)
         case 0x75: /* 4-bit ADPCM output with reference */
             dsp->sbref  = dsp->dma_readb(dsp->dma_priv);
             dsp->sbstep = 0;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x74: /* 4-bit ADPCM output */
             sb_start_dma(dsp, 1, 0, ADPCM_4, dsp->sb_data[0] + (dsp->sb_data[1] << 8));
             dsp->sbdat2 = dsp->dma_readb(dsp->dma_priv);
@@ -1556,7 +1556,7 @@ sb_exec_command(sb_dsp_t *dsp)
         case 0x77: /* 2.6-bit ADPCM output with reference */
             dsp->sbref  = dsp->dma_readb(dsp->dma_priv);
             dsp->sbstep = 0;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x76: /* 2.6-bit ADPCM output */
             sb_start_dma(dsp, 1, 0, ADPCM_26, dsp->sb_data[0] + (dsp->sb_data[1] << 8));
             dsp->sbdat2 = dsp->dma_readb(dsp->dma_priv);
diff --git a/src/video/ramdac/vid_ramdac_tvp3026.c b/src/video/ramdac/vid_ramdac_tvp3026.c
index 73af99075..10f33fe9c 100644
--- a/src/video/ramdac/vid_ramdac_tvp3026.c
+++ b/src/video/ramdac/vid_ramdac_tvp3026.c
@@ -113,7 +113,7 @@ tvp3026_ramdac_out(uint16_t addr, int rs2, int rs3, uint8_t val, void *priv, svg
     switch (rs) {
         case 0x00: /* Palette Write Index Register (RS value = 0000) */
             ramdac->ind_idx = val;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x04: /* Ext Palette Write Index Register (RS value = 0100) */
         case 0x03:
         case 0x07: /* Ext Palette Read Index Register (RS value = 0111) */
diff --git a/src/video/vid_8514a.c b/src/video/vid_8514a.c
index d521c07d5..923c2e80d 100644
--- a/src/video/vid_8514a.c
+++ b/src/video/vid_8514a.c
@@ -4026,7 +4026,7 @@ ibm8514_init(const device_t *info)
                 break;
             }
 
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         default:
             dev->extensions = 0;
             ibm8514_io_set(svga);
diff --git a/src/video/vid_ati_eeprom.c b/src/video/vid_ati_eeprom.c
index 8baa72351..3584f490a 100644
--- a/src/video/vid_ati_eeprom.c
+++ b/src/video/vid_ati_eeprom.c
@@ -121,7 +121,7 @@ ati_eeprom_write(ati_eeprom_t *eeprom, int ena, int clk, int dat)
                     if (!dat)
                         break;
                     eeprom->state = EEPROM_OPCODE;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case EEPROM_OPCODE:
                     eeprom->opcode = (eeprom->opcode << 1) | (dat ? 1 : 0);
                     eeprom->count--;
diff --git a/src/video/vid_ati_mach64.c b/src/video/vid_ati_mach64.c
index 97c869668..b4112afc1 100644
--- a/src/video/vid_ati_mach64.c
+++ b/src/video/vid_ati_mach64.c
@@ -972,7 +972,7 @@ mach64_accel_write_fifo(mach64_t *mach64, uint32_t addr, uint8_t val)
         case 0x11e:
         case 0x11f:
             WRITE8(addr, mach64->dst_height_width, val);
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x113:
             if (((addr & 0x3ff) == 0x11b || (addr & 0x3ff) == 0x11f || (addr & 0x3ff) == 0x113) && !(val & 0x80)) {
 start_blit_op:
@@ -1212,7 +1212,7 @@ start_blit_op:
         case 0x2a4:
         case 0x2a5:
             addr += 2;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x2aa:
         case 0x2ab:
             WRITE8(addr, mach64->sc_left_right, val);
@@ -1227,7 +1227,7 @@ start_blit_op:
         case 0x2b0:
         case 0x2b1:
             addr += 2;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x2b6:
         case 0x2b7:
             WRITE8(addr, mach64->sc_top_bottom, val);
@@ -3020,7 +3020,7 @@ mach64_ext_readb(uint32_t addr, void *priv)
             case 0x110:
             case 0x111:
                 addr += 2;
-                fallthrough;
+                _86BOX_FALLTHROUGH;
             case 0x114:
             case 0x115:
             case 0x118:
@@ -3195,7 +3195,7 @@ mach64_ext_readb(uint32_t addr, void *priv)
             case 0x2a4:
             case 0x2a5:
                 addr += 2;
-                fallthrough;
+                _86BOX_FALLTHROUGH;
             case 0x2aa:
             case 0x2ab:
                 mach64_wait_fifo_idle(mach64);
@@ -3212,7 +3212,7 @@ mach64_ext_readb(uint32_t addr, void *priv)
             case 0x2b0:
             case 0x2b1:
                 addr += 2;
-                fallthrough;
+                _86BOX_FALLTHROUGH;
             case 0x2b6:
             case 0x2b7:
                 mach64_wait_fifo_idle(mach64);
diff --git a/src/video/vid_ati_mach8.c b/src/video/vid_ati_mach8.c
index 81abadd56..c3e70e332 100644
--- a/src/video/vid_ati_mach8.c
+++ b/src/video/vid_ati_mach8.c
@@ -3794,7 +3794,7 @@ mach_accel_out_fifo(mach_t *mach, svga_t *svga, ibm8514_t *dev, uint16_t port, u
         case 0xaae8:
             if (len == 2)
                 mach->accel.dst_clr_cmp_mask = val;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0xaee8:
         case 0xb2e8:
         case 0xb6e8:
@@ -5098,7 +5098,7 @@ mach_accel_in_call(uint16_t port, mach_t *mach, svga_t *svga, ibm8514_t *dev)
                 dev->fifo_idx--;
             if (mach->fifo_test_idx > 0)
                 mach->fifo_test_idx--;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x1aef:
             mach_log("FIFO Test IDX=%d, Data=%04x.\n", mach->fifo_test_idx, mach->fifo_test_data[mach->fifo_test_idx]);
             READ8(port, mach->fifo_test_data[mach->fifo_test_idx]);
diff --git a/src/video/vid_cl54xx.c b/src/video/vid_cl54xx.c
index 95f9e7bf6..de251fca1 100644
--- a/src/video/vid_cl54xx.c
+++ b/src/video/vid_cl54xx.c
@@ -1005,7 +1005,7 @@ gd54xx_out(uint16_t addr, uint8_t val, void *priv)
                             }
                             svga->seqregs[2] &= 0x0f;
                         }
-                        fallthrough;
+                        _86BOX_FALLTHROUGH;
                     case 0x09:
                     case 0x0a:
                         gd54xx_recalc_banking(gd54xx);
@@ -2516,7 +2516,7 @@ gd54xx_readw_linear(uint32_t addr, void *priv)
         case 2:
             /* 0 -> 3, 1 -> 2, 2 -> 1, 3 -> 0 */
             addr ^= 0x00000002;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 1:
             temp = svga_readb_linear(addr + 1, svga);
             temp |= (svga_readb_linear(addr, svga) << 8);
@@ -2791,7 +2791,7 @@ gd54xx_writew_linear(uint32_t addr, uint16_t val, void *priv)
                 return;
             case 2:
                 addr ^= 0x00000002;
-                fallthrough;
+                _86BOX_FALLTHROUGH;
             case 1:
                 svga_write_linear(addr + 1, val & 0xff, svga);
                 svga_write_linear(addr, val >> 8, svga);
diff --git a/src/video/vid_et4000.c b/src/video/vid_et4000.c
index c17495c1e..797bd3683 100644
--- a/src/video/vid_et4000.c
+++ b/src/video/vid_et4000.c
@@ -525,7 +525,7 @@ et4000_kasan_out(uint16_t addr, uint8_t val, void *priv)
                 case 5:
                     et4000->kasan_cfg_regs[5]              = val;
                     et4000->svga.ksc5601_english_font_type = 0x100 | val;
-                    fallthrough;
+                    _86BOX_FALLTHROUGH;
                 case 6:
                 case 7:
                     et4000->svga.ksc5601_udc_area_msb[et4000->kasan_cfg_index - 0xF6] = val;
diff --git a/src/video/vid_ht216.c b/src/video/vid_ht216.c
index 42a337674..c924762e0 100644
--- a/src/video/vid_ht216.c
+++ b/src/video/vid_ht216.c
@@ -359,7 +359,7 @@ ht216_out(uint16_t addr, uint8_t val, void *priv)
                         if (val & 0x04)
                             svga->adv_flags |= FLAG_RAMDAC_SHIFT;
                         svga_recalctimings(svga);
-                        fallthrough;
+                        _86BOX_FALLTHROUGH;
                     /*Bank registers*/
                     case 0xe8:
                     case 0xe9:
diff --git a/src/video/vid_jega.c b/src/video/vid_jega.c
index 8a34b9f61..bbb82ca2e 100644
--- a/src/video/vid_jega.c
+++ b/src/video/vid_jega.c
@@ -637,7 +637,7 @@ jega_in(uint16_t addr, void *priv)
         case 0x3ba:
         case 0x3da:
             jega->attrff = 0;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         default:
             /* Accessing to Slave is redirected to Master in AX-1. */
             if (jega->regs[RMOD1] & 0x0c) {
diff --git a/src/video/vid_mga.c b/src/video/vid_mga.c
index 07b0869f7..efc5d5df1 100644
--- a/src/video/vid_mga.c
+++ b/src/video/vid_mga.c
@@ -704,7 +704,7 @@ mystique_out(uint16_t addr, uint8_t val, void *priv)
     switch (addr) {
         case 0x3c8:
             mystique->xreg_idx = val;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case 0x3c6:
         case 0x3c7:
         case 0x3c9:
diff --git a/src/video/vid_s3_virge.c b/src/video/vid_s3_virge.c
index 5fa84e555..a0c057604 100644
--- a/src/video/vid_s3_virge.c
+++ b/src/video/vid_s3_virge.c
@@ -4538,7 +4538,7 @@ s3_virge_hwcursor_draw(svga_t *svga, int displine)
                 bg = svga->pallook[virge->hwc_bg_col & 0xff];
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 15:
             if (virge->chip != S3_VIRGEGX2) {
@@ -4546,7 +4546,7 @@ s3_virge_hwcursor_draw(svga_t *svga, int displine)
                 bg = video_15to32[virge->hwc_bg_col & 0xffff];
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 16:
             if (virge->chip != S3_VIRGEGX2) {
@@ -4554,7 +4554,7 @@ s3_virge_hwcursor_draw(svga_t *svga, int displine)
                 bg = video_16to32[virge->hwc_bg_col & 0xffff];
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         case 24:
         case 32:
@@ -5483,7 +5483,7 @@ s3_virge_init(const device_t *info)
 
         case S3_VIRGE_GX:
             virge->virge_rev = 0x01;
-            fallthrough;
+            _86BOX_FALLTHROUGH;
 
         default:
             virge->fifo_slots_num    = 8;
diff --git a/src/video/vid_voodoo_reg.c b/src/video/vid_voodoo_reg.c
index 38ac42845..88c45cd67 100644
--- a/src/video/vid_voodoo_reg.c
+++ b/src/video/vid_voodoo_reg.c
@@ -1121,7 +1121,7 @@ voodoo_reg_writel(uint32_t addr, uint32_t val, void *priv)
                 }
                 break;
             }
-            fallthrough;
+            _86BOX_FALLTHROUGH;
         case SST_nccTable0_I2:
             if (!(val & (1 << 31))) {
                 if (chip & CHIP_TREX0) {
