--- b/src/hardware/voodoo_opengl.cpp
+++ a/src/hardware/voodoo_opengl.cpp
@@ -1745,8 +1745,8 @@
     GFX_ReleaseMouse();
     GFX_ForceFullscreenExit();
 
+	if (new_height<v->fbi.height) new_height = v->fbi.height;
+	if (new_width < v->fbi.width) new_width = v->fbi.width;
-	/*if (new_height<v->fbi.height) */new_height = v->fbi.height;
-	/*if (new_width < v->fbi.width) */new_width = v->fbi.width;
 
 	if (GFX_LazyFullscreenRequested()) GFX_SwitchFullscreenNoReset();
 
--- b/src/output/output_opengl.cpp
+++ a/src/output/output_opengl.cpp
@@ -199,11 +199,7 @@
 
 #if DOSBOXMENU_TYPE == DOSBOXMENU_SDLDRAW
     /* scale the menu bar if the window is large enough */
+    {
-    /* SDL drawn menus cannot coexist with 3Dfx emulation. In fact, there is a serious
-     * bug in SDL1 builds that rapidly expands the vertical size of the menu every frame. */
-    if (Voodoo_OGL_GetWidth() != 0 && Voodoo_OGL_GetHeight() != 0 && Voodoo_OGL_Active() && sdl.desktop.prevent_fullscreen) {
-    }
-    else {
         int cw = fixedWidth, ch = fixedHeight;
         int scale = 1;
 
@@ -259,12 +255,8 @@
 #if DOSBOXMENU_TYPE == DOSBOXMENU_SDLDRAW
     if (mainMenu.isVisible() && !sdl.desktop.fullscreen) 
     {
+        windowHeight += mainMenu.menuBox.h;
+        sdl.clip.y += mainMenu.menuBox.h;
-        if (Voodoo_OGL_GetWidth() != 0 && Voodoo_OGL_GetHeight() != 0 && Voodoo_OGL_Active() && sdl.desktop.prevent_fullscreen) {
-        }
-        else {
-            windowHeight += mainMenu.menuBox.h;
-            sdl.clip.y += mainMenu.menuBox.h;
-        }
     }
 #endif
 
