From ea34b34825feeb8415f7171203f8b2b4d544231e Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Wed, 6 Oct 2021 03:48:34 -0400
Subject: [PATCH 4/7] meson: let version get stale

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index e59b8d98da..6f98a3de8b 100644
--- a/meson.build
+++ b/meson.build
@@ -315,7 +315,7 @@ xemu_version = custom_target('xemu-version-macro.h',
                              command: xemu_version_cmd,
                              capture: true,
                              build_by_default: true,
-                             build_always_stale: true)
+                             build_always_stale: false)
 
 
 ###########################################
@@ -2887,7 +2887,7 @@ qemu_version = custom_target('qemu-version.h',
                              command: qemu_version_cmd,
                              capture: true,
                              build_by_default: true,
-                             build_always_stale: true)
+                             build_always_stale: false)
 
 genconfig_cmd = [
   python, files('genconfig/gen_config.py'),
-- 
2.41.0

