diff --git a/meson.build b/meson.build
index 88e1e34..74e29b1 100644
--- a/meson.build
+++ b/meson.build
@@ -2572,7 +2572,7 @@ foreach target : target_dirs
     target_inc += include_directories('linux-headers', is_system: true)
   endif
   if target.endswith('-softmmu')
-    qemu_target_name = 'qemu-system-' + target_name
+    qemu_target_name = 'xemu'
     target_type='system'
     t = target_softmmu_arch[arch].apply(config_target, strict: false)
     arch_srcs += t.sources()
@@ -2588,7 +2588,7 @@ foreach target : target_dirs
   else
     abi = config_target['TARGET_ABI_DIR']
     target_type='user'
-    qemu_target_name = 'qemu-' + target_name
+    qemu_target_name = 'xemu'
     if arch in target_user_arch
       t = target_user_arch[arch].apply(config_target, strict: false)
       arch_srcs += t.sources()
@@ -2637,7 +2637,7 @@ foreach target : target_dirs
   arch_srcs += target_specific.sources()
   arch_deps += target_specific.dependencies()
 
-  lib = static_library('qemu-' + target,
+  lib = static_library('xemu',
                  sources: arch_srcs + genh,
                  dependencies: arch_deps,
                  objects: objects,
@@ -2648,7 +2648,7 @@ foreach target : target_dirs
 
   if target.endswith('-softmmu')
     execs = [{
-      'name': 'qemu-system-' + target_name,
+      'name': 'xemu',
       'gui': false,
       'sources': files('softmmu/main.c'),
       'dependencies': []
@@ -2672,7 +2672,7 @@ foreach target : target_dirs
     endif
   else
     execs = [{
-      'name': 'qemu-' + target_name,
+      'name': 'xemu',
       'gui': false,
       'sources': [],
       'dependencies': []
diff --git a/tests/qtest/meson.build b/tests/qtest/meson.build
index 2bc3efd..54daec0 100644
--- a/tests/qtest/meson.build
+++ b/tests/qtest/meson.build
@@ -265,7 +265,7 @@ foreach dir : target_dirs
   endif
 
   target_base = dir.split('-')[0]
-  qtest_emulator = emulators['qemu-system-' + target_base]
+  qtest_emulator = emulators['xemu']
   target_qtests = get_variable('qtests_' + target_base, []) + qtests_generic
 
   test_deps = []
@@ -275,7 +275,7 @@ foreach dir : target_dirs
     test_deps += [qemu_img]
   endif
   qtest_env.set('G_TEST_DBUS_DAEMON', meson.source_root() / 'tests/dbus-vmstate-daemon.sh')
-  qtest_env.set('QTEST_QEMU_BINARY', './qemu-system-' + target_base)
+  qtest_env.set('QTEST_QEMU_BINARY', './xemu')
   if have_tools and have_vhost_user_blk_server
     qtest_env.set('QTEST_QEMU_STORAGE_DAEMON_BINARY', './storage-daemon/qemu-storage-daemon')
     test_deps += [qsd]
