From b8b8e1c15b2497008283ed67378765813fd87c10 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 7 Oct 2021 07:24:08 -0400
Subject: [PATCH 6/7] not for upstream: remove trace-events-all

---
 trace/meson.build | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/trace/meson.build b/trace/meson.build
index ef18f11..8efd92c 100644
--- a/trace/meson.build
+++ b/trace/meson.build
@@ -53,14 +53,6 @@ foreach dir : [ '.' ] + trace_events_subdirs
   endif
 endforeach
 
-trace_events_all = custom_target('trace-events-all',
-                                 output: 'trace-events-all',
-                                 input: trace_events_files,
-                                 command: [ 'cat', '@INPUT@' ],
-                                 capture: true,
-                                 install: true,
-                                 install_dir: qemu_datadir)
-
 foreach d : [
   ['generated-tcg-tracers.h', 'tcg-h'],
   ['generated-helpers.c', 'tcg-helper-c'],
-- 
2.33.0

