From 11e006cd93394c893ac91638e0473e3d635016de Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Wed, 6 Oct 2021 03:48:34 -0400
Subject: [PATCH 4/7] meson: let version get stale

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index f490991..3b961ce 100644
--- a/meson.build
+++ b/meson.build
@@ -167,7 +167,7 @@ xemu_version = custom_target('xemu-version-macro.h',
                              command: xemu_version_cmd,
                              capture: true,
                              build_by_default: true,
-                             build_always_stale: true)
+                             build_always_stale: false)
 
 
 ###########################################
@@ -2063,7 +2063,7 @@ qemu_version = custom_target('qemu-version.h',
                              command: qemu_version_cmd,
                              capture: true,
                              build_by_default: true,
-                             build_always_stale: true)
+                             build_always_stale: false)
 
 genconfig_cmd = [
   python, files('genconfig/gen_config.py'),
-- 
2.35.1

