From b30507d163114d41b81b1a19910323cc68ca5269 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 7 Oct 2021 07:12:59 -0400
Subject: [PATCH 5/7] allow use of system xxHash header

---
 configure         | 16 ++++++++++++++++
 meson.build       | 16 ++++++++++++++++
 meson_options.txt |  3 +++
 3 files changed, 35 insertions(+)

diff --git a/configure b/configure
index e2a3ca0a67..859ca59713 100755
--- a/configure
+++ b/configure
@@ -803,6 +803,10 @@ for opt do
   ;;
   --enable-fortify-source) fortify_source="yes"
   ;;
+  --with-xxhash=internal) meson_option_add -Dxxhash=internal
+  ;;
+  --with-xxhash=system) meson_option_add -Dxxhash=system
+  ;;
   --disable-tcg) tcg="disabled"
                  plugins="no"
   ;;
@@ -1042,6 +1046,8 @@ Advanced options (experts only):
   --enable-tsan            enable thread sanitizer
   --disable-werror         disable compilation abort on warning
   --disable-stack-protector disable compiler-provided stack protection
+  --with-xxhash=internal   use internal version of xxHash
+  --with-xxhash=system     link against system xxHash
   --cpu=CPU                Build for host CPU [$cpu]
   --with-coroutine=BACKEND coroutine backend. Supported options:
                            ucontext, sigaltstack, windows
@@ -2214,6 +2220,16 @@ write_target_makefile() {
   fi
 }
 
+##########################################
+# check for xxHash
+case "$xxhash" in
+    system)
+        if ! pkg-config --exists libxxhash ; then
+            error_exit "Cannot find xxHash on system."
+        fi
+    ;;
+esac
+
 ##########################################
 # check for vfio_user_server
 
diff --git a/meson.build b/meson.build
index 6f98a3de8b..7d5b5f7e4e 100644
--- a/meson.build
+++ b/meson.build
@@ -2769,6 +2769,19 @@ if have_system and vfio_user_server_allowed
   libvfio_user_dep = declare_dependency(dependencies: [libvfio_user_lib])
 endif
 
+xxhash = not_found
+xxhash_opt = get_option('xxhash')
+if xxhash_opt == 'system'
+  xxhash = dependency('libxxhash', kwargs: static_kwargs,
+                       method: 'pkg-config',
+                       required: xxhash_opt == 'system')
+  if xxhash.found()
+    xxhash_opt = 'system'
+  else
+    xxhash_opt = 'internal'
+  endif
+endif
+
 fdt = not_found
 if have_system
   fdt_opt = get_option('fdt')
@@ -3479,6 +3492,9 @@ foreach target : target_dirs
 
   config_target += config_host
   target_inc = [include_directories('target' / config_target['TARGET_BASE_ARCH'])]
+  if xxhash_opt == 'internal' and target == 'xemu'
+    target_inc += include_directories('hw/xbox/nv2a/xxHash')
+  endif
   if targetos == 'linux'
     target_inc += include_directories('linux-headers', is_system: true)
   endif
diff --git a/meson_options.txt b/meson_options.txt
index 6d353d4130..2ebb58200a 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -328,3 +328,6 @@ option('slirp_smbd', type : 'feature', value : 'auto',
 
 option('renderdoc', type: 'feature', value: 'auto',
        description: 'improved RenderDoc frame capture support')
+option('xxhash', type: 'combo', value: 'internal',
+       choices: ['internal', 'system'],
+       description: 'Whether and how to find the xxHash library')
-- 
2.41.0

