From 0e82d05592a3870a89702c3e6111e2ccfe9082d0 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Wed, 6 Oct 2021 03:48:34 -0400
Subject: [PATCH 4/8] meson: let version get stale

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 3a4a6a071e..355fd1b90b 100644
--- a/meson.build
+++ b/meson.build
@@ -163,7 +163,7 @@ xemu_version = custom_target('xemu-version-macro.h',
                              command: xemu_version_cmd,
                              capture: true,
                              build_by_default: true,
-                             build_always_stale: true)
+                             build_always_stale: false)
 
 #####################
 # Option validation #
@@ -3619,7 +3619,7 @@ qemu_version = custom_target('qemu-version.h',
                              command: qemu_version_cmd,
                              capture: true,
                              build_by_default: true,
-                             build_always_stale: true)
+                             build_always_stale: false)
 
 xemu_config = custom_target('xemu-config.h',
                             output: 'xemu-config.h',
-- 
2.48.1

