From 616a037d36fa93cd78be2474aef862f1281aa6c7 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Mon, 24 Feb 2025 01:51:53 -0500
Subject: [PATCH 7/9] Not for upstream: misc updates

---
 meson.build | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index a25a77bb98..51a4fc0d53 100644
--- a/meson.build
+++ b/meson.build
@@ -4611,20 +4611,20 @@ foreach target : target_dirs
     execs = [{
       'name': 'xemu',
       'gui': false,
-      'sources': files('softmmu/main.c'),
+      'sources': files('system/main.c'),
       'dependencies': [],
       'win_subsystem': 'console'
     }]
-    if targetos == 'windows' and (sdl.found() or gtk.found())
+    if host_os == 'windows' and (sdl.found() or gtk.found())
       execs += [{
         'name': 'xemuw',
         'gui': true,
-        'sources': files('softmmu/main.c'),
+        'sources': files('system/main.c'),
         'dependencies': [],
         'win_subsystem': 'console'
       }]
     endif
-    if config_host.has_key('CONFIG_FUZZ')
+    if get_option('fuzzing')
       specific_fuzz = specific_fuzz_ss.apply(config_target, strict: false)
       execs += [{
         'name': 'xemu-fuzz',
-- 
2.52.0

