From 4da56ab9a0cb7f82e27e8b4f6b67423cfcf1d1a8 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Wed, 6 Oct 2021 03:48:34 -0400
Subject: [PATCH 04/11] meson: let version get stale

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 06dad15756..a25a77bb98 100644
--- a/meson.build
+++ b/meson.build
@@ -190,7 +190,7 @@ xemu_version = custom_target('xemu-version-macro.h',
                              command: xemu_version_cmd,
                              capture: true,
                              build_by_default: true,
-                             build_always_stale: true)
+                             build_always_stale: false)
 
 #####################
 # Option validation #
@@ -3726,7 +3726,7 @@ qemu_version = custom_target('qemu-version.h',
                              command: qemu_version_cmd,
                              capture: true,
                              build_by_default: true,
-                             build_always_stale: true)
+                             build_always_stale: false)
 
 xemu_config = custom_target('xemu-config.h',
                             output: 'xemu-config.h',
-- 
2.52.0

