From b8f7064954cb2dfe9e65b12aea8b8a24a3ca7ec1 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 7 Oct 2021 07:24:08 -0400
Subject: [PATCH 6/7] not for upstream: remove trace-events-all

---
 trace/meson.build | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/trace/meson.build b/trace/meson.build
index 26b54714d5..626299dc64 100644
--- a/trace/meson.build
+++ b/trace/meson.build
@@ -58,14 +58,6 @@ foreach item : [ '.' ] + trace_events_subdirs + qapi_trace_events
   endif
 endforeach
 
-trace_events_all = custom_target('trace-events-all',
-                                 output: 'trace-events-all',
-                                 input: trace_events_files,
-                                 command: [ 'cat', '@INPUT@' ],
-                                 capture: true,
-                                 install: true,
-                                 install_dir: qemu_datadir)
-
 if 'ust' in get_option('trace_backends')
   trace_ust_all_h = custom_target('trace-ust-all.h',
                                   output: 'trace-ust-all.h',
-- 
2.41.0

