From c03b0649e6027cc693111eface997a5bcc84fb9f Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Thu, 7 Oct 2021 07:24:39 -0400
Subject: [PATCH 7/7] not for upstream: remove keymaps

---
 pc-bios/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pc-bios/meson.build b/pc-bios/meson.build
index f2b3259..d8d166e 100644
--- a/pc-bios/meson.build
+++ b/pc-bios/meson.build
@@ -88,4 +88,3 @@ if get_option('install_blobs')
 endif
 
 subdir('descriptors')
-subdir('keymaps')
-- 
2.35.1

