From 5937dfd4ddc13e9ac57036a100de6a3cef80b239 Mon Sep 17 00:00:00 2001
From: SuperSamus <40663462+SuperSamus@users.noreply.github.com>
Date: Mon, 13 Jun 2022 22:01:49 +0200
Subject: [PATCH] Fix icon install location

---
 bsnes/target-bsnes/GNUmakefile | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/bsnes/target-bsnes/GNUmakefile b/bsnes/target-bsnes/GNUmakefile
index 4c67bde5..c693a2ed 100644
--- a/bsnes/target-bsnes/GNUmakefile
+++ b/bsnes/target-bsnes/GNUmakefile
@@ -51,14 +51,16 @@ else ifeq ($(platform),macos)
 else ifneq ($(filter $(platform),linux bsd),)
 	mkdir -p $(prefix)/bin/
 	mkdir -p $(prefix)/share/applications/
-	mkdir -p $(prefix)/share/icons/
+	mkdir -p $(prefix)/share/icons/hicolor/256x256/apps/
+	mkdir -p $(prefix)/share/icons/hicolor/scalable/apps/
 	mkdir -p $(prefix)/share/$(name)/
 	mkdir -p $(prefix)/share/$(name)/Database/
 	mkdir -p $(prefix)/share/$(name)/Firmware/
 	mkdir -p $(prefix)/share/$(name)/Shaders/
 	cp out/$(name) $(prefix)/bin/$(name)
 	cp $(ui)/resource/$(name).desktop $(prefix)/share/applications/$(name).desktop
-	cp $(ui)/resource/$(name).png $(prefix)/share/icons/$(name).png
+	cp $(ui)/resource/$(name).png $(prefix)/share/icons/hicolor/256x256/apps/$(name).png
+	cp $(ui)/resource/$(name).svg $(prefix)/share/icons/hicolor/scalable/apps/$(name).svg
 	cp Database/* $(prefix)/share/$(name)/Database/
 	cp -r ../shaders/* $(prefix)/share/$(name)/Shaders/
 endif
@@ -72,6 +74,7 @@ else ifeq ($(platform),macos)
 else ifneq ($(filter $(platform),linux bsd),)
 	rm -f $(prefix)/bin/$(name)
 	rm -f $(prefix)/share/applications/$(name).desktop
-	rm -f $(prefix)/share/icons/$(name).png
+	rm -f $(prefix)/share/icons/hicolor/256x256/apps/$(name).png
+	rm -f $(prefix)/share/icons/hicolor/scalable/apps/$(name).svg
 	rm -rf $(prefix)/share/$(name)
 endif
