diff --git a/src/mkgcc b/src/mkgcc
index 58f4f31..c0894ab 100755
--- a/src/mkgcc
+++ b/src/mkgcc
@@ -1,4 +1,5 @@
 #!/bin/sh
-MYFLAGS=
-gcc $MYFLAGS -O9 -Wall -Wextra -Werror -fomit-frame-pointer "$1.c" -s -o "$1"
-
+MYFLAGS="${CFLAGS}"
+CMD="gcc $MYFLAGS -Wall -Wextra "$1.c" -o "$1""
+echo "$CMD"
+$CMD
diff --git a/src/mkmacppc b/src/mkmacppc
index 9525c1a..f22a8aa 100755
--- a/src/mkmacppc
+++ b/src/mkmacppc
@@ -1,4 +1,5 @@
 #!/bin/sh
-MYFLAGS="-arch ppc"
-gcc $MYFLAGS -O9 -Wall -Wextra -Werror -fomit-frame-pointer "$1.c" -s -o "$1"
-
+MYFLAGS="${CFLAGS} -arch ppc"
+CMD="gcc $MYFLAGS -Wall -Wextra "$1.c" -o "$1""
+echo "$CMD"
+$CMD
diff --git a/src/mkmacx64 b/src/mkmacx64
index 39d0406..7b60773 100755
--- a/src/mkmacx64
+++ b/src/mkmacx64
@@ -1,4 +1,5 @@
 #!/bin/sh
-MYFLAGS="-mmacosx-version-min=10.4 -arch x86_64"
-gcc $MYFLAGS -O9 -Wall -Wextra -Werror -fomit-frame-pointer "$1.c" -o "$1"
-
+MYFLAGS="${CFLAGS} -mmacosx-version-min=10.4 -arch x86_64"
+CMD="gcc $MYFLAGS -Wall -Wextra "$1.c" -o "$1""
+echo "$CMD"
+$CMD
diff --git a/src/mkmacx86 b/src/mkmacx86
index 5e4b136..2bab4ba 100755
--- a/src/mkmacx86
+++ b/src/mkmacx86
@@ -1,4 +1,5 @@
 #!/bin/sh
-MYFLAGS="-mmacosx-version-min=10.4 -arch i386"
-gcc $MYFLAGS -O9 -Wall -Wextra -Werror -fomit-frame-pointer "$1.c" -o "$1"
-
+MYFLAGS="${CFLAGS} -mmacosx-version-min=10.4 -arch i386"
+CMD="gcc $MYFLAGS -Wall -Wextra "$1.c" -o "$1""
+echo "$CMD"
+$CMD
